// @aws-sdk/client-rds/dist-es/commands/AddRoleToDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AddRoleToDBClusterMessage } from "../models/models_0";
import { deserializeAws_queryAddRoleToDBClusterCommand, serializeAws_queryAddRoleToDBClusterCommand, } from "../protocols/Aws_query";
var AddRoleToDBClusterCommand = (function (_super) {
    __extends(AddRoleToDBClusterCommand, _super);
    function AddRoleToDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AddRoleToDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "AddRoleToDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AddRoleToDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AddRoleToDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryAddRoleToDBClusterCommand(input, context);
    };
    AddRoleToDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryAddRoleToDBClusterCommand(output, context);
    };
    return AddRoleToDBClusterCommand;
}($Command));
export { AddRoleToDBClusterCommand };
