// @aws-sdk/client-rds/dist-es/commands/BacktrackDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BacktrackDBClusterMessage, DBClusterBacktrack } from "../models/models_0";
import { deserializeAws_queryBacktrackDBClusterCommand, serializeAws_queryBacktrackDBClusterCommand, } from "../protocols/Aws_query";
var BacktrackDBClusterCommand = (function (_super) {
    __extends(BacktrackDBClusterCommand, _super);
    function BacktrackDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BacktrackDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "BacktrackDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BacktrackDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterBacktrack.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BacktrackDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryBacktrackDBClusterCommand(input, context);
    };
    BacktrackDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryBacktrackDBClusterCommand(output, context);
    };
    return BacktrackDBClusterCommand;
}($Command));
export { BacktrackDBClusterCommand };
