// @aws-sdk/client-rds/dist-es/commands/CopyDBParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CopyDBParameterGroupMessage, CopyDBParameterGroupResult } from "../models/models_0";
import { deserializeAws_queryCopyDBParameterGroupCommand, serializeAws_queryCopyDBParameterGroupCommand, } from "../protocols/Aws_query";
var CopyDBParameterGroupCommand = (function (_super) {
    __extends(CopyDBParameterGroupCommand, _super);
    function CopyDBParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CopyDBParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CopyDBParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CopyDBParameterGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CopyDBParameterGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CopyDBParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCopyDBParameterGroupCommand(input, context);
    };
    CopyDBParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCopyDBParameterGroupCommand(output, context);
    };
    return CopyDBParameterGroupCommand;
}($Command));
export { CopyDBParameterGroupCommand };
