// @aws-sdk/client-rds/dist-es/commands/CopyOptionGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CopyOptionGroupMessage, CopyOptionGroupResult } from "../models/models_0";
import { deserializeAws_queryCopyOptionGroupCommand, serializeAws_queryCopyOptionGroupCommand, } from "../protocols/Aws_query";
var CopyOptionGroupCommand = (function (_super) {
    __extends(CopyOptionGroupCommand, _super);
    function CopyOptionGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CopyOptionGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CopyOptionGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CopyOptionGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CopyOptionGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CopyOptionGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCopyOptionGroupCommand(input, context);
    };
    CopyOptionGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCopyOptionGroupCommand(output, context);
    };
    return CopyOptionGroupCommand;
}($Command));
export { CopyOptionGroupCommand };
