// @aws-sdk/client-rds/dist-es/commands/CreateCustomDBEngineVersionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCustomDBEngineVersionMessage, DBEngineVersion } from "../models/models_0";
import { deserializeAws_queryCreateCustomDBEngineVersionCommand, serializeAws_queryCreateCustomDBEngineVersionCommand, } from "../protocols/Aws_query";
var CreateCustomDBEngineVersionCommand = (function (_super) {
    __extends(CreateCustomDBEngineVersionCommand, _super);
    function CreateCustomDBEngineVersionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCustomDBEngineVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateCustomDBEngineVersionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCustomDBEngineVersionMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBEngineVersion.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCustomDBEngineVersionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateCustomDBEngineVersionCommand(input, context);
    };
    CreateCustomDBEngineVersionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateCustomDBEngineVersionCommand(output, context);
    };
    return CreateCustomDBEngineVersionCommand;
}($Command));
export { CreateCustomDBEngineVersionCommand };
