// @aws-sdk/client-rds/dist-es/commands/CreateOptionGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateOptionGroupMessage, CreateOptionGroupResult } from "../models/models_0";
import { deserializeAws_queryCreateOptionGroupCommand, serializeAws_queryCreateOptionGroupCommand, } from "../protocols/Aws_query";
var CreateOptionGroupCommand = (function (_super) {
    __extends(CreateOptionGroupCommand, _super);
    function CreateOptionGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateOptionGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateOptionGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateOptionGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateOptionGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateOptionGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateOptionGroupCommand(input, context);
    };
    CreateOptionGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateOptionGroupCommand(output, context);
    };
    return CreateOptionGroupCommand;
}($Command));
export { CreateOptionGroupCommand };
