// @aws-sdk/client-rds/dist-es/commands/CreateGlobalClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateGlobalClusterMessage, CreateGlobalClusterResult } from "../models/models_0";
import { deserializeAws_queryCreateGlobalClusterCommand, serializeAws_queryCreateGlobalClusterCommand, } from "../protocols/Aws_query";
var CreateGlobalClusterCommand = (function (_super) {
    __extends(CreateGlobalClusterCommand, _super);
    function CreateGlobalClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateGlobalClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateGlobalClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateGlobalClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateGlobalClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateGlobalClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateGlobalClusterCommand(input, context);
    };
    CreateGlobalClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateGlobalClusterCommand(output, context);
    };
    return CreateGlobalClusterCommand;
}($Command));
export { CreateGlobalClusterCommand };
