// @aws-sdk/client-rds/dist-es/commands/CreateDBClusterEndpointCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBClusterEndpointMessage, DBClusterEndpoint } from "../models/models_0";
import { deserializeAws_queryCreateDBClusterEndpointCommand, serializeAws_queryCreateDBClusterEndpointCommand, } from "../protocols/Aws_query";
var CreateDBClusterEndpointCommand = (function (_super) {
    __extends(CreateDBClusterEndpointCommand, _super);
    function CreateDBClusterEndpointCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBClusterEndpointCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBClusterEndpointCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBClusterEndpointMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterEndpoint.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBClusterEndpointCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBClusterEndpointCommand(input, context);
    };
    CreateDBClusterEndpointCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBClusterEndpointCommand(output, context);
    };
    return CreateDBClusterEndpointCommand;
}($Command));
export { CreateDBClusterEndpointCommand };
