// @aws-sdk/client-rds/dist-es/commands/CreateDBClusterCommand.js
import { __extends } from "tslib";
import { getCrossRegionPresignedUrlPlugin } from "@aws-sdk/middleware-sdk-rds";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBClusterMessage, CreateDBClusterResult } from "../models/models_0";
import { deserializeAws_queryCreateDBClusterCommand, serializeAws_queryCreateDBClusterCommand, } from "../protocols/Aws_query";
var CreateDBClusterCommand = (function (_super) {
    __extends(CreateDBClusterCommand, _super);
    function CreateDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getCrossRegionPresignedUrlPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBClusterCommand(input, context);
    };
    CreateDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBClusterCommand(output, context);
    };
    return CreateDBClusterCommand;
}($Command));
export { CreateDBClusterCommand };
