// @aws-sdk/client-rds/dist-es/commands/CreateDBInstanceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBInstanceMessage, CreateDBInstanceResult } from "../models/models_0";
import { deserializeAws_queryCreateDBInstanceCommand, serializeAws_queryCreateDBInstanceCommand, } from "../protocols/Aws_query";
var CreateDBInstanceCommand = (function (_super) {
    __extends(CreateDBInstanceCommand, _super);
    function CreateDBInstanceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBInstanceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBInstanceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBInstanceMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBInstanceResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBInstanceCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBInstanceCommand(input, context);
    };
    CreateDBInstanceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBInstanceCommand(output, context);
    };
    return CreateDBInstanceCommand;
}($Command));
export { CreateDBInstanceCommand };
