// @aws-sdk/client-rds/dist-es/commands/CreateDBProxyEndpointCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBProxyEndpointRequest, CreateDBProxyEndpointResponse } from "../models/models_0";
import { deserializeAws_queryCreateDBProxyEndpointCommand, serializeAws_queryCreateDBProxyEndpointCommand, } from "../protocols/Aws_query";
var CreateDBProxyEndpointCommand = (function (_super) {
    __extends(CreateDBProxyEndpointCommand, _super);
    function CreateDBProxyEndpointCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBProxyEndpointCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBProxyEndpointCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBProxyEndpointRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBProxyEndpointResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBProxyEndpointCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBProxyEndpointCommand(input, context);
    };
    CreateDBProxyEndpointCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBProxyEndpointCommand(output, context);
    };
    return CreateDBProxyEndpointCommand;
}($Command));
export { CreateDBProxyEndpointCommand };
