// @aws-sdk/client-rds/dist-es/commands/CreateDBSecurityGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBSecurityGroupMessage, CreateDBSecurityGroupResult } from "../models/models_0";
import { deserializeAws_queryCreateDBSecurityGroupCommand, serializeAws_queryCreateDBSecurityGroupCommand, } from "../protocols/Aws_query";
var CreateDBSecurityGroupCommand = (function (_super) {
    __extends(CreateDBSecurityGroupCommand, _super);
    function CreateDBSecurityGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBSecurityGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBSecurityGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBSecurityGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBSecurityGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBSecurityGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBSecurityGroupCommand(input, context);
    };
    CreateDBSecurityGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBSecurityGroupCommand(output, context);
    };
    return CreateDBSecurityGroupCommand;
}($Command));
export { CreateDBSecurityGroupCommand };
