// @aws-sdk/client-rds/dist-es/commands/CreateCustomAvailabilityZoneCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCustomAvailabilityZoneMessage, CreateCustomAvailabilityZoneResult } from "../models/models_0";
import { deserializeAws_queryCreateCustomAvailabilityZoneCommand, serializeAws_queryCreateCustomAvailabilityZoneCommand, } from "../protocols/Aws_query";
var CreateCustomAvailabilityZoneCommand = (function (_super) {
    __extends(CreateCustomAvailabilityZoneCommand, _super);
    function CreateCustomAvailabilityZoneCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCustomAvailabilityZoneCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateCustomAvailabilityZoneCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCustomAvailabilityZoneMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCustomAvailabilityZoneResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCustomAvailabilityZoneCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateCustomAvailabilityZoneCommand(input, context);
    };
    CreateCustomAvailabilityZoneCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateCustomAvailabilityZoneCommand(output, context);
    };
    return CreateCustomAvailabilityZoneCommand;
}($Command));
export { CreateCustomAvailabilityZoneCommand };
