// @aws-sdk/client-rds/dist-es/commands/CreateDBSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBSnapshotMessage, CreateDBSnapshotResult } from "../models/models_0";
import { deserializeAws_queryCreateDBSnapshotCommand, serializeAws_queryCreateDBSnapshotCommand, } from "../protocols/Aws_query";
var CreateDBSnapshotCommand = (function (_super) {
    __extends(CreateDBSnapshotCommand, _super);
    function CreateDBSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBSnapshotMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBSnapshotResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBSnapshotCommand(input, context);
    };
    CreateDBSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBSnapshotCommand(output, context);
    };
    return CreateDBSnapshotCommand;
}($Command));
export { CreateDBSnapshotCommand };
