// @aws-sdk/client-rds/dist-es/commands/CreateDBSubnetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBSubnetGroupMessage, CreateDBSubnetGroupResult } from "../models/models_0";
import { deserializeAws_queryCreateDBSubnetGroupCommand, serializeAws_queryCreateDBSubnetGroupCommand, } from "../protocols/Aws_query";
var CreateDBSubnetGroupCommand = (function (_super) {
    __extends(CreateDBSubnetGroupCommand, _super);
    function CreateDBSubnetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBSubnetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBSubnetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBSubnetGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBSubnetGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBSubnetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBSubnetGroupCommand(input, context);
    };
    CreateDBSubnetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBSubnetGroupCommand(output, context);
    };
    return CreateDBSubnetGroupCommand;
}($Command));
export { CreateDBSubnetGroupCommand };
