// @aws-sdk/client-rds/dist-es/commands/CreateDBInstanceReadReplicaCommand.js
import { __extends } from "tslib";
import { getCrossRegionPresignedUrlPlugin } from "@aws-sdk/middleware-sdk-rds";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDBInstanceReadReplicaMessage, CreateDBInstanceReadReplicaResult } from "../models/models_0";
import { deserializeAws_queryCreateDBInstanceReadReplicaCommand, serializeAws_queryCreateDBInstanceReadReplicaCommand, } from "../protocols/Aws_query";
var CreateDBInstanceReadReplicaCommand = (function (_super) {
    __extends(CreateDBInstanceReadReplicaCommand, _super);
    function CreateDBInstanceReadReplicaCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDBInstanceReadReplicaCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getCrossRegionPresignedUrlPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "CreateDBInstanceReadReplicaCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDBInstanceReadReplicaMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDBInstanceReadReplicaResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDBInstanceReadReplicaCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateDBInstanceReadReplicaCommand(input, context);
    };
    CreateDBInstanceReadReplicaCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateDBInstanceReadReplicaCommand(output, context);
    };
    return CreateDBInstanceReadReplicaCommand;
}($Command));
export { CreateDBInstanceReadReplicaCommand };
