// @aws-sdk/client-rds/dist-es/commands/DeleteDBSubnetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDBSubnetGroupMessage } from "../models/models_0";
import { deserializeAws_queryDeleteDBSubnetGroupCommand, serializeAws_queryDeleteDBSubnetGroupCommand, } from "../protocols/Aws_query";
var DeleteDBSubnetGroupCommand = (function (_super) {
    __extends(DeleteDBSubnetGroupCommand, _super);
    function DeleteDBSubnetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDBSubnetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteDBSubnetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDBSubnetGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDBSubnetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteDBSubnetGroupCommand(input, context);
    };
    DeleteDBSubnetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteDBSubnetGroupCommand(output, context);
    };
    return DeleteDBSubnetGroupCommand;
}($Command));
export { DeleteDBSubnetGroupCommand };
