// @aws-sdk/client-rds/dist-es/commands/DeleteOptionGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteOptionGroupMessage } from "../models/models_0";
import { deserializeAws_queryDeleteOptionGroupCommand, serializeAws_queryDeleteOptionGroupCommand, } from "../protocols/Aws_query";
var DeleteOptionGroupCommand = (function (_super) {
    __extends(DeleteOptionGroupCommand, _super);
    function DeleteOptionGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteOptionGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteOptionGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteOptionGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteOptionGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteOptionGroupCommand(input, context);
    };
    DeleteOptionGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteOptionGroupCommand(output, context);
    };
    return DeleteOptionGroupCommand;
}($Command));
export { DeleteOptionGroupCommand };
