// @aws-sdk/client-rds/dist-es/commands/DeleteDBClusterEndpointCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBClusterEndpoint, DeleteDBClusterEndpointMessage } from "../models/models_0";
import { deserializeAws_queryDeleteDBClusterEndpointCommand, serializeAws_queryDeleteDBClusterEndpointCommand, } from "../protocols/Aws_query";
var DeleteDBClusterEndpointCommand = (function (_super) {
    __extends(DeleteDBClusterEndpointCommand, _super);
    function DeleteDBClusterEndpointCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDBClusterEndpointCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteDBClusterEndpointCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDBClusterEndpointMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterEndpoint.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDBClusterEndpointCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteDBClusterEndpointCommand(input, context);
    };
    DeleteDBClusterEndpointCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteDBClusterEndpointCommand(output, context);
    };
    return DeleteDBClusterEndpointCommand;
}($Command));
export { DeleteDBClusterEndpointCommand };
