// @aws-sdk/client-rds/dist-es/commands/DeleteDBSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDBSnapshotMessage, DeleteDBSnapshotResult } from "../models/models_0";
import { deserializeAws_queryDeleteDBSnapshotCommand, serializeAws_queryDeleteDBSnapshotCommand, } from "../protocols/Aws_query";
var DeleteDBSnapshotCommand = (function (_super) {
    __extends(DeleteDBSnapshotCommand, _super);
    function DeleteDBSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDBSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteDBSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDBSnapshotMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteDBSnapshotResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDBSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteDBSnapshotCommand(input, context);
    };
    DeleteDBSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteDBSnapshotCommand(output, context);
    };
    return DeleteDBSnapshotCommand;
}($Command));
export { DeleteDBSnapshotCommand };
