// @aws-sdk/client-rds/dist-es/commands/DeleteDBProxyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDBProxyRequest, DeleteDBProxyResponse } from "../models/models_0";
import { deserializeAws_queryDeleteDBProxyCommand, serializeAws_queryDeleteDBProxyCommand, } from "../protocols/Aws_query";
var DeleteDBProxyCommand = (function (_super) {
    __extends(DeleteDBProxyCommand, _super);
    function DeleteDBProxyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDBProxyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteDBProxyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDBProxyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteDBProxyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDBProxyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteDBProxyCommand(input, context);
    };
    DeleteDBProxyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteDBProxyCommand(output, context);
    };
    return DeleteDBProxyCommand;
}($Command));
export { DeleteDBProxyCommand };
