// @aws-sdk/client-rds/dist-es/commands/DeleteGlobalClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteGlobalClusterMessage, DeleteGlobalClusterResult } from "../models/models_0";
import { deserializeAws_queryDeleteGlobalClusterCommand, serializeAws_queryDeleteGlobalClusterCommand, } from "../protocols/Aws_query";
var DeleteGlobalClusterCommand = (function (_super) {
    __extends(DeleteGlobalClusterCommand, _super);
    function DeleteGlobalClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteGlobalClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteGlobalClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteGlobalClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteGlobalClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteGlobalClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteGlobalClusterCommand(input, context);
    };
    DeleteGlobalClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteGlobalClusterCommand(output, context);
    };
    return DeleteGlobalClusterCommand;
}($Command));
export { DeleteGlobalClusterCommand };
