// @aws-sdk/client-rds/dist-es/commands/DeleteDBClusterSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDBClusterSnapshotMessage, DeleteDBClusterSnapshotResult } from "../models/models_0";
import { deserializeAws_queryDeleteDBClusterSnapshotCommand, serializeAws_queryDeleteDBClusterSnapshotCommand, } from "../protocols/Aws_query";
var DeleteDBClusterSnapshotCommand = (function (_super) {
    __extends(DeleteDBClusterSnapshotCommand, _super);
    function DeleteDBClusterSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDBClusterSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteDBClusterSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDBClusterSnapshotMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteDBClusterSnapshotResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDBClusterSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteDBClusterSnapshotCommand(input, context);
    };
    DeleteDBClusterSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteDBClusterSnapshotCommand(output, context);
    };
    return DeleteDBClusterSnapshotCommand;
}($Command));
export { DeleteDBClusterSnapshotCommand };
