// @aws-sdk/client-rds/dist-es/commands/DeleteDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDBClusterMessage, DeleteDBClusterResult } from "../models/models_0";
import { deserializeAws_queryDeleteDBClusterCommand, serializeAws_queryDeleteDBClusterCommand, } from "../protocols/Aws_query";
var DeleteDBClusterCommand = (function (_super) {
    __extends(DeleteDBClusterCommand, _super);
    function DeleteDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeleteDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteDBClusterCommand(input, context);
    };
    DeleteDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteDBClusterCommand(output, context);
    };
    return DeleteDBClusterCommand;
}($Command));
export { DeleteDBClusterCommand };
