// @aws-sdk/client-rds/dist-es/commands/DeregisterDBProxyTargetsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeregisterDBProxyTargetsRequest, DeregisterDBProxyTargetsResponse } from "../models/models_0";
import { deserializeAws_queryDeregisterDBProxyTargetsCommand, serializeAws_queryDeregisterDBProxyTargetsCommand, } from "../protocols/Aws_query";
var DeregisterDBProxyTargetsCommand = (function (_super) {
    __extends(DeregisterDBProxyTargetsCommand, _super);
    function DeregisterDBProxyTargetsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeregisterDBProxyTargetsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DeregisterDBProxyTargetsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeregisterDBProxyTargetsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeregisterDBProxyTargetsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeregisterDBProxyTargetsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeregisterDBProxyTargetsCommand(input, context);
    };
    DeregisterDBProxyTargetsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeregisterDBProxyTargetsCommand(output, context);
    };
    return DeregisterDBProxyTargetsCommand;
}($Command));
export { DeregisterDBProxyTargetsCommand };
