// @aws-sdk/client-rds/dist-es/commands/DescribeReservedDBInstancesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeReservedDBInstancesMessage, ReservedDBInstanceMessage } from "../models/models_1";
import { deserializeAws_queryDescribeReservedDBInstancesCommand, serializeAws_queryDescribeReservedDBInstancesCommand, } from "../protocols/Aws_query";
var DescribeReservedDBInstancesCommand = (function (_super) {
    __extends(DescribeReservedDBInstancesCommand, _super);
    function DescribeReservedDBInstancesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeReservedDBInstancesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeReservedDBInstancesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeReservedDBInstancesMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ReservedDBInstanceMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeReservedDBInstancesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeReservedDBInstancesCommand(input, context);
    };
    DescribeReservedDBInstancesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeReservedDBInstancesCommand(output, context);
    };
    return DescribeReservedDBInstancesCommand;
}($Command));
export { DescribeReservedDBInstancesCommand };
