// @aws-sdk/client-rds/dist-es/commands/DescribeDBParametersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBParameterGroupDetails, DescribeDBParametersMessage } from "../models/models_0";
import { deserializeAws_queryDescribeDBParametersCommand, serializeAws_queryDescribeDBParametersCommand, } from "../protocols/Aws_query";
var DescribeDBParametersCommand = (function (_super) {
    __extends(DescribeDBParametersCommand, _super);
    function DescribeDBParametersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBParametersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBParametersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBParametersMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBParameterGroupDetails.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBParametersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBParametersCommand(input, context);
    };
    DescribeDBParametersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBParametersCommand(output, context);
    };
    return DescribeDBParametersCommand;
}($Command));
export { DescribeDBParametersCommand };
