// @aws-sdk/client-rds/dist-es/commands/DescribeDBProxyEndpointsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeDBProxyEndpointsRequest, DescribeDBProxyEndpointsResponse } from "../models/models_0";
import { deserializeAws_queryDescribeDBProxyEndpointsCommand, serializeAws_queryDescribeDBProxyEndpointsCommand, } from "../protocols/Aws_query";
var DescribeDBProxyEndpointsCommand = (function (_super) {
    __extends(DescribeDBProxyEndpointsCommand, _super);
    function DescribeDBProxyEndpointsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBProxyEndpointsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBProxyEndpointsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBProxyEndpointsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDBProxyEndpointsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBProxyEndpointsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBProxyEndpointsCommand(input, context);
    };
    DescribeDBProxyEndpointsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBProxyEndpointsCommand(output, context);
    };
    return DescribeDBProxyEndpointsCommand;
}($Command));
export { DescribeDBProxyEndpointsCommand };
