// @aws-sdk/client-rds/dist-es/commands/DescribeEventSubscriptionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeEventSubscriptionsMessage, EventSubscriptionsMessage } from "../models/models_1";
import { deserializeAws_queryDescribeEventSubscriptionsCommand, serializeAws_queryDescribeEventSubscriptionsCommand, } from "../protocols/Aws_query";
var DescribeEventSubscriptionsCommand = (function (_super) {
    __extends(DescribeEventSubscriptionsCommand, _super);
    function DescribeEventSubscriptionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeEventSubscriptionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeEventSubscriptionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeEventSubscriptionsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: EventSubscriptionsMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeEventSubscriptionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeEventSubscriptionsCommand(input, context);
    };
    DescribeEventSubscriptionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeEventSubscriptionsCommand(output, context);
    };
    return DescribeEventSubscriptionsCommand;
}($Command));
export { DescribeEventSubscriptionsCommand };
