// @aws-sdk/client-rds/dist-es/commands/DescribeSourceRegionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeSourceRegionsMessage, SourceRegionMessage } from "../models/models_1";
import { deserializeAws_queryDescribeSourceRegionsCommand, serializeAws_queryDescribeSourceRegionsCommand, } from "../protocols/Aws_query";
var DescribeSourceRegionsCommand = (function (_super) {
    __extends(DescribeSourceRegionsCommand, _super);
    function DescribeSourceRegionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeSourceRegionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeSourceRegionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeSourceRegionsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: SourceRegionMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeSourceRegionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeSourceRegionsCommand(input, context);
    };
    DescribeSourceRegionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeSourceRegionsCommand(output, context);
    };
    return DescribeSourceRegionsCommand;
}($Command));
export { DescribeSourceRegionsCommand };
