// @aws-sdk/client-rds/dist-es/commands/DescribeDBSecurityGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBSecurityGroupMessage, DescribeDBSecurityGroupsMessage } from "../models/models_1";
import { deserializeAws_queryDescribeDBSecurityGroupsCommand, serializeAws_queryDescribeDBSecurityGroupsCommand, } from "../protocols/Aws_query";
var DescribeDBSecurityGroupsCommand = (function (_super) {
    __extends(DescribeDBSecurityGroupsCommand, _super);
    function DescribeDBSecurityGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBSecurityGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBSecurityGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBSecurityGroupsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBSecurityGroupMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBSecurityGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBSecurityGroupsCommand(input, context);
    };
    DescribeDBSecurityGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBSecurityGroupsCommand(output, context);
    };
    return DescribeDBSecurityGroupsCommand;
}($Command));
export { DescribeDBSecurityGroupsCommand };
