// @aws-sdk/client-rds/dist-es/commands/DescribeDBSnapshotsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBSnapshotMessage, DescribeDBSnapshotsMessage } from "../models/models_1";
import { deserializeAws_queryDescribeDBSnapshotsCommand, serializeAws_queryDescribeDBSnapshotsCommand, } from "../protocols/Aws_query";
var DescribeDBSnapshotsCommand = (function (_super) {
    __extends(DescribeDBSnapshotsCommand, _super);
    function DescribeDBSnapshotsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBSnapshotsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBSnapshotsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBSnapshotsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBSnapshotMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBSnapshotsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBSnapshotsCommand(input, context);
    };
    DescribeDBSnapshotsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBSnapshotsCommand(output, context);
    };
    return DescribeDBSnapshotsCommand;
}($Command));
export { DescribeDBSnapshotsCommand };
