// @aws-sdk/client-rds/dist-es/commands/DescribeDBEngineVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBEngineVersionMessage, DescribeDBEngineVersionsMessage } from "../models/models_0";
import { deserializeAws_queryDescribeDBEngineVersionsCommand, serializeAws_queryDescribeDBEngineVersionsCommand, } from "../protocols/Aws_query";
var DescribeDBEngineVersionsCommand = (function (_super) {
    __extends(DescribeDBEngineVersionsCommand, _super);
    function DescribeDBEngineVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBEngineVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBEngineVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBEngineVersionsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBEngineVersionMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBEngineVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBEngineVersionsCommand(input, context);
    };
    DescribeDBEngineVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBEngineVersionsCommand(output, context);
    };
    return DescribeDBEngineVersionsCommand;
}($Command));
export { DescribeDBEngineVersionsCommand };
