// @aws-sdk/client-rds/dist-es/commands/DescribeCertificatesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CertificateMessage, DescribeCertificatesMessage } from "../models/models_0";
import { deserializeAws_queryDescribeCertificatesCommand, serializeAws_queryDescribeCertificatesCommand, } from "../protocols/Aws_query";
var DescribeCertificatesCommand = (function (_super) {
    __extends(DescribeCertificatesCommand, _super);
    function DescribeCertificatesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeCertificatesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeCertificatesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeCertificatesMessage.filterSensitiveLog,
            outputFilterSensitiveLog: CertificateMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeCertificatesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeCertificatesCommand(input, context);
    };
    DescribeCertificatesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeCertificatesCommand(output, context);
    };
    return DescribeCertificatesCommand;
}($Command));
export { DescribeCertificatesCommand };
