// @aws-sdk/client-rds/dist-es/commands/DescribeDBParameterGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBParameterGroupsMessage, DescribeDBParameterGroupsMessage } from "../models/models_0";
import { deserializeAws_queryDescribeDBParameterGroupsCommand, serializeAws_queryDescribeDBParameterGroupsCommand, } from "../protocols/Aws_query";
var DescribeDBParameterGroupsCommand = (function (_super) {
    __extends(DescribeDBParameterGroupsCommand, _super);
    function DescribeDBParameterGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBParameterGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBParameterGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBParameterGroupsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBParameterGroupsMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBParameterGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBParameterGroupsCommand(input, context);
    };
    DescribeDBParameterGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBParameterGroupsCommand(output, context);
    };
    return DescribeDBParameterGroupsCommand;
}($Command));
export { DescribeDBParameterGroupsCommand };
