// @aws-sdk/client-rds/dist-es/commands/DescribeDBInstancesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBInstanceMessage, DescribeDBInstancesMessage } from "../models/models_0";
import { deserializeAws_queryDescribeDBInstancesCommand, serializeAws_queryDescribeDBInstancesCommand, } from "../protocols/Aws_query";
var DescribeDBInstancesCommand = (function (_super) {
    __extends(DescribeDBInstancesCommand, _super);
    function DescribeDBInstancesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBInstancesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBInstancesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBInstancesMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBInstanceMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBInstancesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBInstancesCommand(input, context);
    };
    DescribeDBInstancesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBInstancesCommand(output, context);
    };
    return DescribeDBInstancesCommand;
}($Command));
export { DescribeDBInstancesCommand };
