// @aws-sdk/client-rds/dist-es/commands/DescribeInstallationMediaCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeInstallationMediaMessage, InstallationMediaMessage } from "../models/models_1";
import { deserializeAws_queryDescribeInstallationMediaCommand, serializeAws_queryDescribeInstallationMediaCommand, } from "../protocols/Aws_query";
var DescribeInstallationMediaCommand = (function (_super) {
    __extends(DescribeInstallationMediaCommand, _super);
    function DescribeInstallationMediaCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeInstallationMediaCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeInstallationMediaCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeInstallationMediaMessage.filterSensitiveLog,
            outputFilterSensitiveLog: InstallationMediaMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeInstallationMediaCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeInstallationMediaCommand(input, context);
    };
    DescribeInstallationMediaCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeInstallationMediaCommand(output, context);
    };
    return DescribeInstallationMediaCommand;
}($Command));
export { DescribeInstallationMediaCommand };
