// @aws-sdk/client-rds/dist-es/commands/DescribeDBClusterSnapshotAttributesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeDBClusterSnapshotAttributesMessage, DescribeDBClusterSnapshotAttributesResult, } from "../models/models_0";
import { deserializeAws_queryDescribeDBClusterSnapshotAttributesCommand, serializeAws_queryDescribeDBClusterSnapshotAttributesCommand, } from "../protocols/Aws_query";
var DescribeDBClusterSnapshotAttributesCommand = (function (_super) {
    __extends(DescribeDBClusterSnapshotAttributesCommand, _super);
    function DescribeDBClusterSnapshotAttributesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBClusterSnapshotAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBClusterSnapshotAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBClusterSnapshotAttributesMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDBClusterSnapshotAttributesResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBClusterSnapshotAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBClusterSnapshotAttributesCommand(input, context);
    };
    DescribeDBClusterSnapshotAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBClusterSnapshotAttributesCommand(output, context);
    };
    return DescribeDBClusterSnapshotAttributesCommand;
}($Command));
export { DescribeDBClusterSnapshotAttributesCommand };
