// @aws-sdk/client-rds/dist-es/commands/DescribeDBClusterEndpointsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBClusterEndpointMessage, DescribeDBClusterEndpointsMessage } from "../models/models_0";
import { deserializeAws_queryDescribeDBClusterEndpointsCommand, serializeAws_queryDescribeDBClusterEndpointsCommand, } from "../protocols/Aws_query";
var DescribeDBClusterEndpointsCommand = (function (_super) {
    __extends(DescribeDBClusterEndpointsCommand, _super);
    function DescribeDBClusterEndpointsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDBClusterEndpointsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DescribeDBClusterEndpointsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDBClusterEndpointsMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterEndpointMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDBClusterEndpointsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDBClusterEndpointsCommand(input, context);
    };
    DescribeDBClusterEndpointsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDBClusterEndpointsCommand(output, context);
    };
    return DescribeDBClusterEndpointsCommand;
}($Command));
export { DescribeDBClusterEndpointsCommand };
