// @aws-sdk/client-rds/dist-es/commands/DownloadDBLogFilePortionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DownloadDBLogFilePortionDetails, DownloadDBLogFilePortionMessage } from "../models/models_1";
import { deserializeAws_queryDownloadDBLogFilePortionCommand, serializeAws_queryDownloadDBLogFilePortionCommand, } from "../protocols/Aws_query";
var DownloadDBLogFilePortionCommand = (function (_super) {
    __extends(DownloadDBLogFilePortionCommand, _super);
    function DownloadDBLogFilePortionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DownloadDBLogFilePortionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "DownloadDBLogFilePortionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DownloadDBLogFilePortionMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DownloadDBLogFilePortionDetails.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DownloadDBLogFilePortionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDownloadDBLogFilePortionCommand(input, context);
    };
    DownloadDBLogFilePortionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDownloadDBLogFilePortionCommand(output, context);
    };
    return DownloadDBLogFilePortionCommand;
}($Command));
export { DownloadDBLogFilePortionCommand };
