// @aws-sdk/client-rds/dist-es/commands/FailoverDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FailoverDBClusterMessage, FailoverDBClusterResult } from "../models/models_1";
import { deserializeAws_queryFailoverDBClusterCommand, serializeAws_queryFailoverDBClusterCommand, } from "../protocols/Aws_query";
var FailoverDBClusterCommand = (function (_super) {
    __extends(FailoverDBClusterCommand, _super);
    function FailoverDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    FailoverDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "FailoverDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: FailoverDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: FailoverDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    FailoverDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryFailoverDBClusterCommand(input, context);
    };
    FailoverDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryFailoverDBClusterCommand(output, context);
    };
    return FailoverDBClusterCommand;
}($Command));
export { FailoverDBClusterCommand };
