// @aws-sdk/client-rds/dist-es/commands/FailoverGlobalClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FailoverGlobalClusterMessage, FailoverGlobalClusterResult } from "../models/models_1";
import { deserializeAws_queryFailoverGlobalClusterCommand, serializeAws_queryFailoverGlobalClusterCommand, } from "../protocols/Aws_query";
var FailoverGlobalClusterCommand = (function (_super) {
    __extends(FailoverGlobalClusterCommand, _super);
    function FailoverGlobalClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    FailoverGlobalClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "FailoverGlobalClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: FailoverGlobalClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: FailoverGlobalClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    FailoverGlobalClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryFailoverGlobalClusterCommand(input, context);
    };
    FailoverGlobalClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryFailoverGlobalClusterCommand(output, context);
    };
    return FailoverGlobalClusterCommand;
}($Command));
export { FailoverGlobalClusterCommand };
