// @aws-sdk/client-rds/dist-es/commands/ModifyDBSubnetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBSubnetGroupMessage, ModifyDBSubnetGroupResult } from "../models/models_1";
import { deserializeAws_queryModifyDBSubnetGroupCommand, serializeAws_queryModifyDBSubnetGroupCommand, } from "../protocols/Aws_query";
var ModifyDBSubnetGroupCommand = (function (_super) {
    __extends(ModifyDBSubnetGroupCommand, _super);
    function ModifyDBSubnetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBSubnetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBSubnetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBSubnetGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBSubnetGroupResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBSubnetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBSubnetGroupCommand(input, context);
    };
    ModifyDBSubnetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBSubnetGroupCommand(output, context);
    };
    return ModifyDBSubnetGroupCommand;
}($Command));
export { ModifyDBSubnetGroupCommand };
