// @aws-sdk/client-rds/dist-es/commands/ModifyDBSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBSnapshotMessage, ModifyDBSnapshotResult } from "../models/models_1";
import { deserializeAws_queryModifyDBSnapshotCommand, serializeAws_queryModifyDBSnapshotCommand, } from "../protocols/Aws_query";
var ModifyDBSnapshotCommand = (function (_super) {
    __extends(ModifyDBSnapshotCommand, _super);
    function ModifyDBSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBSnapshotMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBSnapshotResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBSnapshotCommand(input, context);
    };
    ModifyDBSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBSnapshotCommand(output, context);
    };
    return ModifyDBSnapshotCommand;
}($Command));
export { ModifyDBSnapshotCommand };
