// @aws-sdk/client-rds/dist-es/commands/ModifyDBProxyTargetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBProxyTargetGroupRequest, ModifyDBProxyTargetGroupResponse } from "../models/models_1";
import { deserializeAws_queryModifyDBProxyTargetGroupCommand, serializeAws_queryModifyDBProxyTargetGroupCommand, } from "../protocols/Aws_query";
var ModifyDBProxyTargetGroupCommand = (function (_super) {
    __extends(ModifyDBProxyTargetGroupCommand, _super);
    function ModifyDBProxyTargetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBProxyTargetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBProxyTargetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBProxyTargetGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBProxyTargetGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBProxyTargetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBProxyTargetGroupCommand(input, context);
    };
    ModifyDBProxyTargetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBProxyTargetGroupCommand(output, context);
    };
    return ModifyDBProxyTargetGroupCommand;
}($Command));
export { ModifyDBProxyTargetGroupCommand };
