// @aws-sdk/client-rds/dist-es/commands/ModifyDBSnapshotAttributeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBSnapshotAttributeMessage, ModifyDBSnapshotAttributeResult } from "../models/models_1";
import { deserializeAws_queryModifyDBSnapshotAttributeCommand, serializeAws_queryModifyDBSnapshotAttributeCommand, } from "../protocols/Aws_query";
var ModifyDBSnapshotAttributeCommand = (function (_super) {
    __extends(ModifyDBSnapshotAttributeCommand, _super);
    function ModifyDBSnapshotAttributeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBSnapshotAttributeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBSnapshotAttributeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBSnapshotAttributeMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBSnapshotAttributeResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBSnapshotAttributeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBSnapshotAttributeCommand(input, context);
    };
    ModifyDBSnapshotAttributeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBSnapshotAttributeCommand(output, context);
    };
    return ModifyDBSnapshotAttributeCommand;
}($Command));
export { ModifyDBSnapshotAttributeCommand };
