// @aws-sdk/client-rds/dist-es/commands/ModifyCurrentDBClusterCapacityCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBClusterCapacityInfo, ModifyCurrentDBClusterCapacityMessage } from "../models/models_1";
import { deserializeAws_queryModifyCurrentDBClusterCapacityCommand, serializeAws_queryModifyCurrentDBClusterCapacityCommand, } from "../protocols/Aws_query";
var ModifyCurrentDBClusterCapacityCommand = (function (_super) {
    __extends(ModifyCurrentDBClusterCapacityCommand, _super);
    function ModifyCurrentDBClusterCapacityCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyCurrentDBClusterCapacityCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyCurrentDBClusterCapacityCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyCurrentDBClusterCapacityMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterCapacityInfo.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyCurrentDBClusterCapacityCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyCurrentDBClusterCapacityCommand(input, context);
    };
    ModifyCurrentDBClusterCapacityCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyCurrentDBClusterCapacityCommand(output, context);
    };
    return ModifyCurrentDBClusterCapacityCommand;
}($Command));
export { ModifyCurrentDBClusterCapacityCommand };
