// @aws-sdk/client-rds/dist-es/commands/ModifyEventSubscriptionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyEventSubscriptionMessage, ModifyEventSubscriptionResult } from "../models/models_1";
import { deserializeAws_queryModifyEventSubscriptionCommand, serializeAws_queryModifyEventSubscriptionCommand, } from "../protocols/Aws_query";
var ModifyEventSubscriptionCommand = (function (_super) {
    __extends(ModifyEventSubscriptionCommand, _super);
    function ModifyEventSubscriptionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyEventSubscriptionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyEventSubscriptionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyEventSubscriptionMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyEventSubscriptionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyEventSubscriptionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyEventSubscriptionCommand(input, context);
    };
    ModifyEventSubscriptionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyEventSubscriptionCommand(output, context);
    };
    return ModifyEventSubscriptionCommand;
}($Command));
export { ModifyEventSubscriptionCommand };
