// @aws-sdk/client-rds/dist-es/commands/ModifyDBParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBParameterGroupNameMessage, ModifyDBParameterGroupMessage } from "../models/models_1";
import { deserializeAws_queryModifyDBParameterGroupCommand, serializeAws_queryModifyDBParameterGroupCommand, } from "../protocols/Aws_query";
var ModifyDBParameterGroupCommand = (function (_super) {
    __extends(ModifyDBParameterGroupCommand, _super);
    function ModifyDBParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBParameterGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBParameterGroupNameMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBParameterGroupCommand(input, context);
    };
    ModifyDBParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBParameterGroupCommand(output, context);
    };
    return ModifyDBParameterGroupCommand;
}($Command));
export { ModifyDBParameterGroupCommand };
