// @aws-sdk/client-rds/dist-es/commands/ModifyDBProxyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBProxyRequest, ModifyDBProxyResponse } from "../models/models_1";
import { deserializeAws_queryModifyDBProxyCommand, serializeAws_queryModifyDBProxyCommand, } from "../protocols/Aws_query";
var ModifyDBProxyCommand = (function (_super) {
    __extends(ModifyDBProxyCommand, _super);
    function ModifyDBProxyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBProxyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBProxyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBProxyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBProxyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBProxyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBProxyCommand(input, context);
    };
    ModifyDBProxyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBProxyCommand(output, context);
    };
    return ModifyDBProxyCommand;
}($Command));
export { ModifyDBProxyCommand };
