// @aws-sdk/client-rds/dist-es/commands/ModifyDBProxyEndpointCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBProxyEndpointRequest, ModifyDBProxyEndpointResponse } from "../models/models_1";
import { deserializeAws_queryModifyDBProxyEndpointCommand, serializeAws_queryModifyDBProxyEndpointCommand, } from "../protocols/Aws_query";
var ModifyDBProxyEndpointCommand = (function (_super) {
    __extends(ModifyDBProxyEndpointCommand, _super);
    function ModifyDBProxyEndpointCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBProxyEndpointCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBProxyEndpointCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBProxyEndpointRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBProxyEndpointResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBProxyEndpointCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBProxyEndpointCommand(input, context);
    };
    ModifyDBProxyEndpointCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBProxyEndpointCommand(output, context);
    };
    return ModifyDBProxyEndpointCommand;
}($Command));
export { ModifyDBProxyEndpointCommand };
