// @aws-sdk/client-rds/dist-es/commands/ModifyDBInstanceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBInstanceMessage, ModifyDBInstanceResult } from "../models/models_1";
import { deserializeAws_queryModifyDBInstanceCommand, serializeAws_queryModifyDBInstanceCommand, } from "../protocols/Aws_query";
var ModifyDBInstanceCommand = (function (_super) {
    __extends(ModifyDBInstanceCommand, _super);
    function ModifyDBInstanceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBInstanceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBInstanceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBInstanceMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBInstanceResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBInstanceCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBInstanceCommand(input, context);
    };
    ModifyDBInstanceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBInstanceCommand(output, context);
    };
    return ModifyDBInstanceCommand;
}($Command));
export { ModifyDBInstanceCommand };
