// @aws-sdk/client-rds/dist-es/commands/ModifyDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyDBClusterMessage, ModifyDBClusterResult } from "../models/models_1";
import { deserializeAws_queryModifyDBClusterCommand, serializeAws_queryModifyDBClusterCommand, } from "../protocols/Aws_query";
var ModifyDBClusterCommand = (function (_super) {
    __extends(ModifyDBClusterCommand, _super);
    function ModifyDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ModifyDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBClusterCommand(input, context);
    };
    ModifyDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBClusterCommand(output, context);
    };
    return ModifyDBClusterCommand;
}($Command));
export { ModifyDBClusterCommand };
