// @aws-sdk/client-rds/dist-es/commands/ModifyDBClusterParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBClusterParameterGroupNameMessage, ModifyDBClusterParameterGroupMessage } from "../models/models_1";
import { deserializeAws_queryModifyDBClusterParameterGroupCommand, serializeAws_queryModifyDBClusterParameterGroupCommand, } from "../protocols/Aws_query";
var ModifyDBClusterParameterGroupCommand = (function (_super) {
    __extends(ModifyDBClusterParameterGroupCommand, _super);
    function ModifyDBClusterParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyDBClusterParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ModifyDBClusterParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyDBClusterParameterGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterParameterGroupNameMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyDBClusterParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyDBClusterParameterGroupCommand(input, context);
    };
    ModifyDBClusterParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyDBClusterParameterGroupCommand(output, context);
    };
    return ModifyDBClusterParameterGroupCommand;
}($Command));
export { ModifyDBClusterParameterGroupCommand };
