// @aws-sdk/client-rds/dist-es/commands/PromoteReadReplicaCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PromoteReadReplicaMessage, PromoteReadReplicaResult } from "../models/models_1";
import { deserializeAws_queryPromoteReadReplicaCommand, serializeAws_queryPromoteReadReplicaCommand, } from "../protocols/Aws_query";
var PromoteReadReplicaCommand = (function (_super) {
    __extends(PromoteReadReplicaCommand, _super);
    function PromoteReadReplicaCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PromoteReadReplicaCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "PromoteReadReplicaCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PromoteReadReplicaMessage.filterSensitiveLog,
            outputFilterSensitiveLog: PromoteReadReplicaResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PromoteReadReplicaCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPromoteReadReplicaCommand(input, context);
    };
    PromoteReadReplicaCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPromoteReadReplicaCommand(output, context);
    };
    return PromoteReadReplicaCommand;
}($Command));
export { PromoteReadReplicaCommand };
