// @aws-sdk/client-rds/dist-es/commands/RebootDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RebootDBClusterMessage, RebootDBClusterResult } from "../models/models_1";
import { deserializeAws_queryRebootDBClusterCommand, serializeAws_queryRebootDBClusterCommand, } from "../protocols/Aws_query";
var RebootDBClusterCommand = (function (_super) {
    __extends(RebootDBClusterCommand, _super);
    function RebootDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RebootDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "RebootDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RebootDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: RebootDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RebootDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRebootDBClusterCommand(input, context);
    };
    RebootDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRebootDBClusterCommand(output, context);
    };
    return RebootDBClusterCommand;
}($Command));
export { RebootDBClusterCommand };
