// @aws-sdk/client-rds/dist-es/commands/RemoveFromGlobalClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RemoveFromGlobalClusterMessage, RemoveFromGlobalClusterResult } from "../models/models_1";
import { deserializeAws_queryRemoveFromGlobalClusterCommand, serializeAws_queryRemoveFromGlobalClusterCommand, } from "../protocols/Aws_query";
var RemoveFromGlobalClusterCommand = (function (_super) {
    __extends(RemoveFromGlobalClusterCommand, _super);
    function RemoveFromGlobalClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RemoveFromGlobalClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "RemoveFromGlobalClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RemoveFromGlobalClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: RemoveFromGlobalClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RemoveFromGlobalClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRemoveFromGlobalClusterCommand(input, context);
    };
    RemoveFromGlobalClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRemoveFromGlobalClusterCommand(output, context);
    };
    return RemoveFromGlobalClusterCommand;
}($Command));
export { RemoveFromGlobalClusterCommand };
