// @aws-sdk/client-rds/dist-es/commands/ResetDBClusterParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DBClusterParameterGroupNameMessage, ResetDBClusterParameterGroupMessage } from "../models/models_1";
import { deserializeAws_queryResetDBClusterParameterGroupCommand, serializeAws_queryResetDBClusterParameterGroupCommand, } from "../protocols/Aws_query";
var ResetDBClusterParameterGroupCommand = (function (_super) {
    __extends(ResetDBClusterParameterGroupCommand, _super);
    function ResetDBClusterParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ResetDBClusterParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "ResetDBClusterParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResetDBClusterParameterGroupMessage.filterSensitiveLog,
            outputFilterSensitiveLog: DBClusterParameterGroupNameMessage.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResetDBClusterParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryResetDBClusterParameterGroupCommand(input, context);
    };
    ResetDBClusterParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryResetDBClusterParameterGroupCommand(output, context);
    };
    return ResetDBClusterParameterGroupCommand;
}($Command));
export { ResetDBClusterParameterGroupCommand };
