// @aws-sdk/client-rds/dist-es/commands/RestoreDBInstanceFromS3Command.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RestoreDBInstanceFromS3Message, RestoreDBInstanceFromS3Result } from "../models/models_1";
import { deserializeAws_queryRestoreDBInstanceFromS3Command, serializeAws_queryRestoreDBInstanceFromS3Command, } from "../protocols/Aws_query";
var RestoreDBInstanceFromS3Command = (function (_super) {
    __extends(RestoreDBInstanceFromS3Command, _super);
    function RestoreDBInstanceFromS3Command(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RestoreDBInstanceFromS3Command.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "RestoreDBInstanceFromS3Command";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RestoreDBInstanceFromS3Message.filterSensitiveLog,
            outputFilterSensitiveLog: RestoreDBInstanceFromS3Result.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RestoreDBInstanceFromS3Command.prototype.serialize = function (input, context) {
        return serializeAws_queryRestoreDBInstanceFromS3Command(input, context);
    };
    RestoreDBInstanceFromS3Command.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRestoreDBInstanceFromS3Command(output, context);
    };
    return RestoreDBInstanceFromS3Command;
}($Command));
export { RestoreDBInstanceFromS3Command };
