// @aws-sdk/client-rds/dist-es/commands/RestoreDBInstanceFromDBSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RestoreDBInstanceFromDBSnapshotMessage, RestoreDBInstanceFromDBSnapshotResult } from "../models/models_1";
import { deserializeAws_queryRestoreDBInstanceFromDBSnapshotCommand, serializeAws_queryRestoreDBInstanceFromDBSnapshotCommand, } from "../protocols/Aws_query";
var RestoreDBInstanceFromDBSnapshotCommand = (function (_super) {
    __extends(RestoreDBInstanceFromDBSnapshotCommand, _super);
    function RestoreDBInstanceFromDBSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RestoreDBInstanceFromDBSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "RestoreDBInstanceFromDBSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RestoreDBInstanceFromDBSnapshotMessage.filterSensitiveLog,
            outputFilterSensitiveLog: RestoreDBInstanceFromDBSnapshotResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RestoreDBInstanceFromDBSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRestoreDBInstanceFromDBSnapshotCommand(input, context);
    };
    RestoreDBInstanceFromDBSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRestoreDBInstanceFromDBSnapshotCommand(output, context);
    };
    return RestoreDBInstanceFromDBSnapshotCommand;
}($Command));
export { RestoreDBInstanceFromDBSnapshotCommand };
