// @aws-sdk/client-rds/dist-es/commands/RevokeDBSecurityGroupIngressCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RevokeDBSecurityGroupIngressMessage, RevokeDBSecurityGroupIngressResult } from "../models/models_1";
import { deserializeAws_queryRevokeDBSecurityGroupIngressCommand, serializeAws_queryRevokeDBSecurityGroupIngressCommand, } from "../protocols/Aws_query";
var RevokeDBSecurityGroupIngressCommand = (function (_super) {
    __extends(RevokeDBSecurityGroupIngressCommand, _super);
    function RevokeDBSecurityGroupIngressCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RevokeDBSecurityGroupIngressCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "RevokeDBSecurityGroupIngressCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RevokeDBSecurityGroupIngressMessage.filterSensitiveLog,
            outputFilterSensitiveLog: RevokeDBSecurityGroupIngressResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RevokeDBSecurityGroupIngressCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRevokeDBSecurityGroupIngressCommand(input, context);
    };
    RevokeDBSecurityGroupIngressCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRevokeDBSecurityGroupIngressCommand(output, context);
    };
    return RevokeDBSecurityGroupIngressCommand;
}($Command));
export { RevokeDBSecurityGroupIngressCommand };
