// @aws-sdk/client-rds/dist-es/commands/StartDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartDBClusterMessage, StartDBClusterResult } from "../models/models_1";
import { deserializeAws_queryStartDBClusterCommand, serializeAws_queryStartDBClusterCommand, } from "../protocols/Aws_query";
var StartDBClusterCommand = (function (_super) {
    __extends(StartDBClusterCommand, _super);
    function StartDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "StartDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: StartDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryStartDBClusterCommand(input, context);
    };
    StartDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryStartDBClusterCommand(output, context);
    };
    return StartDBClusterCommand;
}($Command));
export { StartDBClusterCommand };
