// @aws-sdk/client-rds/dist-es/commands/StartExportTaskCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExportTask } from "../models/models_0";
import { StartExportTaskMessage } from "../models/models_1";
import { deserializeAws_queryStartExportTaskCommand, serializeAws_queryStartExportTaskCommand, } from "../protocols/Aws_query";
var StartExportTaskCommand = (function (_super) {
    __extends(StartExportTaskCommand, _super);
    function StartExportTaskCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartExportTaskCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "StartExportTaskCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartExportTaskMessage.filterSensitiveLog,
            outputFilterSensitiveLog: ExportTask.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartExportTaskCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryStartExportTaskCommand(input, context);
    };
    StartExportTaskCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryStartExportTaskCommand(output, context);
    };
    return StartExportTaskCommand;
}($Command));
export { StartExportTaskCommand };
