// @aws-sdk/client-rds/dist-es/commands/StartActivityStreamCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartActivityStreamRequest, StartActivityStreamResponse } from "../models/models_1";
import { deserializeAws_queryStartActivityStreamCommand, serializeAws_queryStartActivityStreamCommand, } from "../protocols/Aws_query";
var StartActivityStreamCommand = (function (_super) {
    __extends(StartActivityStreamCommand, _super);
    function StartActivityStreamCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartActivityStreamCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "StartActivityStreamCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartActivityStreamRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartActivityStreamResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartActivityStreamCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryStartActivityStreamCommand(input, context);
    };
    StartActivityStreamCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryStartActivityStreamCommand(output, context);
    };
    return StartActivityStreamCommand;
}($Command));
export { StartActivityStreamCommand };
