// @aws-sdk/client-rds/dist-es/commands/StopDBInstanceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopDBInstanceMessage, StopDBInstanceResult } from "../models/models_1";
import { deserializeAws_queryStopDBInstanceCommand, serializeAws_queryStopDBInstanceCommand, } from "../protocols/Aws_query";
var StopDBInstanceCommand = (function (_super) {
    __extends(StopDBInstanceCommand, _super);
    function StopDBInstanceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopDBInstanceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "StopDBInstanceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopDBInstanceMessage.filterSensitiveLog,
            outputFilterSensitiveLog: StopDBInstanceResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopDBInstanceCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryStopDBInstanceCommand(input, context);
    };
    StopDBInstanceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryStopDBInstanceCommand(output, context);
    };
    return StopDBInstanceCommand;
}($Command));
export { StopDBInstanceCommand };
