// @aws-sdk/client-rds/dist-es/commands/StopDBClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopDBClusterMessage, StopDBClusterResult } from "../models/models_1";
import { deserializeAws_queryStopDBClusterCommand, serializeAws_queryStopDBClusterCommand, } from "../protocols/Aws_query";
var StopDBClusterCommand = (function (_super) {
    __extends(StopDBClusterCommand, _super);
    function StopDBClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopDBClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSClient";
        var commandName = "StopDBClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopDBClusterMessage.filterSensitiveLog,
            outputFilterSensitiveLog: StopDBClusterResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopDBClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryStopDBClusterCommand(input, context);
    };
    StopDBClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryStopDBClusterCommand(output, context);
    };
    return StopDBClusterCommand;
}($Command));
export { StopDBClusterCommand };
