// @aws-sdk/client-rds/dist-es/models/models_0.js
import { __assign } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
export var AccountQuota;
(function (AccountQuota) {
    AccountQuota.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccountQuota || (AccountQuota = {}));
export var AccountAttributesMessage;
(function (AccountAttributesMessage) {
    AccountAttributesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccountAttributesMessage || (AccountAttributesMessage = {}));
export var AddRoleToDBClusterMessage;
(function (AddRoleToDBClusterMessage) {
    AddRoleToDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddRoleToDBClusterMessage || (AddRoleToDBClusterMessage = {}));
export var DBClusterNotFoundFault;
(function (DBClusterNotFoundFault) {
    DBClusterNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterNotFoundFault || (DBClusterNotFoundFault = {}));
export var DBClusterRoleAlreadyExistsFault;
(function (DBClusterRoleAlreadyExistsFault) {
    DBClusterRoleAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterRoleAlreadyExistsFault || (DBClusterRoleAlreadyExistsFault = {}));
export var DBClusterRoleQuotaExceededFault;
(function (DBClusterRoleQuotaExceededFault) {
    DBClusterRoleQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterRoleQuotaExceededFault || (DBClusterRoleQuotaExceededFault = {}));
export var InvalidDBClusterStateFault;
(function (InvalidDBClusterStateFault) {
    InvalidDBClusterStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBClusterStateFault || (InvalidDBClusterStateFault = {}));
export var AddRoleToDBInstanceMessage;
(function (AddRoleToDBInstanceMessage) {
    AddRoleToDBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddRoleToDBInstanceMessage || (AddRoleToDBInstanceMessage = {}));
export var DBInstanceNotFoundFault;
(function (DBInstanceNotFoundFault) {
    DBInstanceNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceNotFoundFault || (DBInstanceNotFoundFault = {}));
export var DBInstanceRoleAlreadyExistsFault;
(function (DBInstanceRoleAlreadyExistsFault) {
    DBInstanceRoleAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceRoleAlreadyExistsFault || (DBInstanceRoleAlreadyExistsFault = {}));
export var DBInstanceRoleQuotaExceededFault;
(function (DBInstanceRoleQuotaExceededFault) {
    DBInstanceRoleQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceRoleQuotaExceededFault || (DBInstanceRoleQuotaExceededFault = {}));
export var InvalidDBInstanceStateFault;
(function (InvalidDBInstanceStateFault) {
    InvalidDBInstanceStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBInstanceStateFault || (InvalidDBInstanceStateFault = {}));
export var AddSourceIdentifierToSubscriptionMessage;
(function (AddSourceIdentifierToSubscriptionMessage) {
    AddSourceIdentifierToSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddSourceIdentifierToSubscriptionMessage || (AddSourceIdentifierToSubscriptionMessage = {}));
export var EventSubscription;
(function (EventSubscription) {
    EventSubscription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventSubscription || (EventSubscription = {}));
export var AddSourceIdentifierToSubscriptionResult;
(function (AddSourceIdentifierToSubscriptionResult) {
    AddSourceIdentifierToSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddSourceIdentifierToSubscriptionResult || (AddSourceIdentifierToSubscriptionResult = {}));
export var SourceNotFoundFault;
(function (SourceNotFoundFault) {
    SourceNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SourceNotFoundFault || (SourceNotFoundFault = {}));
export var SubscriptionNotFoundFault;
(function (SubscriptionNotFoundFault) {
    SubscriptionNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubscriptionNotFoundFault || (SubscriptionNotFoundFault = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var AddTagsToResourceMessage;
(function (AddTagsToResourceMessage) {
    AddTagsToResourceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddTagsToResourceMessage || (AddTagsToResourceMessage = {}));
export var DBProxyNotFoundFault;
(function (DBProxyNotFoundFault) {
    DBProxyNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyNotFoundFault || (DBProxyNotFoundFault = {}));
export var DBProxyTargetGroupNotFoundFault;
(function (DBProxyTargetGroupNotFoundFault) {
    DBProxyTargetGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyTargetGroupNotFoundFault || (DBProxyTargetGroupNotFoundFault = {}));
export var DBSnapshotNotFoundFault;
(function (DBSnapshotNotFoundFault) {
    DBSnapshotNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSnapshotNotFoundFault || (DBSnapshotNotFoundFault = {}));
export var ApplyPendingMaintenanceActionMessage;
(function (ApplyPendingMaintenanceActionMessage) {
    ApplyPendingMaintenanceActionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApplyPendingMaintenanceActionMessage || (ApplyPendingMaintenanceActionMessage = {}));
export var PendingMaintenanceAction;
(function (PendingMaintenanceAction) {
    PendingMaintenanceAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingMaintenanceAction || (PendingMaintenanceAction = {}));
export var ResourcePendingMaintenanceActions;
(function (ResourcePendingMaintenanceActions) {
    ResourcePendingMaintenanceActions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourcePendingMaintenanceActions || (ResourcePendingMaintenanceActions = {}));
export var ApplyPendingMaintenanceActionResult;
(function (ApplyPendingMaintenanceActionResult) {
    ApplyPendingMaintenanceActionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApplyPendingMaintenanceActionResult || (ApplyPendingMaintenanceActionResult = {}));
export var ResourceNotFoundFault;
(function (ResourceNotFoundFault) {
    ResourceNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceNotFoundFault || (ResourceNotFoundFault = {}));
export var AuthorizationAlreadyExistsFault;
(function (AuthorizationAlreadyExistsFault) {
    AuthorizationAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizationAlreadyExistsFault || (AuthorizationAlreadyExistsFault = {}));
export var AuthorizationQuotaExceededFault;
(function (AuthorizationQuotaExceededFault) {
    AuthorizationQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizationQuotaExceededFault || (AuthorizationQuotaExceededFault = {}));
export var AuthorizeDBSecurityGroupIngressMessage;
(function (AuthorizeDBSecurityGroupIngressMessage) {
    AuthorizeDBSecurityGroupIngressMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizeDBSecurityGroupIngressMessage || (AuthorizeDBSecurityGroupIngressMessage = {}));
export var EC2SecurityGroup;
(function (EC2SecurityGroup) {
    EC2SecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EC2SecurityGroup || (EC2SecurityGroup = {}));
export var IPRange;
(function (IPRange) {
    IPRange.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IPRange || (IPRange = {}));
export var DBSecurityGroup;
(function (DBSecurityGroup) {
    DBSecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroup || (DBSecurityGroup = {}));
export var AuthorizeDBSecurityGroupIngressResult;
(function (AuthorizeDBSecurityGroupIngressResult) {
    AuthorizeDBSecurityGroupIngressResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizeDBSecurityGroupIngressResult || (AuthorizeDBSecurityGroupIngressResult = {}));
export var DBSecurityGroupNotFoundFault;
(function (DBSecurityGroupNotFoundFault) {
    DBSecurityGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroupNotFoundFault || (DBSecurityGroupNotFoundFault = {}));
export var InvalidDBSecurityGroupStateFault;
(function (InvalidDBSecurityGroupStateFault) {
    InvalidDBSecurityGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBSecurityGroupStateFault || (InvalidDBSecurityGroupStateFault = {}));
export var BacktrackDBClusterMessage;
(function (BacktrackDBClusterMessage) {
    BacktrackDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BacktrackDBClusterMessage || (BacktrackDBClusterMessage = {}));
export var DBClusterBacktrack;
(function (DBClusterBacktrack) {
    DBClusterBacktrack.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterBacktrack || (DBClusterBacktrack = {}));
export var CancelExportTaskMessage;
(function (CancelExportTaskMessage) {
    CancelExportTaskMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelExportTaskMessage || (CancelExportTaskMessage = {}));
export var ExportTask;
(function (ExportTask) {
    ExportTask.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportTask || (ExportTask = {}));
export var ExportTaskNotFoundFault;
(function (ExportTaskNotFoundFault) {
    ExportTaskNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportTaskNotFoundFault || (ExportTaskNotFoundFault = {}));
export var InvalidExportTaskStateFault;
(function (InvalidExportTaskStateFault) {
    InvalidExportTaskStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidExportTaskStateFault || (InvalidExportTaskStateFault = {}));
export var CopyDBClusterParameterGroupMessage;
(function (CopyDBClusterParameterGroupMessage) {
    CopyDBClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterParameterGroupMessage || (CopyDBClusterParameterGroupMessage = {}));
export var DBClusterParameterGroup;
(function (DBClusterParameterGroup) {
    DBClusterParameterGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroup || (DBClusterParameterGroup = {}));
export var CopyDBClusterParameterGroupResult;
(function (CopyDBClusterParameterGroupResult) {
    CopyDBClusterParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterParameterGroupResult || (CopyDBClusterParameterGroupResult = {}));
export var DBParameterGroupAlreadyExistsFault;
(function (DBParameterGroupAlreadyExistsFault) {
    DBParameterGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupAlreadyExistsFault || (DBParameterGroupAlreadyExistsFault = {}));
export var DBParameterGroupNotFoundFault;
(function (DBParameterGroupNotFoundFault) {
    DBParameterGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupNotFoundFault || (DBParameterGroupNotFoundFault = {}));
export var DBParameterGroupQuotaExceededFault;
(function (DBParameterGroupQuotaExceededFault) {
    DBParameterGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupQuotaExceededFault || (DBParameterGroupQuotaExceededFault = {}));
export var CopyDBClusterSnapshotMessage;
(function (CopyDBClusterSnapshotMessage) {
    CopyDBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterSnapshotMessage || (CopyDBClusterSnapshotMessage = {}));
export var DBClusterSnapshot;
(function (DBClusterSnapshot) {
    DBClusterSnapshot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshot || (DBClusterSnapshot = {}));
export var CopyDBClusterSnapshotResult;
(function (CopyDBClusterSnapshotResult) {
    CopyDBClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterSnapshotResult || (CopyDBClusterSnapshotResult = {}));
export var DBClusterSnapshotAlreadyExistsFault;
(function (DBClusterSnapshotAlreadyExistsFault) {
    DBClusterSnapshotAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotAlreadyExistsFault || (DBClusterSnapshotAlreadyExistsFault = {}));
export var DBClusterSnapshotNotFoundFault;
(function (DBClusterSnapshotNotFoundFault) {
    DBClusterSnapshotNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotNotFoundFault || (DBClusterSnapshotNotFoundFault = {}));
export var InvalidDBClusterSnapshotStateFault;
(function (InvalidDBClusterSnapshotStateFault) {
    InvalidDBClusterSnapshotStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBClusterSnapshotStateFault || (InvalidDBClusterSnapshotStateFault = {}));
export var KMSKeyNotAccessibleFault;
(function (KMSKeyNotAccessibleFault) {
    KMSKeyNotAccessibleFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KMSKeyNotAccessibleFault || (KMSKeyNotAccessibleFault = {}));
export var SnapshotQuotaExceededFault;
(function (SnapshotQuotaExceededFault) {
    SnapshotQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotQuotaExceededFault || (SnapshotQuotaExceededFault = {}));
export var CopyDBParameterGroupMessage;
(function (CopyDBParameterGroupMessage) {
    CopyDBParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBParameterGroupMessage || (CopyDBParameterGroupMessage = {}));
export var DBParameterGroup;
(function (DBParameterGroup) {
    DBParameterGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroup || (DBParameterGroup = {}));
export var CopyDBParameterGroupResult;
(function (CopyDBParameterGroupResult) {
    CopyDBParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBParameterGroupResult || (CopyDBParameterGroupResult = {}));
export var CopyDBSnapshotMessage;
(function (CopyDBSnapshotMessage) {
    CopyDBSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBSnapshotMessage || (CopyDBSnapshotMessage = {}));
export var ProcessorFeature;
(function (ProcessorFeature) {
    ProcessorFeature.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProcessorFeature || (ProcessorFeature = {}));
export var DBSnapshot;
(function (DBSnapshot) {
    DBSnapshot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSnapshot || (DBSnapshot = {}));
export var CopyDBSnapshotResult;
(function (CopyDBSnapshotResult) {
    CopyDBSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBSnapshotResult || (CopyDBSnapshotResult = {}));
export var CustomAvailabilityZoneNotFoundFault;
(function (CustomAvailabilityZoneNotFoundFault) {
    CustomAvailabilityZoneNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomAvailabilityZoneNotFoundFault || (CustomAvailabilityZoneNotFoundFault = {}));
export var DBSnapshotAlreadyExistsFault;
(function (DBSnapshotAlreadyExistsFault) {
    DBSnapshotAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSnapshotAlreadyExistsFault || (DBSnapshotAlreadyExistsFault = {}));
export var InvalidDBSnapshotStateFault;
(function (InvalidDBSnapshotStateFault) {
    InvalidDBSnapshotStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBSnapshotStateFault || (InvalidDBSnapshotStateFault = {}));
export var CopyOptionGroupMessage;
(function (CopyOptionGroupMessage) {
    CopyOptionGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyOptionGroupMessage || (CopyOptionGroupMessage = {}));
export var DBSecurityGroupMembership;
(function (DBSecurityGroupMembership) {
    DBSecurityGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroupMembership || (DBSecurityGroupMembership = {}));
export var OptionSetting;
(function (OptionSetting) {
    OptionSetting.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionSetting || (OptionSetting = {}));
export var VpcSecurityGroupMembership;
(function (VpcSecurityGroupMembership) {
    VpcSecurityGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VpcSecurityGroupMembership || (VpcSecurityGroupMembership = {}));
export var Option;
(function (Option) {
    Option.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Option || (Option = {}));
export var OptionGroup;
(function (OptionGroup) {
    OptionGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroup || (OptionGroup = {}));
export var CopyOptionGroupResult;
(function (CopyOptionGroupResult) {
    CopyOptionGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyOptionGroupResult || (CopyOptionGroupResult = {}));
export var OptionGroupAlreadyExistsFault;
(function (OptionGroupAlreadyExistsFault) {
    OptionGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroupAlreadyExistsFault || (OptionGroupAlreadyExistsFault = {}));
export var OptionGroupNotFoundFault;
(function (OptionGroupNotFoundFault) {
    OptionGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroupNotFoundFault || (OptionGroupNotFoundFault = {}));
export var OptionGroupQuotaExceededFault;
(function (OptionGroupQuotaExceededFault) {
    OptionGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroupQuotaExceededFault || (OptionGroupQuotaExceededFault = {}));
export var CreateCustomAvailabilityZoneMessage;
(function (CreateCustomAvailabilityZoneMessage) {
    CreateCustomAvailabilityZoneMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCustomAvailabilityZoneMessage || (CreateCustomAvailabilityZoneMessage = {}));
export var VpnDetails;
(function (VpnDetails) {
    VpnDetails.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.VpnPSK && { VpnPSK: SENSITIVE_STRING }))); };
})(VpnDetails || (VpnDetails = {}));
export var CustomAvailabilityZone;
(function (CustomAvailabilityZone) {
    CustomAvailabilityZone.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.VpnDetails && { VpnDetails: VpnDetails.filterSensitiveLog(obj.VpnDetails) }))); };
})(CustomAvailabilityZone || (CustomAvailabilityZone = {}));
export var CreateCustomAvailabilityZoneResult;
(function (CreateCustomAvailabilityZoneResult) {
    CreateCustomAvailabilityZoneResult.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CustomAvailabilityZone && {
        CustomAvailabilityZone: CustomAvailabilityZone.filterSensitiveLog(obj.CustomAvailabilityZone),
    }))); };
})(CreateCustomAvailabilityZoneResult || (CreateCustomAvailabilityZoneResult = {}));
export var CustomAvailabilityZoneAlreadyExistsFault;
(function (CustomAvailabilityZoneAlreadyExistsFault) {
    CustomAvailabilityZoneAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomAvailabilityZoneAlreadyExistsFault || (CustomAvailabilityZoneAlreadyExistsFault = {}));
export var CustomAvailabilityZoneQuotaExceededFault;
(function (CustomAvailabilityZoneQuotaExceededFault) {
    CustomAvailabilityZoneQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomAvailabilityZoneQuotaExceededFault || (CustomAvailabilityZoneQuotaExceededFault = {}));
export var CreateCustomDBEngineVersionMessage;
(function (CreateCustomDBEngineVersionMessage) {
    CreateCustomDBEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCustomDBEngineVersionMessage || (CreateCustomDBEngineVersionMessage = {}));
export var CustomDBEngineVersionAlreadyExistsFault;
(function (CustomDBEngineVersionAlreadyExistsFault) {
    CustomDBEngineVersionAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomDBEngineVersionAlreadyExistsFault || (CustomDBEngineVersionAlreadyExistsFault = {}));
export var CustomDBEngineVersionQuotaExceededFault;
(function (CustomDBEngineVersionQuotaExceededFault) {
    CustomDBEngineVersionQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomDBEngineVersionQuotaExceededFault || (CustomDBEngineVersionQuotaExceededFault = {}));
export var CharacterSet;
(function (CharacterSet) {
    CharacterSet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CharacterSet || (CharacterSet = {}));
export var Timezone;
(function (Timezone) {
    Timezone.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Timezone || (Timezone = {}));
export var UpgradeTarget;
(function (UpgradeTarget) {
    UpgradeTarget.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpgradeTarget || (UpgradeTarget = {}));
export var DBEngineVersion;
(function (DBEngineVersion) {
    DBEngineVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBEngineVersion || (DBEngineVersion = {}));
export var ScalingConfiguration;
(function (ScalingConfiguration) {
    ScalingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingConfiguration || (ScalingConfiguration = {}));
export var CreateDBClusterMessage;
(function (CreateDBClusterMessage) {
    CreateDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterMessage || (CreateDBClusterMessage = {}));
export var DBClusterRole;
(function (DBClusterRole) {
    DBClusterRole.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterRole || (DBClusterRole = {}));
export var DBClusterMember;
(function (DBClusterMember) {
    DBClusterMember.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterMember || (DBClusterMember = {}));
export var DBClusterOptionGroupStatus;
(function (DBClusterOptionGroupStatus) {
    DBClusterOptionGroupStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterOptionGroupStatus || (DBClusterOptionGroupStatus = {}));
export var DomainMembership;
(function (DomainMembership) {
    DomainMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainMembership || (DomainMembership = {}));
export var WriteForwardingStatus;
(function (WriteForwardingStatus) {
    WriteForwardingStatus["DISABLED"] = "disabled";
    WriteForwardingStatus["DISABLING"] = "disabling";
    WriteForwardingStatus["ENABLED"] = "enabled";
    WriteForwardingStatus["ENABLING"] = "enabling";
    WriteForwardingStatus["UNKNOWN"] = "unknown";
})(WriteForwardingStatus || (WriteForwardingStatus = {}));
export var PendingCloudwatchLogsExports;
(function (PendingCloudwatchLogsExports) {
    PendingCloudwatchLogsExports.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingCloudwatchLogsExports || (PendingCloudwatchLogsExports = {}));
export var ClusterPendingModifiedValues;
(function (ClusterPendingModifiedValues) {
    ClusterPendingModifiedValues.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ClusterPendingModifiedValues || (ClusterPendingModifiedValues = {}));
export var ScalingConfigurationInfo;
(function (ScalingConfigurationInfo) {
    ScalingConfigurationInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingConfigurationInfo || (ScalingConfigurationInfo = {}));
export var DBCluster;
(function (DBCluster) {
    DBCluster.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBCluster || (DBCluster = {}));
export var CreateDBClusterResult;
(function (CreateDBClusterResult) {
    CreateDBClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterResult || (CreateDBClusterResult = {}));
export var DBClusterAlreadyExistsFault;
(function (DBClusterAlreadyExistsFault) {
    DBClusterAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterAlreadyExistsFault || (DBClusterAlreadyExistsFault = {}));
export var DBClusterParameterGroupNotFoundFault;
(function (DBClusterParameterGroupNotFoundFault) {
    DBClusterParameterGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroupNotFoundFault || (DBClusterParameterGroupNotFoundFault = {}));
export var DBClusterQuotaExceededFault;
(function (DBClusterQuotaExceededFault) {
    DBClusterQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterQuotaExceededFault || (DBClusterQuotaExceededFault = {}));
export var DBSubnetGroupDoesNotCoverEnoughAZs;
(function (DBSubnetGroupDoesNotCoverEnoughAZs) {
    DBSubnetGroupDoesNotCoverEnoughAZs.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroupDoesNotCoverEnoughAZs || (DBSubnetGroupDoesNotCoverEnoughAZs = {}));
export var DBSubnetGroupNotFoundFault;
(function (DBSubnetGroupNotFoundFault) {
    DBSubnetGroupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroupNotFoundFault || (DBSubnetGroupNotFoundFault = {}));
export var DomainNotFoundFault;
(function (DomainNotFoundFault) {
    DomainNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainNotFoundFault || (DomainNotFoundFault = {}));
export var GlobalClusterNotFoundFault;
(function (GlobalClusterNotFoundFault) {
    GlobalClusterNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalClusterNotFoundFault || (GlobalClusterNotFoundFault = {}));
export var InsufficientStorageClusterCapacityFault;
(function (InsufficientStorageClusterCapacityFault) {
    InsufficientStorageClusterCapacityFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsufficientStorageClusterCapacityFault || (InsufficientStorageClusterCapacityFault = {}));
export var InvalidDBSubnetGroupStateFault;
(function (InvalidDBSubnetGroupStateFault) {
    InvalidDBSubnetGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBSubnetGroupStateFault || (InvalidDBSubnetGroupStateFault = {}));
export var InvalidGlobalClusterStateFault;
(function (InvalidGlobalClusterStateFault) {
    InvalidGlobalClusterStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidGlobalClusterStateFault || (InvalidGlobalClusterStateFault = {}));
export var InvalidSubnet;
(function (InvalidSubnet) {
    InvalidSubnet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidSubnet || (InvalidSubnet = {}));
export var InvalidVPCNetworkStateFault;
(function (InvalidVPCNetworkStateFault) {
    InvalidVPCNetworkStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidVPCNetworkStateFault || (InvalidVPCNetworkStateFault = {}));
export var StorageQuotaExceededFault;
(function (StorageQuotaExceededFault) {
    StorageQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StorageQuotaExceededFault || (StorageQuotaExceededFault = {}));
export var CreateDBClusterEndpointMessage;
(function (CreateDBClusterEndpointMessage) {
    CreateDBClusterEndpointMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterEndpointMessage || (CreateDBClusterEndpointMessage = {}));
export var DBClusterEndpoint;
(function (DBClusterEndpoint) {
    DBClusterEndpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpoint || (DBClusterEndpoint = {}));
export var DBClusterEndpointAlreadyExistsFault;
(function (DBClusterEndpointAlreadyExistsFault) {
    DBClusterEndpointAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpointAlreadyExistsFault || (DBClusterEndpointAlreadyExistsFault = {}));
export var DBClusterEndpointQuotaExceededFault;
(function (DBClusterEndpointQuotaExceededFault) {
    DBClusterEndpointQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpointQuotaExceededFault || (DBClusterEndpointQuotaExceededFault = {}));
export var CreateDBClusterParameterGroupMessage;
(function (CreateDBClusterParameterGroupMessage) {
    CreateDBClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterParameterGroupMessage || (CreateDBClusterParameterGroupMessage = {}));
export var CreateDBClusterParameterGroupResult;
(function (CreateDBClusterParameterGroupResult) {
    CreateDBClusterParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterParameterGroupResult || (CreateDBClusterParameterGroupResult = {}));
export var CreateDBClusterSnapshotMessage;
(function (CreateDBClusterSnapshotMessage) {
    CreateDBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterSnapshotMessage || (CreateDBClusterSnapshotMessage = {}));
export var CreateDBClusterSnapshotResult;
(function (CreateDBClusterSnapshotResult) {
    CreateDBClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterSnapshotResult || (CreateDBClusterSnapshotResult = {}));
export var AuthorizationNotFoundFault;
(function (AuthorizationNotFoundFault) {
    AuthorizationNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizationNotFoundFault || (AuthorizationNotFoundFault = {}));
export var BackupPolicyNotFoundFault;
(function (BackupPolicyNotFoundFault) {
    BackupPolicyNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackupPolicyNotFoundFault || (BackupPolicyNotFoundFault = {}));
export var CreateDBInstanceMessage;
(function (CreateDBInstanceMessage) {
    CreateDBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceMessage || (CreateDBInstanceMessage = {}));
export var DBInstanceRole;
(function (DBInstanceRole) {
    DBInstanceRole.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceRole || (DBInstanceRole = {}));
export var AutomationMode;
(function (AutomationMode) {
    AutomationMode["ALL_PAUSED"] = "all-paused";
    AutomationMode["FULL"] = "full";
})(AutomationMode || (AutomationMode = {}));
export var DBInstanceAutomatedBackupsReplication;
(function (DBInstanceAutomatedBackupsReplication) {
    DBInstanceAutomatedBackupsReplication.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackupsReplication || (DBInstanceAutomatedBackupsReplication = {}));
export var DBParameterGroupStatus;
(function (DBParameterGroupStatus) {
    DBParameterGroupStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupStatus || (DBParameterGroupStatus = {}));
export var AvailabilityZone;
(function (AvailabilityZone) {
    AvailabilityZone.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AvailabilityZone || (AvailabilityZone = {}));
export var Outpost;
(function (Outpost) {
    Outpost.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Outpost || (Outpost = {}));
export var Subnet;
(function (Subnet) {
    Subnet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Subnet || (Subnet = {}));
export var DBSubnetGroup;
(function (DBSubnetGroup) {
    DBSubnetGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroup || (DBSubnetGroup = {}));
export var Endpoint;
(function (Endpoint) {
    Endpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Endpoint || (Endpoint = {}));
export var OptionGroupMembership;
(function (OptionGroupMembership) {
    OptionGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroupMembership || (OptionGroupMembership = {}));
export var PendingModifiedValues;
(function (PendingModifiedValues) {
    PendingModifiedValues.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingModifiedValues || (PendingModifiedValues = {}));
export var ReplicaMode;
(function (ReplicaMode) {
    ReplicaMode["MOUNTED"] = "mounted";
    ReplicaMode["OPEN_READ_ONLY"] = "open-read-only";
})(ReplicaMode || (ReplicaMode = {}));
export var DBInstanceStatusInfo;
(function (DBInstanceStatusInfo) {
    DBInstanceStatusInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceStatusInfo || (DBInstanceStatusInfo = {}));
export var DBInstance;
(function (DBInstance) {
    DBInstance.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstance || (DBInstance = {}));
export var CreateDBInstanceResult;
(function (CreateDBInstanceResult) {
    CreateDBInstanceResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceResult || (CreateDBInstanceResult = {}));
export var DBInstanceAlreadyExistsFault;
(function (DBInstanceAlreadyExistsFault) {
    DBInstanceAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAlreadyExistsFault || (DBInstanceAlreadyExistsFault = {}));
export var InstanceQuotaExceededFault;
(function (InstanceQuotaExceededFault) {
    InstanceQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceQuotaExceededFault || (InstanceQuotaExceededFault = {}));
export var InsufficientDBInstanceCapacityFault;
(function (InsufficientDBInstanceCapacityFault) {
    InsufficientDBInstanceCapacityFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InsufficientDBInstanceCapacityFault || (InsufficientDBInstanceCapacityFault = {}));
export var ProvisionedIopsNotAvailableInAZFault;
(function (ProvisionedIopsNotAvailableInAZFault) {
    ProvisionedIopsNotAvailableInAZFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisionedIopsNotAvailableInAZFault || (ProvisionedIopsNotAvailableInAZFault = {}));
export var StorageTypeNotSupportedFault;
(function (StorageTypeNotSupportedFault) {
    StorageTypeNotSupportedFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StorageTypeNotSupportedFault || (StorageTypeNotSupportedFault = {}));
export var CreateDBInstanceReadReplicaMessage;
(function (CreateDBInstanceReadReplicaMessage) {
    CreateDBInstanceReadReplicaMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceReadReplicaMessage || (CreateDBInstanceReadReplicaMessage = {}));
export var CreateDBInstanceReadReplicaResult;
(function (CreateDBInstanceReadReplicaResult) {
    CreateDBInstanceReadReplicaResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceReadReplicaResult || (CreateDBInstanceReadReplicaResult = {}));
export var DBSubnetGroupNotAllowedFault;
(function (DBSubnetGroupNotAllowedFault) {
    DBSubnetGroupNotAllowedFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroupNotAllowedFault || (DBSubnetGroupNotAllowedFault = {}));
export var InvalidDBSubnetGroupFault;
(function (InvalidDBSubnetGroupFault) {
    InvalidDBSubnetGroupFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBSubnetGroupFault || (InvalidDBSubnetGroupFault = {}));
export var CreateDBParameterGroupMessage;
(function (CreateDBParameterGroupMessage) {
    CreateDBParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBParameterGroupMessage || (CreateDBParameterGroupMessage = {}));
export var CreateDBParameterGroupResult;
(function (CreateDBParameterGroupResult) {
    CreateDBParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBParameterGroupResult || (CreateDBParameterGroupResult = {}));
export var AuthScheme;
(function (AuthScheme) {
    AuthScheme["SECRETS"] = "SECRETS";
})(AuthScheme || (AuthScheme = {}));
export var IAMAuthMode;
(function (IAMAuthMode) {
    IAMAuthMode["DISABLED"] = "DISABLED";
    IAMAuthMode["REQUIRED"] = "REQUIRED";
})(IAMAuthMode || (IAMAuthMode = {}));
export var UserAuthConfig;
(function (UserAuthConfig) {
    UserAuthConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserAuthConfig || (UserAuthConfig = {}));
export var EngineFamily;
(function (EngineFamily) {
    EngineFamily["MYSQL"] = "MYSQL";
    EngineFamily["POSTGRESQL"] = "POSTGRESQL";
})(EngineFamily || (EngineFamily = {}));
export var CreateDBProxyRequest;
(function (CreateDBProxyRequest) {
    CreateDBProxyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyRequest || (CreateDBProxyRequest = {}));
export var UserAuthConfigInfo;
(function (UserAuthConfigInfo) {
    UserAuthConfigInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserAuthConfigInfo || (UserAuthConfigInfo = {}));
export var DBProxyStatus;
(function (DBProxyStatus) {
    DBProxyStatus["AVAILABLE"] = "available";
    DBProxyStatus["CREATING"] = "creating";
    DBProxyStatus["DELETING"] = "deleting";
    DBProxyStatus["INCOMPATIBLE_NETWORK"] = "incompatible-network";
    DBProxyStatus["INSUFFICIENT_RESOURCE_LIMITS"] = "insufficient-resource-limits";
    DBProxyStatus["MODIFYING"] = "modifying";
    DBProxyStatus["REACTIVATING"] = "reactivating";
    DBProxyStatus["SUSPENDED"] = "suspended";
    DBProxyStatus["SUSPENDING"] = "suspending";
})(DBProxyStatus || (DBProxyStatus = {}));
export var DBProxy;
(function (DBProxy) {
    DBProxy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxy || (DBProxy = {}));
export var CreateDBProxyResponse;
(function (CreateDBProxyResponse) {
    CreateDBProxyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyResponse || (CreateDBProxyResponse = {}));
export var DBProxyAlreadyExistsFault;
(function (DBProxyAlreadyExistsFault) {
    DBProxyAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyAlreadyExistsFault || (DBProxyAlreadyExistsFault = {}));
export var DBProxyQuotaExceededFault;
(function (DBProxyQuotaExceededFault) {
    DBProxyQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyQuotaExceededFault || (DBProxyQuotaExceededFault = {}));
export var DBProxyEndpointTargetRole;
(function (DBProxyEndpointTargetRole) {
    DBProxyEndpointTargetRole["READ_ONLY"] = "READ_ONLY";
    DBProxyEndpointTargetRole["READ_WRITE"] = "READ_WRITE";
})(DBProxyEndpointTargetRole || (DBProxyEndpointTargetRole = {}));
export var CreateDBProxyEndpointRequest;
(function (CreateDBProxyEndpointRequest) {
    CreateDBProxyEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyEndpointRequest || (CreateDBProxyEndpointRequest = {}));
export var DBProxyEndpointStatus;
(function (DBProxyEndpointStatus) {
    DBProxyEndpointStatus["AVAILABLE"] = "available";
    DBProxyEndpointStatus["CREATING"] = "creating";
    DBProxyEndpointStatus["DELETING"] = "deleting";
    DBProxyEndpointStatus["INCOMPATIBLE_NETWORK"] = "incompatible-network";
    DBProxyEndpointStatus["INSUFFICIENT_RESOURCE_LIMITS"] = "insufficient-resource-limits";
    DBProxyEndpointStatus["MODIFYING"] = "modifying";
})(DBProxyEndpointStatus || (DBProxyEndpointStatus = {}));
export var DBProxyEndpoint;
(function (DBProxyEndpoint) {
    DBProxyEndpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyEndpoint || (DBProxyEndpoint = {}));
export var CreateDBProxyEndpointResponse;
(function (CreateDBProxyEndpointResponse) {
    CreateDBProxyEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyEndpointResponse || (CreateDBProxyEndpointResponse = {}));
export var DBProxyEndpointAlreadyExistsFault;
(function (DBProxyEndpointAlreadyExistsFault) {
    DBProxyEndpointAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyEndpointAlreadyExistsFault || (DBProxyEndpointAlreadyExistsFault = {}));
export var DBProxyEndpointQuotaExceededFault;
(function (DBProxyEndpointQuotaExceededFault) {
    DBProxyEndpointQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyEndpointQuotaExceededFault || (DBProxyEndpointQuotaExceededFault = {}));
export var InvalidDBProxyStateFault;
(function (InvalidDBProxyStateFault) {
    InvalidDBProxyStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBProxyStateFault || (InvalidDBProxyStateFault = {}));
export var CreateDBSecurityGroupMessage;
(function (CreateDBSecurityGroupMessage) {
    CreateDBSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSecurityGroupMessage || (CreateDBSecurityGroupMessage = {}));
export var CreateDBSecurityGroupResult;
(function (CreateDBSecurityGroupResult) {
    CreateDBSecurityGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSecurityGroupResult || (CreateDBSecurityGroupResult = {}));
export var DBSecurityGroupAlreadyExistsFault;
(function (DBSecurityGroupAlreadyExistsFault) {
    DBSecurityGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroupAlreadyExistsFault || (DBSecurityGroupAlreadyExistsFault = {}));
export var DBSecurityGroupNotSupportedFault;
(function (DBSecurityGroupNotSupportedFault) {
    DBSecurityGroupNotSupportedFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroupNotSupportedFault || (DBSecurityGroupNotSupportedFault = {}));
export var DBSecurityGroupQuotaExceededFault;
(function (DBSecurityGroupQuotaExceededFault) {
    DBSecurityGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroupQuotaExceededFault || (DBSecurityGroupQuotaExceededFault = {}));
export var CreateDBSnapshotMessage;
(function (CreateDBSnapshotMessage) {
    CreateDBSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSnapshotMessage || (CreateDBSnapshotMessage = {}));
export var CreateDBSnapshotResult;
(function (CreateDBSnapshotResult) {
    CreateDBSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSnapshotResult || (CreateDBSnapshotResult = {}));
export var CreateDBSubnetGroupMessage;
(function (CreateDBSubnetGroupMessage) {
    CreateDBSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSubnetGroupMessage || (CreateDBSubnetGroupMessage = {}));
export var CreateDBSubnetGroupResult;
(function (CreateDBSubnetGroupResult) {
    CreateDBSubnetGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSubnetGroupResult || (CreateDBSubnetGroupResult = {}));
export var DBSubnetGroupAlreadyExistsFault;
(function (DBSubnetGroupAlreadyExistsFault) {
    DBSubnetGroupAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroupAlreadyExistsFault || (DBSubnetGroupAlreadyExistsFault = {}));
export var DBSubnetGroupQuotaExceededFault;
(function (DBSubnetGroupQuotaExceededFault) {
    DBSubnetGroupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroupQuotaExceededFault || (DBSubnetGroupQuotaExceededFault = {}));
export var DBSubnetQuotaExceededFault;
(function (DBSubnetQuotaExceededFault) {
    DBSubnetQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetQuotaExceededFault || (DBSubnetQuotaExceededFault = {}));
export var CreateEventSubscriptionMessage;
(function (CreateEventSubscriptionMessage) {
    CreateEventSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventSubscriptionMessage || (CreateEventSubscriptionMessage = {}));
export var CreateEventSubscriptionResult;
(function (CreateEventSubscriptionResult) {
    CreateEventSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventSubscriptionResult || (CreateEventSubscriptionResult = {}));
export var EventSubscriptionQuotaExceededFault;
(function (EventSubscriptionQuotaExceededFault) {
    EventSubscriptionQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventSubscriptionQuotaExceededFault || (EventSubscriptionQuotaExceededFault = {}));
export var SNSInvalidTopicFault;
(function (SNSInvalidTopicFault) {
    SNSInvalidTopicFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SNSInvalidTopicFault || (SNSInvalidTopicFault = {}));
export var SNSNoAuthorizationFault;
(function (SNSNoAuthorizationFault) {
    SNSNoAuthorizationFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SNSNoAuthorizationFault || (SNSNoAuthorizationFault = {}));
export var SNSTopicArnNotFoundFault;
(function (SNSTopicArnNotFoundFault) {
    SNSTopicArnNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SNSTopicArnNotFoundFault || (SNSTopicArnNotFoundFault = {}));
export var SubscriptionAlreadyExistFault;
(function (SubscriptionAlreadyExistFault) {
    SubscriptionAlreadyExistFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubscriptionAlreadyExistFault || (SubscriptionAlreadyExistFault = {}));
export var SubscriptionCategoryNotFoundFault;
(function (SubscriptionCategoryNotFoundFault) {
    SubscriptionCategoryNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SubscriptionCategoryNotFoundFault || (SubscriptionCategoryNotFoundFault = {}));
export var CreateGlobalClusterMessage;
(function (CreateGlobalClusterMessage) {
    CreateGlobalClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalClusterMessage || (CreateGlobalClusterMessage = {}));
export var FailoverStatus;
(function (FailoverStatus) {
    FailoverStatus["CANCELLING"] = "cancelling";
    FailoverStatus["FAILING_OVER"] = "failing-over";
    FailoverStatus["PENDING"] = "pending";
})(FailoverStatus || (FailoverStatus = {}));
export var FailoverState;
(function (FailoverState) {
    FailoverState.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailoverState || (FailoverState = {}));
export var GlobalClusterMember;
(function (GlobalClusterMember) {
    GlobalClusterMember.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalClusterMember || (GlobalClusterMember = {}));
export var GlobalCluster;
(function (GlobalCluster) {
    GlobalCluster.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalCluster || (GlobalCluster = {}));
export var CreateGlobalClusterResult;
(function (CreateGlobalClusterResult) {
    CreateGlobalClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalClusterResult || (CreateGlobalClusterResult = {}));
export var GlobalClusterAlreadyExistsFault;
(function (GlobalClusterAlreadyExistsFault) {
    GlobalClusterAlreadyExistsFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalClusterAlreadyExistsFault || (GlobalClusterAlreadyExistsFault = {}));
export var GlobalClusterQuotaExceededFault;
(function (GlobalClusterQuotaExceededFault) {
    GlobalClusterQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalClusterQuotaExceededFault || (GlobalClusterQuotaExceededFault = {}));
export var CreateOptionGroupMessage;
(function (CreateOptionGroupMessage) {
    CreateOptionGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOptionGroupMessage || (CreateOptionGroupMessage = {}));
export var CreateOptionGroupResult;
(function (CreateOptionGroupResult) {
    CreateOptionGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOptionGroupResult || (CreateOptionGroupResult = {}));
export var DeleteCustomAvailabilityZoneMessage;
(function (DeleteCustomAvailabilityZoneMessage) {
    DeleteCustomAvailabilityZoneMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCustomAvailabilityZoneMessage || (DeleteCustomAvailabilityZoneMessage = {}));
export var DeleteCustomAvailabilityZoneResult;
(function (DeleteCustomAvailabilityZoneResult) {
    DeleteCustomAvailabilityZoneResult.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CustomAvailabilityZone && {
        CustomAvailabilityZone: CustomAvailabilityZone.filterSensitiveLog(obj.CustomAvailabilityZone),
    }))); };
})(DeleteCustomAvailabilityZoneResult || (DeleteCustomAvailabilityZoneResult = {}));
export var CustomDBEngineVersionNotFoundFault;
(function (CustomDBEngineVersionNotFoundFault) {
    CustomDBEngineVersionNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomDBEngineVersionNotFoundFault || (CustomDBEngineVersionNotFoundFault = {}));
export var DeleteCustomDBEngineVersionMessage;
(function (DeleteCustomDBEngineVersionMessage) {
    DeleteCustomDBEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCustomDBEngineVersionMessage || (DeleteCustomDBEngineVersionMessage = {}));
export var InvalidCustomDBEngineVersionStateFault;
(function (InvalidCustomDBEngineVersionStateFault) {
    InvalidCustomDBEngineVersionStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidCustomDBEngineVersionStateFault || (InvalidCustomDBEngineVersionStateFault = {}));
export var DeleteDBClusterMessage;
(function (DeleteDBClusterMessage) {
    DeleteDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterMessage || (DeleteDBClusterMessage = {}));
export var DeleteDBClusterResult;
(function (DeleteDBClusterResult) {
    DeleteDBClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterResult || (DeleteDBClusterResult = {}));
export var DBClusterEndpointNotFoundFault;
(function (DBClusterEndpointNotFoundFault) {
    DBClusterEndpointNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpointNotFoundFault || (DBClusterEndpointNotFoundFault = {}));
export var DeleteDBClusterEndpointMessage;
(function (DeleteDBClusterEndpointMessage) {
    DeleteDBClusterEndpointMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterEndpointMessage || (DeleteDBClusterEndpointMessage = {}));
export var InvalidDBClusterEndpointStateFault;
(function (InvalidDBClusterEndpointStateFault) {
    InvalidDBClusterEndpointStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBClusterEndpointStateFault || (InvalidDBClusterEndpointStateFault = {}));
export var DeleteDBClusterParameterGroupMessage;
(function (DeleteDBClusterParameterGroupMessage) {
    DeleteDBClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterParameterGroupMessage || (DeleteDBClusterParameterGroupMessage = {}));
export var InvalidDBParameterGroupStateFault;
(function (InvalidDBParameterGroupStateFault) {
    InvalidDBParameterGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBParameterGroupStateFault || (InvalidDBParameterGroupStateFault = {}));
export var DeleteDBClusterSnapshotMessage;
(function (DeleteDBClusterSnapshotMessage) {
    DeleteDBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterSnapshotMessage || (DeleteDBClusterSnapshotMessage = {}));
export var DeleteDBClusterSnapshotResult;
(function (DeleteDBClusterSnapshotResult) {
    DeleteDBClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterSnapshotResult || (DeleteDBClusterSnapshotResult = {}));
export var DBInstanceAutomatedBackupQuotaExceededFault;
(function (DBInstanceAutomatedBackupQuotaExceededFault) {
    DBInstanceAutomatedBackupQuotaExceededFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackupQuotaExceededFault || (DBInstanceAutomatedBackupQuotaExceededFault = {}));
export var DeleteDBInstanceMessage;
(function (DeleteDBInstanceMessage) {
    DeleteDBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceMessage || (DeleteDBInstanceMessage = {}));
export var DeleteDBInstanceResult;
(function (DeleteDBInstanceResult) {
    DeleteDBInstanceResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceResult || (DeleteDBInstanceResult = {}));
export var DBInstanceAutomatedBackupNotFoundFault;
(function (DBInstanceAutomatedBackupNotFoundFault) {
    DBInstanceAutomatedBackupNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackupNotFoundFault || (DBInstanceAutomatedBackupNotFoundFault = {}));
export var DeleteDBInstanceAutomatedBackupMessage;
(function (DeleteDBInstanceAutomatedBackupMessage) {
    DeleteDBInstanceAutomatedBackupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceAutomatedBackupMessage || (DeleteDBInstanceAutomatedBackupMessage = {}));
export var RestoreWindow;
(function (RestoreWindow) {
    RestoreWindow.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreWindow || (RestoreWindow = {}));
export var DBInstanceAutomatedBackup;
(function (DBInstanceAutomatedBackup) {
    DBInstanceAutomatedBackup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackup || (DBInstanceAutomatedBackup = {}));
export var DeleteDBInstanceAutomatedBackupResult;
(function (DeleteDBInstanceAutomatedBackupResult) {
    DeleteDBInstanceAutomatedBackupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceAutomatedBackupResult || (DeleteDBInstanceAutomatedBackupResult = {}));
export var InvalidDBInstanceAutomatedBackupStateFault;
(function (InvalidDBInstanceAutomatedBackupStateFault) {
    InvalidDBInstanceAutomatedBackupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBInstanceAutomatedBackupStateFault || (InvalidDBInstanceAutomatedBackupStateFault = {}));
export var DeleteDBParameterGroupMessage;
(function (DeleteDBParameterGroupMessage) {
    DeleteDBParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBParameterGroupMessage || (DeleteDBParameterGroupMessage = {}));
export var DeleteDBProxyRequest;
(function (DeleteDBProxyRequest) {
    DeleteDBProxyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyRequest || (DeleteDBProxyRequest = {}));
export var DeleteDBProxyResponse;
(function (DeleteDBProxyResponse) {
    DeleteDBProxyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyResponse || (DeleteDBProxyResponse = {}));
export var DBProxyEndpointNotFoundFault;
(function (DBProxyEndpointNotFoundFault) {
    DBProxyEndpointNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyEndpointNotFoundFault || (DBProxyEndpointNotFoundFault = {}));
export var DeleteDBProxyEndpointRequest;
(function (DeleteDBProxyEndpointRequest) {
    DeleteDBProxyEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyEndpointRequest || (DeleteDBProxyEndpointRequest = {}));
export var DeleteDBProxyEndpointResponse;
(function (DeleteDBProxyEndpointResponse) {
    DeleteDBProxyEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyEndpointResponse || (DeleteDBProxyEndpointResponse = {}));
export var InvalidDBProxyEndpointStateFault;
(function (InvalidDBProxyEndpointStateFault) {
    InvalidDBProxyEndpointStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBProxyEndpointStateFault || (InvalidDBProxyEndpointStateFault = {}));
export var DeleteDBSecurityGroupMessage;
(function (DeleteDBSecurityGroupMessage) {
    DeleteDBSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSecurityGroupMessage || (DeleteDBSecurityGroupMessage = {}));
export var DeleteDBSnapshotMessage;
(function (DeleteDBSnapshotMessage) {
    DeleteDBSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSnapshotMessage || (DeleteDBSnapshotMessage = {}));
export var DeleteDBSnapshotResult;
(function (DeleteDBSnapshotResult) {
    DeleteDBSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSnapshotResult || (DeleteDBSnapshotResult = {}));
export var DeleteDBSubnetGroupMessage;
(function (DeleteDBSubnetGroupMessage) {
    DeleteDBSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSubnetGroupMessage || (DeleteDBSubnetGroupMessage = {}));
export var InvalidDBSubnetStateFault;
(function (InvalidDBSubnetStateFault) {
    InvalidDBSubnetStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidDBSubnetStateFault || (InvalidDBSubnetStateFault = {}));
export var DeleteEventSubscriptionMessage;
(function (DeleteEventSubscriptionMessage) {
    DeleteEventSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEventSubscriptionMessage || (DeleteEventSubscriptionMessage = {}));
export var DeleteEventSubscriptionResult;
(function (DeleteEventSubscriptionResult) {
    DeleteEventSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEventSubscriptionResult || (DeleteEventSubscriptionResult = {}));
export var InvalidEventSubscriptionStateFault;
(function (InvalidEventSubscriptionStateFault) {
    InvalidEventSubscriptionStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidEventSubscriptionStateFault || (InvalidEventSubscriptionStateFault = {}));
export var DeleteGlobalClusterMessage;
(function (DeleteGlobalClusterMessage) {
    DeleteGlobalClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalClusterMessage || (DeleteGlobalClusterMessage = {}));
export var DeleteGlobalClusterResult;
(function (DeleteGlobalClusterResult) {
    DeleteGlobalClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalClusterResult || (DeleteGlobalClusterResult = {}));
export var DeleteInstallationMediaMessage;
(function (DeleteInstallationMediaMessage) {
    DeleteInstallationMediaMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInstallationMediaMessage || (DeleteInstallationMediaMessage = {}));
export var InstallationMediaFailureCause;
(function (InstallationMediaFailureCause) {
    InstallationMediaFailureCause.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstallationMediaFailureCause || (InstallationMediaFailureCause = {}));
export var InstallationMedia;
(function (InstallationMedia) {
    InstallationMedia.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstallationMedia || (InstallationMedia = {}));
export var InstallationMediaNotFoundFault;
(function (InstallationMediaNotFoundFault) {
    InstallationMediaNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstallationMediaNotFoundFault || (InstallationMediaNotFoundFault = {}));
export var DeleteOptionGroupMessage;
(function (DeleteOptionGroupMessage) {
    DeleteOptionGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteOptionGroupMessage || (DeleteOptionGroupMessage = {}));
export var InvalidOptionGroupStateFault;
(function (InvalidOptionGroupStateFault) {
    InvalidOptionGroupStateFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidOptionGroupStateFault || (InvalidOptionGroupStateFault = {}));
export var DBProxyTargetNotFoundFault;
(function (DBProxyTargetNotFoundFault) {
    DBProxyTargetNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyTargetNotFoundFault || (DBProxyTargetNotFoundFault = {}));
export var DeregisterDBProxyTargetsRequest;
(function (DeregisterDBProxyTargetsRequest) {
    DeregisterDBProxyTargetsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeregisterDBProxyTargetsRequest || (DeregisterDBProxyTargetsRequest = {}));
export var DeregisterDBProxyTargetsResponse;
(function (DeregisterDBProxyTargetsResponse) {
    DeregisterDBProxyTargetsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeregisterDBProxyTargetsResponse || (DeregisterDBProxyTargetsResponse = {}));
export var DescribeAccountAttributesMessage;
(function (DescribeAccountAttributesMessage) {
    DescribeAccountAttributesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountAttributesMessage || (DescribeAccountAttributesMessage = {}));
export var Certificate;
(function (Certificate) {
    Certificate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Certificate || (Certificate = {}));
export var CertificateMessage;
(function (CertificateMessage) {
    CertificateMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CertificateMessage || (CertificateMessage = {}));
export var CertificateNotFoundFault;
(function (CertificateNotFoundFault) {
    CertificateNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CertificateNotFoundFault || (CertificateNotFoundFault = {}));
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var DescribeCertificatesMessage;
(function (DescribeCertificatesMessage) {
    DescribeCertificatesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCertificatesMessage || (DescribeCertificatesMessage = {}));
export var CustomAvailabilityZoneMessage;
(function (CustomAvailabilityZoneMessage) {
    CustomAvailabilityZoneMessage.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CustomAvailabilityZones && {
        CustomAvailabilityZones: obj.CustomAvailabilityZones.map(function (item) {
            return CustomAvailabilityZone.filterSensitiveLog(item);
        }),
    }))); };
})(CustomAvailabilityZoneMessage || (CustomAvailabilityZoneMessage = {}));
export var DescribeCustomAvailabilityZonesMessage;
(function (DescribeCustomAvailabilityZonesMessage) {
    DescribeCustomAvailabilityZonesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCustomAvailabilityZonesMessage || (DescribeCustomAvailabilityZonesMessage = {}));
export var DBClusterBacktrackMessage;
(function (DBClusterBacktrackMessage) {
    DBClusterBacktrackMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterBacktrackMessage || (DBClusterBacktrackMessage = {}));
export var DBClusterBacktrackNotFoundFault;
(function (DBClusterBacktrackNotFoundFault) {
    DBClusterBacktrackNotFoundFault.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterBacktrackNotFoundFault || (DBClusterBacktrackNotFoundFault = {}));
export var DescribeDBClusterBacktracksMessage;
(function (DescribeDBClusterBacktracksMessage) {
    DescribeDBClusterBacktracksMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterBacktracksMessage || (DescribeDBClusterBacktracksMessage = {}));
export var DBClusterEndpointMessage;
(function (DBClusterEndpointMessage) {
    DBClusterEndpointMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpointMessage || (DBClusterEndpointMessage = {}));
export var DescribeDBClusterEndpointsMessage;
(function (DescribeDBClusterEndpointsMessage) {
    DescribeDBClusterEndpointsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterEndpointsMessage || (DescribeDBClusterEndpointsMessage = {}));
export var DBClusterParameterGroupsMessage;
(function (DBClusterParameterGroupsMessage) {
    DBClusterParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroupsMessage || (DBClusterParameterGroupsMessage = {}));
export var DescribeDBClusterParameterGroupsMessage;
(function (DescribeDBClusterParameterGroupsMessage) {
    DescribeDBClusterParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterParameterGroupsMessage || (DescribeDBClusterParameterGroupsMessage = {}));
export var Parameter;
(function (Parameter) {
    Parameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Parameter || (Parameter = {}));
export var DBClusterParameterGroupDetails;
(function (DBClusterParameterGroupDetails) {
    DBClusterParameterGroupDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroupDetails || (DBClusterParameterGroupDetails = {}));
export var DescribeDBClusterParametersMessage;
(function (DescribeDBClusterParametersMessage) {
    DescribeDBClusterParametersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterParametersMessage || (DescribeDBClusterParametersMessage = {}));
export var DBClusterMessage;
(function (DBClusterMessage) {
    DBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterMessage || (DBClusterMessage = {}));
export var DescribeDBClustersMessage;
(function (DescribeDBClustersMessage) {
    DescribeDBClustersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClustersMessage || (DescribeDBClustersMessage = {}));
export var DescribeDBClusterSnapshotAttributesMessage;
(function (DescribeDBClusterSnapshotAttributesMessage) {
    DescribeDBClusterSnapshotAttributesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterSnapshotAttributesMessage || (DescribeDBClusterSnapshotAttributesMessage = {}));
export var DBClusterSnapshotAttribute;
(function (DBClusterSnapshotAttribute) {
    DBClusterSnapshotAttribute.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotAttribute || (DBClusterSnapshotAttribute = {}));
export var DBClusterSnapshotAttributesResult;
(function (DBClusterSnapshotAttributesResult) {
    DBClusterSnapshotAttributesResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotAttributesResult || (DBClusterSnapshotAttributesResult = {}));
export var DescribeDBClusterSnapshotAttributesResult;
(function (DescribeDBClusterSnapshotAttributesResult) {
    DescribeDBClusterSnapshotAttributesResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterSnapshotAttributesResult || (DescribeDBClusterSnapshotAttributesResult = {}));
export var DBClusterSnapshotMessage;
(function (DBClusterSnapshotMessage) {
    DBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotMessage || (DBClusterSnapshotMessage = {}));
export var DescribeDBClusterSnapshotsMessage;
(function (DescribeDBClusterSnapshotsMessage) {
    DescribeDBClusterSnapshotsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterSnapshotsMessage || (DescribeDBClusterSnapshotsMessage = {}));
export var DBEngineVersionMessage;
(function (DBEngineVersionMessage) {
    DBEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBEngineVersionMessage || (DBEngineVersionMessage = {}));
export var DescribeDBEngineVersionsMessage;
(function (DescribeDBEngineVersionsMessage) {
    DescribeDBEngineVersionsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBEngineVersionsMessage || (DescribeDBEngineVersionsMessage = {}));
export var DBInstanceAutomatedBackupMessage;
(function (DBInstanceAutomatedBackupMessage) {
    DBInstanceAutomatedBackupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackupMessage || (DBInstanceAutomatedBackupMessage = {}));
export var DescribeDBInstanceAutomatedBackupsMessage;
(function (DescribeDBInstanceAutomatedBackupsMessage) {
    DescribeDBInstanceAutomatedBackupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBInstanceAutomatedBackupsMessage || (DescribeDBInstanceAutomatedBackupsMessage = {}));
export var DBInstanceMessage;
(function (DBInstanceMessage) {
    DBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceMessage || (DBInstanceMessage = {}));
export var DescribeDBInstancesMessage;
(function (DescribeDBInstancesMessage) {
    DescribeDBInstancesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBInstancesMessage || (DescribeDBInstancesMessage = {}));
export var DescribeDBLogFilesMessage;
(function (DescribeDBLogFilesMessage) {
    DescribeDBLogFilesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBLogFilesMessage || (DescribeDBLogFilesMessage = {}));
export var DescribeDBLogFilesDetails;
(function (DescribeDBLogFilesDetails) {
    DescribeDBLogFilesDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBLogFilesDetails || (DescribeDBLogFilesDetails = {}));
export var DescribeDBLogFilesResponse;
(function (DescribeDBLogFilesResponse) {
    DescribeDBLogFilesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBLogFilesResponse || (DescribeDBLogFilesResponse = {}));
export var DBParameterGroupsMessage;
(function (DBParameterGroupsMessage) {
    DBParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupsMessage || (DBParameterGroupsMessage = {}));
export var DescribeDBParameterGroupsMessage;
(function (DescribeDBParameterGroupsMessage) {
    DescribeDBParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBParameterGroupsMessage || (DescribeDBParameterGroupsMessage = {}));
export var DBParameterGroupDetails;
(function (DBParameterGroupDetails) {
    DBParameterGroupDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupDetails || (DBParameterGroupDetails = {}));
export var DescribeDBParametersMessage;
(function (DescribeDBParametersMessage) {
    DescribeDBParametersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBParametersMessage || (DescribeDBParametersMessage = {}));
export var DescribeDBProxiesRequest;
(function (DescribeDBProxiesRequest) {
    DescribeDBProxiesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxiesRequest || (DescribeDBProxiesRequest = {}));
export var DescribeDBProxiesResponse;
(function (DescribeDBProxiesResponse) {
    DescribeDBProxiesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxiesResponse || (DescribeDBProxiesResponse = {}));
export var DescribeDBProxyEndpointsRequest;
(function (DescribeDBProxyEndpointsRequest) {
    DescribeDBProxyEndpointsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyEndpointsRequest || (DescribeDBProxyEndpointsRequest = {}));
export var DescribeDBProxyEndpointsResponse;
(function (DescribeDBProxyEndpointsResponse) {
    DescribeDBProxyEndpointsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyEndpointsResponse || (DescribeDBProxyEndpointsResponse = {}));
export var DescribeDBProxyTargetGroupsRequest;
(function (DescribeDBProxyTargetGroupsRequest) {
    DescribeDBProxyTargetGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyTargetGroupsRequest || (DescribeDBProxyTargetGroupsRequest = {}));
export var ConnectionPoolConfigurationInfo;
(function (ConnectionPoolConfigurationInfo) {
    ConnectionPoolConfigurationInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionPoolConfigurationInfo || (ConnectionPoolConfigurationInfo = {}));
export var DBProxyTargetGroup;
(function (DBProxyTargetGroup) {
    DBProxyTargetGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyTargetGroup || (DBProxyTargetGroup = {}));
export var DescribeDBProxyTargetGroupsResponse;
(function (DescribeDBProxyTargetGroupsResponse) {
    DescribeDBProxyTargetGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyTargetGroupsResponse || (DescribeDBProxyTargetGroupsResponse = {}));
export var DescribeDBProxyTargetsRequest;
(function (DescribeDBProxyTargetsRequest) {
    DescribeDBProxyTargetsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyTargetsRequest || (DescribeDBProxyTargetsRequest = {}));
