// @aws-sdk/client-rds-data/dist-es/commands/BatchExecuteStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchExecuteStatementRequest, BatchExecuteStatementResponse } from "../models/models_0";
import { deserializeAws_restJson1BatchExecuteStatementCommand, serializeAws_restJson1BatchExecuteStatementCommand, } from "../protocols/Aws_restJson1";
var BatchExecuteStatementCommand = (function (_super) {
    __extends(BatchExecuteStatementCommand, _super);
    function BatchExecuteStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchExecuteStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSDataClient";
        var commandName = "BatchExecuteStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchExecuteStatementRequest.filterSensitiveLog,
            outputFilterSensitiveLog: BatchExecuteStatementResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchExecuteStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1BatchExecuteStatementCommand(input, context);
    };
    BatchExecuteStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1BatchExecuteStatementCommand(output, context);
    };
    return BatchExecuteStatementCommand;
}($Command));
export { BatchExecuteStatementCommand };
