// @aws-sdk/client-rds-data/dist-es/commands/BeginTransactionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BeginTransactionRequest, BeginTransactionResponse } from "../models/models_0";
import { deserializeAws_restJson1BeginTransactionCommand, serializeAws_restJson1BeginTransactionCommand, } from "../protocols/Aws_restJson1";
var BeginTransactionCommand = (function (_super) {
    __extends(BeginTransactionCommand, _super);
    function BeginTransactionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BeginTransactionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSDataClient";
        var commandName = "BeginTransactionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BeginTransactionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: BeginTransactionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BeginTransactionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1BeginTransactionCommand(input, context);
    };
    BeginTransactionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1BeginTransactionCommand(output, context);
    };
    return BeginTransactionCommand;
}($Command));
export { BeginTransactionCommand };
