// @aws-sdk/client-rds-data/dist-es/commands/CommitTransactionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CommitTransactionRequest, CommitTransactionResponse } from "../models/models_0";
import { deserializeAws_restJson1CommitTransactionCommand, serializeAws_restJson1CommitTransactionCommand, } from "../protocols/Aws_restJson1";
var CommitTransactionCommand = (function (_super) {
    __extends(CommitTransactionCommand, _super);
    function CommitTransactionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CommitTransactionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSDataClient";
        var commandName = "CommitTransactionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CommitTransactionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CommitTransactionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CommitTransactionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CommitTransactionCommand(input, context);
    };
    CommitTransactionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CommitTransactionCommand(output, context);
    };
    return CommitTransactionCommand;
}($Command));
export { CommitTransactionCommand };
