// @aws-sdk/client-rds-data/dist-es/commands/ExecuteStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExecuteStatementRequest, ExecuteStatementResponse } from "../models/models_0";
import { deserializeAws_restJson1ExecuteStatementCommand, serializeAws_restJson1ExecuteStatementCommand, } from "../protocols/Aws_restJson1";
var ExecuteStatementCommand = (function (_super) {
    __extends(ExecuteStatementCommand, _super);
    function ExecuteStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ExecuteStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSDataClient";
        var commandName = "ExecuteStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ExecuteStatementRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ExecuteStatementResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ExecuteStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ExecuteStatementCommand(input, context);
    };
    ExecuteStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ExecuteStatementCommand(output, context);
    };
    return ExecuteStatementCommand;
}($Command));
export { ExecuteStatementCommand };
