// @aws-sdk/client-rds-data/dist-es/commands/ExecuteSqlCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExecuteSqlRequest, ExecuteSqlResponse } from "../models/models_0";
import { deserializeAws_restJson1ExecuteSqlCommand, serializeAws_restJson1ExecuteSqlCommand, } from "../protocols/Aws_restJson1";
var ExecuteSqlCommand = (function (_super) {
    __extends(ExecuteSqlCommand, _super);
    function ExecuteSqlCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ExecuteSqlCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RDSDataClient";
        var commandName = "ExecuteSqlCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ExecuteSqlRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ExecuteSqlResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ExecuteSqlCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ExecuteSqlCommand(input, context);
    };
    ExecuteSqlCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ExecuteSqlCommand(output, context);
    };
    return ExecuteSqlCommand;
}($Command));
export { ExecuteSqlCommand };
