// @aws-sdk/client-sqs/dist-es/commands/ChangeMessageVisibilityBatchCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ChangeMessageVisibilityBatchRequest, ChangeMessageVisibilityBatchResult } from "../models/models_0";
import { deserializeAws_queryChangeMessageVisibilityBatchCommand, serializeAws_queryChangeMessageVisibilityBatchCommand, } from "../protocols/Aws_query";
var ChangeMessageVisibilityBatchCommand = (function (_super) {
    __extends(ChangeMessageVisibilityBatchCommand, _super);
    function ChangeMessageVisibilityBatchCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ChangeMessageVisibilityBatchCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SQSClient";
        var commandName = "ChangeMessageVisibilityBatchCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ChangeMessageVisibilityBatchRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ChangeMessageVisibilityBatchResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ChangeMessageVisibilityBatchCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryChangeMessageVisibilityBatchCommand(input, context);
    };
    ChangeMessageVisibilityBatchCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryChangeMessageVisibilityBatchCommand(output, context);
    };
    return ChangeMessageVisibilityBatchCommand;
}($Command));
export { ChangeMessageVisibilityBatchCommand };
