// @aws-sdk/client-sqs/dist-es/commands/SetQueueAttributesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetQueueAttributesRequest } from "../models/models_0";
import { deserializeAws_querySetQueueAttributesCommand, serializeAws_querySetQueueAttributesCommand, } from "../protocols/Aws_query";
var SetQueueAttributesCommand = (function (_super) {
    __extends(SetQueueAttributesCommand, _super);
    function SetQueueAttributesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetQueueAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SQSClient";
        var commandName = "SetQueueAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetQueueAttributesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetQueueAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetQueueAttributesCommand(input, context);
    };
    SetQueueAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetQueueAttributesCommand(output, context);
    };
    return SetQueueAttributesCommand;
}($Command));
export { SetQueueAttributesCommand };
