// @aws-sdk/client-sqs/dist-es/commands/TagQueueCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TagQueueRequest } from "../models/models_0";
import { deserializeAws_queryTagQueueCommand, serializeAws_queryTagQueueCommand } from "../protocols/Aws_query";
var TagQueueCommand = (function (_super) {
    __extends(TagQueueCommand, _super);
    function TagQueueCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TagQueueCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SQSClient";
        var commandName = "TagQueueCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TagQueueRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TagQueueCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryTagQueueCommand(input, context);
    };
    TagQueueCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryTagQueueCommand(output, context);
    };
    return TagQueueCommand;
}($Command));
export { TagQueueCommand };
