// @aws-sdk/client-rds/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DBClusterSnapshot = exports.CopyDBClusterSnapshotMessage = exports.DBParameterGroupQuotaExceededFault = exports.DBParameterGroupNotFoundFault = exports.DBParameterGroupAlreadyExistsFault = exports.CopyDBClusterParameterGroupResult = exports.DBClusterParameterGroup = exports.CopyDBClusterParameterGroupMessage = exports.InvalidExportTaskStateFault = exports.ExportTaskNotFoundFault = exports.ExportTask = exports.CancelExportTaskMessage = exports.DBClusterBacktrack = exports.BacktrackDBClusterMessage = exports.InvalidDBSecurityGroupStateFault = exports.DBSecurityGroupNotFoundFault = exports.AuthorizeDBSecurityGroupIngressResult = exports.DBSecurityGroup = exports.IPRange = exports.EC2SecurityGroup = exports.AuthorizeDBSecurityGroupIngressMessage = exports.AuthorizationQuotaExceededFault = exports.AuthorizationAlreadyExistsFault = exports.ResourceNotFoundFault = exports.ApplyPendingMaintenanceActionResult = exports.ResourcePendingMaintenanceActions = exports.PendingMaintenanceAction = exports.ApplyPendingMaintenanceActionMessage = exports.DBSnapshotNotFoundFault = exports.DBProxyTargetGroupNotFoundFault = exports.DBProxyNotFoundFault = exports.AddTagsToResourceMessage = exports.Tag = exports.SubscriptionNotFoundFault = exports.SourceNotFoundFault = exports.AddSourceIdentifierToSubscriptionResult = exports.EventSubscription = exports.AddSourceIdentifierToSubscriptionMessage = exports.InvalidDBInstanceStateFault = exports.DBInstanceRoleQuotaExceededFault = exports.DBInstanceRoleAlreadyExistsFault = exports.DBInstanceNotFoundFault = exports.AddRoleToDBInstanceMessage = exports.InvalidDBClusterStateFault = exports.DBClusterRoleQuotaExceededFault = exports.DBClusterRoleAlreadyExistsFault = exports.DBClusterNotFoundFault = exports.AddRoleToDBClusterMessage = exports.AccountAttributesMessage = exports.AccountQuota = void 0;
exports.DBCluster = exports.ScalingConfigurationInfo = exports.ClusterPendingModifiedValues = exports.PendingCloudwatchLogsExports = exports.WriteForwardingStatus = exports.DomainMembership = exports.DBClusterOptionGroupStatus = exports.DBClusterMember = exports.DBClusterRole = exports.CreateDBClusterMessage = exports.ScalingConfiguration = exports.DBEngineVersion = exports.UpgradeTarget = exports.Timezone = exports.CharacterSet = exports.CustomDBEngineVersionQuotaExceededFault = exports.CustomDBEngineVersionAlreadyExistsFault = exports.CreateCustomDBEngineVersionMessage = exports.CustomAvailabilityZoneQuotaExceededFault = exports.CustomAvailabilityZoneAlreadyExistsFault = exports.CreateCustomAvailabilityZoneResult = exports.CustomAvailabilityZone = exports.VpnDetails = exports.CreateCustomAvailabilityZoneMessage = exports.OptionGroupQuotaExceededFault = exports.OptionGroupNotFoundFault = exports.OptionGroupAlreadyExistsFault = exports.CopyOptionGroupResult = exports.OptionGroup = exports.Option = exports.VpcSecurityGroupMembership = exports.OptionSetting = exports.DBSecurityGroupMembership = exports.CopyOptionGroupMessage = exports.InvalidDBSnapshotStateFault = exports.DBSnapshotAlreadyExistsFault = exports.CustomAvailabilityZoneNotFoundFault = exports.CopyDBSnapshotResult = exports.DBSnapshot = exports.ProcessorFeature = exports.CopyDBSnapshotMessage = exports.CopyDBParameterGroupResult = exports.DBParameterGroup = exports.CopyDBParameterGroupMessage = exports.SnapshotQuotaExceededFault = exports.KMSKeyNotAccessibleFault = exports.InvalidDBClusterSnapshotStateFault = exports.DBClusterSnapshotNotFoundFault = exports.DBClusterSnapshotAlreadyExistsFault = exports.CopyDBClusterSnapshotResult = void 0;
exports.CreateDBParameterGroupMessage = exports.InvalidDBSubnetGroupFault = exports.DBSubnetGroupNotAllowedFault = exports.CreateDBInstanceReadReplicaResult = exports.CreateDBInstanceReadReplicaMessage = exports.StorageTypeNotSupportedFault = exports.ProvisionedIopsNotAvailableInAZFault = exports.InsufficientDBInstanceCapacityFault = exports.InstanceQuotaExceededFault = exports.DBInstanceAlreadyExistsFault = exports.CreateDBInstanceResult = exports.DBInstance = exports.DBInstanceStatusInfo = exports.ReplicaMode = exports.PendingModifiedValues = exports.OptionGroupMembership = exports.Endpoint = exports.DBSubnetGroup = exports.Subnet = exports.Outpost = exports.AvailabilityZone = exports.DBParameterGroupStatus = exports.DBInstanceAutomatedBackupsReplication = exports.AutomationMode = exports.DBInstanceRole = exports.CreateDBInstanceMessage = exports.BackupPolicyNotFoundFault = exports.AuthorizationNotFoundFault = exports.CreateDBClusterSnapshotResult = exports.CreateDBClusterSnapshotMessage = exports.CreateDBClusterParameterGroupResult = exports.CreateDBClusterParameterGroupMessage = exports.DBClusterEndpointQuotaExceededFault = exports.DBClusterEndpointAlreadyExistsFault = exports.DBClusterEndpoint = exports.CreateDBClusterEndpointMessage = exports.StorageQuotaExceededFault = exports.InvalidVPCNetworkStateFault = exports.InvalidSubnet = exports.InvalidGlobalClusterStateFault = exports.InvalidDBSubnetGroupStateFault = exports.InsufficientStorageClusterCapacityFault = exports.GlobalClusterNotFoundFault = exports.DomainNotFoundFault = exports.DBSubnetGroupNotFoundFault = exports.DBSubnetGroupDoesNotCoverEnoughAZs = exports.DBClusterQuotaExceededFault = exports.DBClusterParameterGroupNotFoundFault = exports.DBClusterAlreadyExistsFault = exports.CreateDBClusterResult = void 0;
exports.CreateOptionGroupResult = exports.CreateOptionGroupMessage = exports.GlobalClusterQuotaExceededFault = exports.GlobalClusterAlreadyExistsFault = exports.CreateGlobalClusterResult = exports.GlobalCluster = exports.GlobalClusterMember = exports.FailoverState = exports.FailoverStatus = exports.CreateGlobalClusterMessage = exports.SubscriptionCategoryNotFoundFault = exports.SubscriptionAlreadyExistFault = exports.SNSTopicArnNotFoundFault = exports.SNSNoAuthorizationFault = exports.SNSInvalidTopicFault = exports.EventSubscriptionQuotaExceededFault = exports.CreateEventSubscriptionResult = exports.CreateEventSubscriptionMessage = exports.DBSubnetQuotaExceededFault = exports.DBSubnetGroupQuotaExceededFault = exports.DBSubnetGroupAlreadyExistsFault = exports.CreateDBSubnetGroupResult = exports.CreateDBSubnetGroupMessage = exports.CreateDBSnapshotResult = exports.CreateDBSnapshotMessage = exports.DBSecurityGroupQuotaExceededFault = exports.DBSecurityGroupNotSupportedFault = exports.DBSecurityGroupAlreadyExistsFault = exports.CreateDBSecurityGroupResult = exports.CreateDBSecurityGroupMessage = exports.InvalidDBProxyStateFault = exports.DBProxyEndpointQuotaExceededFault = exports.DBProxyEndpointAlreadyExistsFault = exports.CreateDBProxyEndpointResponse = exports.DBProxyEndpoint = exports.DBProxyEndpointStatus = exports.CreateDBProxyEndpointRequest = exports.DBProxyEndpointTargetRole = exports.DBProxyQuotaExceededFault = exports.DBProxyAlreadyExistsFault = exports.CreateDBProxyResponse = exports.DBProxy = exports.DBProxyStatus = exports.UserAuthConfigInfo = exports.CreateDBProxyRequest = exports.EngineFamily = exports.UserAuthConfig = exports.IAMAuthMode = exports.AuthScheme = exports.CreateDBParameterGroupResult = void 0;
exports.DescribeAccountAttributesMessage = exports.DeregisterDBProxyTargetsResponse = exports.DeregisterDBProxyTargetsRequest = exports.DBProxyTargetNotFoundFault = exports.InvalidOptionGroupStateFault = exports.DeleteOptionGroupMessage = exports.InstallationMediaNotFoundFault = exports.InstallationMedia = exports.InstallationMediaFailureCause = exports.DeleteInstallationMediaMessage = exports.DeleteGlobalClusterResult = exports.DeleteGlobalClusterMessage = exports.InvalidEventSubscriptionStateFault = exports.DeleteEventSubscriptionResult = exports.DeleteEventSubscriptionMessage = exports.InvalidDBSubnetStateFault = exports.DeleteDBSubnetGroupMessage = exports.DeleteDBSnapshotResult = exports.DeleteDBSnapshotMessage = exports.DeleteDBSecurityGroupMessage = exports.InvalidDBProxyEndpointStateFault = exports.DeleteDBProxyEndpointResponse = exports.DeleteDBProxyEndpointRequest = exports.DBProxyEndpointNotFoundFault = exports.DeleteDBProxyResponse = exports.DeleteDBProxyRequest = exports.DeleteDBParameterGroupMessage = exports.InvalidDBInstanceAutomatedBackupStateFault = exports.DeleteDBInstanceAutomatedBackupResult = exports.DBInstanceAutomatedBackup = exports.RestoreWindow = exports.DeleteDBInstanceAutomatedBackupMessage = exports.DBInstanceAutomatedBackupNotFoundFault = exports.DeleteDBInstanceResult = exports.DeleteDBInstanceMessage = exports.DBInstanceAutomatedBackupQuotaExceededFault = exports.DeleteDBClusterSnapshotResult = exports.DeleteDBClusterSnapshotMessage = exports.InvalidDBParameterGroupStateFault = exports.DeleteDBClusterParameterGroupMessage = exports.InvalidDBClusterEndpointStateFault = exports.DeleteDBClusterEndpointMessage = exports.DBClusterEndpointNotFoundFault = exports.DeleteDBClusterResult = exports.DeleteDBClusterMessage = exports.InvalidCustomDBEngineVersionStateFault = exports.DeleteCustomDBEngineVersionMessage = exports.CustomDBEngineVersionNotFoundFault = exports.DeleteCustomAvailabilityZoneResult = exports.DeleteCustomAvailabilityZoneMessage = void 0;
exports.DescribeDBProxyTargetsRequest = exports.DescribeDBProxyTargetGroupsResponse = exports.DBProxyTargetGroup = exports.ConnectionPoolConfigurationInfo = exports.DescribeDBProxyTargetGroupsRequest = exports.DescribeDBProxyEndpointsResponse = exports.DescribeDBProxyEndpointsRequest = exports.DescribeDBProxiesResponse = exports.DescribeDBProxiesRequest = exports.DescribeDBParametersMessage = exports.DBParameterGroupDetails = exports.DescribeDBParameterGroupsMessage = exports.DBParameterGroupsMessage = exports.DescribeDBLogFilesResponse = exports.DescribeDBLogFilesDetails = exports.DescribeDBLogFilesMessage = exports.DescribeDBInstancesMessage = exports.DBInstanceMessage = exports.DescribeDBInstanceAutomatedBackupsMessage = exports.DBInstanceAutomatedBackupMessage = exports.DescribeDBEngineVersionsMessage = exports.DBEngineVersionMessage = exports.DescribeDBClusterSnapshotsMessage = exports.DBClusterSnapshotMessage = exports.DescribeDBClusterSnapshotAttributesResult = exports.DBClusterSnapshotAttributesResult = exports.DBClusterSnapshotAttribute = exports.DescribeDBClusterSnapshotAttributesMessage = exports.DescribeDBClustersMessage = exports.DBClusterMessage = exports.DescribeDBClusterParametersMessage = exports.DBClusterParameterGroupDetails = exports.Parameter = exports.DescribeDBClusterParameterGroupsMessage = exports.DBClusterParameterGroupsMessage = exports.DescribeDBClusterEndpointsMessage = exports.DBClusterEndpointMessage = exports.DescribeDBClusterBacktracksMessage = exports.DBClusterBacktrackNotFoundFault = exports.DBClusterBacktrackMessage = exports.DescribeCustomAvailabilityZonesMessage = exports.CustomAvailabilityZoneMessage = exports.DescribeCertificatesMessage = exports.Filter = exports.CertificateNotFoundFault = exports.CertificateMessage = exports.Certificate = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
var AccountQuota;
(function (AccountQuota) {
    AccountQuota.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountQuota = exports.AccountQuota || (exports.AccountQuota = {}));
var AccountAttributesMessage;
(function (AccountAttributesMessage) {
    AccountAttributesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountAttributesMessage = exports.AccountAttributesMessage || (exports.AccountAttributesMessage = {}));
var AddRoleToDBClusterMessage;
(function (AddRoleToDBClusterMessage) {
    AddRoleToDBClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddRoleToDBClusterMessage = exports.AddRoleToDBClusterMessage || (exports.AddRoleToDBClusterMessage = {}));
var DBClusterNotFoundFault;
(function (DBClusterNotFoundFault) {
    DBClusterNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterNotFoundFault = exports.DBClusterNotFoundFault || (exports.DBClusterNotFoundFault = {}));
var DBClusterRoleAlreadyExistsFault;
(function (DBClusterRoleAlreadyExistsFault) {
    DBClusterRoleAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterRoleAlreadyExistsFault = exports.DBClusterRoleAlreadyExistsFault || (exports.DBClusterRoleAlreadyExistsFault = {}));
var DBClusterRoleQuotaExceededFault;
(function (DBClusterRoleQuotaExceededFault) {
    DBClusterRoleQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterRoleQuotaExceededFault = exports.DBClusterRoleQuotaExceededFault || (exports.DBClusterRoleQuotaExceededFault = {}));
var InvalidDBClusterStateFault;
(function (InvalidDBClusterStateFault) {
    InvalidDBClusterStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBClusterStateFault = exports.InvalidDBClusterStateFault || (exports.InvalidDBClusterStateFault = {}));
var AddRoleToDBInstanceMessage;
(function (AddRoleToDBInstanceMessage) {
    AddRoleToDBInstanceMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddRoleToDBInstanceMessage = exports.AddRoleToDBInstanceMessage || (exports.AddRoleToDBInstanceMessage = {}));
var DBInstanceNotFoundFault;
(function (DBInstanceNotFoundFault) {
    DBInstanceNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceNotFoundFault = exports.DBInstanceNotFoundFault || (exports.DBInstanceNotFoundFault = {}));
var DBInstanceRoleAlreadyExistsFault;
(function (DBInstanceRoleAlreadyExistsFault) {
    DBInstanceRoleAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceRoleAlreadyExistsFault = exports.DBInstanceRoleAlreadyExistsFault || (exports.DBInstanceRoleAlreadyExistsFault = {}));
var DBInstanceRoleQuotaExceededFault;
(function (DBInstanceRoleQuotaExceededFault) {
    DBInstanceRoleQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceRoleQuotaExceededFault = exports.DBInstanceRoleQuotaExceededFault || (exports.DBInstanceRoleQuotaExceededFault = {}));
var InvalidDBInstanceStateFault;
(function (InvalidDBInstanceStateFault) {
    InvalidDBInstanceStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBInstanceStateFault = exports.InvalidDBInstanceStateFault || (exports.InvalidDBInstanceStateFault = {}));
var AddSourceIdentifierToSubscriptionMessage;
(function (AddSourceIdentifierToSubscriptionMessage) {
    AddSourceIdentifierToSubscriptionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddSourceIdentifierToSubscriptionMessage = exports.AddSourceIdentifierToSubscriptionMessage || (exports.AddSourceIdentifierToSubscriptionMessage = {}));
var EventSubscription;
(function (EventSubscription) {
    EventSubscription.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EventSubscription = exports.EventSubscription || (exports.EventSubscription = {}));
var AddSourceIdentifierToSubscriptionResult;
(function (AddSourceIdentifierToSubscriptionResult) {
    AddSourceIdentifierToSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddSourceIdentifierToSubscriptionResult = exports.AddSourceIdentifierToSubscriptionResult || (exports.AddSourceIdentifierToSubscriptionResult = {}));
var SourceNotFoundFault;
(function (SourceNotFoundFault) {
    SourceNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SourceNotFoundFault = exports.SourceNotFoundFault || (exports.SourceNotFoundFault = {}));
var SubscriptionNotFoundFault;
(function (SubscriptionNotFoundFault) {
    SubscriptionNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SubscriptionNotFoundFault = exports.SubscriptionNotFoundFault || (exports.SubscriptionNotFoundFault = {}));
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var AddTagsToResourceMessage;
(function (AddTagsToResourceMessage) {
    AddTagsToResourceMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AddTagsToResourceMessage = exports.AddTagsToResourceMessage || (exports.AddTagsToResourceMessage = {}));
var DBProxyNotFoundFault;
(function (DBProxyNotFoundFault) {
    DBProxyNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyNotFoundFault = exports.DBProxyNotFoundFault || (exports.DBProxyNotFoundFault = {}));
var DBProxyTargetGroupNotFoundFault;
(function (DBProxyTargetGroupNotFoundFault) {
    DBProxyTargetGroupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyTargetGroupNotFoundFault = exports.DBProxyTargetGroupNotFoundFault || (exports.DBProxyTargetGroupNotFoundFault = {}));
var DBSnapshotNotFoundFault;
(function (DBSnapshotNotFoundFault) {
    DBSnapshotNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSnapshotNotFoundFault = exports.DBSnapshotNotFoundFault || (exports.DBSnapshotNotFoundFault = {}));
var ApplyPendingMaintenanceActionMessage;
(function (ApplyPendingMaintenanceActionMessage) {
    ApplyPendingMaintenanceActionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ApplyPendingMaintenanceActionMessage = exports.ApplyPendingMaintenanceActionMessage || (exports.ApplyPendingMaintenanceActionMessage = {}));
var PendingMaintenanceAction;
(function (PendingMaintenanceAction) {
    PendingMaintenanceAction.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PendingMaintenanceAction = exports.PendingMaintenanceAction || (exports.PendingMaintenanceAction = {}));
var ResourcePendingMaintenanceActions;
(function (ResourcePendingMaintenanceActions) {
    ResourcePendingMaintenanceActions.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourcePendingMaintenanceActions = exports.ResourcePendingMaintenanceActions || (exports.ResourcePendingMaintenanceActions = {}));
var ApplyPendingMaintenanceActionResult;
(function (ApplyPendingMaintenanceActionResult) {
    ApplyPendingMaintenanceActionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ApplyPendingMaintenanceActionResult = exports.ApplyPendingMaintenanceActionResult || (exports.ApplyPendingMaintenanceActionResult = {}));
var ResourceNotFoundFault;
(function (ResourceNotFoundFault) {
    ResourceNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceNotFoundFault = exports.ResourceNotFoundFault || (exports.ResourceNotFoundFault = {}));
var AuthorizationAlreadyExistsFault;
(function (AuthorizationAlreadyExistsFault) {
    AuthorizationAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AuthorizationAlreadyExistsFault = exports.AuthorizationAlreadyExistsFault || (exports.AuthorizationAlreadyExistsFault = {}));
var AuthorizationQuotaExceededFault;
(function (AuthorizationQuotaExceededFault) {
    AuthorizationQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AuthorizationQuotaExceededFault = exports.AuthorizationQuotaExceededFault || (exports.AuthorizationQuotaExceededFault = {}));
var AuthorizeDBSecurityGroupIngressMessage;
(function (AuthorizeDBSecurityGroupIngressMessage) {
    AuthorizeDBSecurityGroupIngressMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AuthorizeDBSecurityGroupIngressMessage = exports.AuthorizeDBSecurityGroupIngressMessage || (exports.AuthorizeDBSecurityGroupIngressMessage = {}));
var EC2SecurityGroup;
(function (EC2SecurityGroup) {
    EC2SecurityGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EC2SecurityGroup = exports.EC2SecurityGroup || (exports.EC2SecurityGroup = {}));
var IPRange;
(function (IPRange) {
    IPRange.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(IPRange = exports.IPRange || (exports.IPRange = {}));
var DBSecurityGroup;
(function (DBSecurityGroup) {
    DBSecurityGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSecurityGroup = exports.DBSecurityGroup || (exports.DBSecurityGroup = {}));
var AuthorizeDBSecurityGroupIngressResult;
(function (AuthorizeDBSecurityGroupIngressResult) {
    AuthorizeDBSecurityGroupIngressResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AuthorizeDBSecurityGroupIngressResult = exports.AuthorizeDBSecurityGroupIngressResult || (exports.AuthorizeDBSecurityGroupIngressResult = {}));
var DBSecurityGroupNotFoundFault;
(function (DBSecurityGroupNotFoundFault) {
    DBSecurityGroupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSecurityGroupNotFoundFault = exports.DBSecurityGroupNotFoundFault || (exports.DBSecurityGroupNotFoundFault = {}));
var InvalidDBSecurityGroupStateFault;
(function (InvalidDBSecurityGroupStateFault) {
    InvalidDBSecurityGroupStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBSecurityGroupStateFault = exports.InvalidDBSecurityGroupStateFault || (exports.InvalidDBSecurityGroupStateFault = {}));
var BacktrackDBClusterMessage;
(function (BacktrackDBClusterMessage) {
    BacktrackDBClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BacktrackDBClusterMessage = exports.BacktrackDBClusterMessage || (exports.BacktrackDBClusterMessage = {}));
var DBClusterBacktrack;
(function (DBClusterBacktrack) {
    DBClusterBacktrack.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterBacktrack = exports.DBClusterBacktrack || (exports.DBClusterBacktrack = {}));
var CancelExportTaskMessage;
(function (CancelExportTaskMessage) {
    CancelExportTaskMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CancelExportTaskMessage = exports.CancelExportTaskMessage || (exports.CancelExportTaskMessage = {}));
var ExportTask;
(function (ExportTask) {
    ExportTask.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ExportTask = exports.ExportTask || (exports.ExportTask = {}));
var ExportTaskNotFoundFault;
(function (ExportTaskNotFoundFault) {
    ExportTaskNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ExportTaskNotFoundFault = exports.ExportTaskNotFoundFault || (exports.ExportTaskNotFoundFault = {}));
var InvalidExportTaskStateFault;
(function (InvalidExportTaskStateFault) {
    InvalidExportTaskStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidExportTaskStateFault = exports.InvalidExportTaskStateFault || (exports.InvalidExportTaskStateFault = {}));
var CopyDBClusterParameterGroupMessage;
(function (CopyDBClusterParameterGroupMessage) {
    CopyDBClusterParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBClusterParameterGroupMessage = exports.CopyDBClusterParameterGroupMessage || (exports.CopyDBClusterParameterGroupMessage = {}));
var DBClusterParameterGroup;
(function (DBClusterParameterGroup) {
    DBClusterParameterGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterParameterGroup = exports.DBClusterParameterGroup || (exports.DBClusterParameterGroup = {}));
var CopyDBClusterParameterGroupResult;
(function (CopyDBClusterParameterGroupResult) {
    CopyDBClusterParameterGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBClusterParameterGroupResult = exports.CopyDBClusterParameterGroupResult || (exports.CopyDBClusterParameterGroupResult = {}));
var DBParameterGroupAlreadyExistsFault;
(function (DBParameterGroupAlreadyExistsFault) {
    DBParameterGroupAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroupAlreadyExistsFault = exports.DBParameterGroupAlreadyExistsFault || (exports.DBParameterGroupAlreadyExistsFault = {}));
var DBParameterGroupNotFoundFault;
(function (DBParameterGroupNotFoundFault) {
    DBParameterGroupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroupNotFoundFault = exports.DBParameterGroupNotFoundFault || (exports.DBParameterGroupNotFoundFault = {}));
var DBParameterGroupQuotaExceededFault;
(function (DBParameterGroupQuotaExceededFault) {
    DBParameterGroupQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroupQuotaExceededFault = exports.DBParameterGroupQuotaExceededFault || (exports.DBParameterGroupQuotaExceededFault = {}));
var CopyDBClusterSnapshotMessage;
(function (CopyDBClusterSnapshotMessage) {
    CopyDBClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBClusterSnapshotMessage = exports.CopyDBClusterSnapshotMessage || (exports.CopyDBClusterSnapshotMessage = {}));
var DBClusterSnapshot;
(function (DBClusterSnapshot) {
    DBClusterSnapshot.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterSnapshot = exports.DBClusterSnapshot || (exports.DBClusterSnapshot = {}));
var CopyDBClusterSnapshotResult;
(function (CopyDBClusterSnapshotResult) {
    CopyDBClusterSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBClusterSnapshotResult = exports.CopyDBClusterSnapshotResult || (exports.CopyDBClusterSnapshotResult = {}));
var DBClusterSnapshotAlreadyExistsFault;
(function (DBClusterSnapshotAlreadyExistsFault) {
    DBClusterSnapshotAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterSnapshotAlreadyExistsFault = exports.DBClusterSnapshotAlreadyExistsFault || (exports.DBClusterSnapshotAlreadyExistsFault = {}));
var DBClusterSnapshotNotFoundFault;
(function (DBClusterSnapshotNotFoundFault) {
    DBClusterSnapshotNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterSnapshotNotFoundFault = exports.DBClusterSnapshotNotFoundFault || (exports.DBClusterSnapshotNotFoundFault = {}));
var InvalidDBClusterSnapshotStateFault;
(function (InvalidDBClusterSnapshotStateFault) {
    InvalidDBClusterSnapshotStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBClusterSnapshotStateFault = exports.InvalidDBClusterSnapshotStateFault || (exports.InvalidDBClusterSnapshotStateFault = {}));
var KMSKeyNotAccessibleFault;
(function (KMSKeyNotAccessibleFault) {
    KMSKeyNotAccessibleFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KMSKeyNotAccessibleFault = exports.KMSKeyNotAccessibleFault || (exports.KMSKeyNotAccessibleFault = {}));
var SnapshotQuotaExceededFault;
(function (SnapshotQuotaExceededFault) {
    SnapshotQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SnapshotQuotaExceededFault = exports.SnapshotQuotaExceededFault || (exports.SnapshotQuotaExceededFault = {}));
var CopyDBParameterGroupMessage;
(function (CopyDBParameterGroupMessage) {
    CopyDBParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBParameterGroupMessage = exports.CopyDBParameterGroupMessage || (exports.CopyDBParameterGroupMessage = {}));
var DBParameterGroup;
(function (DBParameterGroup) {
    DBParameterGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroup = exports.DBParameterGroup || (exports.DBParameterGroup = {}));
var CopyDBParameterGroupResult;
(function (CopyDBParameterGroupResult) {
    CopyDBParameterGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBParameterGroupResult = exports.CopyDBParameterGroupResult || (exports.CopyDBParameterGroupResult = {}));
var CopyDBSnapshotMessage;
(function (CopyDBSnapshotMessage) {
    CopyDBSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBSnapshotMessage = exports.CopyDBSnapshotMessage || (exports.CopyDBSnapshotMessage = {}));
var ProcessorFeature;
(function (ProcessorFeature) {
    ProcessorFeature.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ProcessorFeature = exports.ProcessorFeature || (exports.ProcessorFeature = {}));
var DBSnapshot;
(function (DBSnapshot) {
    DBSnapshot.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSnapshot = exports.DBSnapshot || (exports.DBSnapshot = {}));
var CopyDBSnapshotResult;
(function (CopyDBSnapshotResult) {
    CopyDBSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyDBSnapshotResult = exports.CopyDBSnapshotResult || (exports.CopyDBSnapshotResult = {}));
var CustomAvailabilityZoneNotFoundFault;
(function (CustomAvailabilityZoneNotFoundFault) {
    CustomAvailabilityZoneNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomAvailabilityZoneNotFoundFault = exports.CustomAvailabilityZoneNotFoundFault || (exports.CustomAvailabilityZoneNotFoundFault = {}));
var DBSnapshotAlreadyExistsFault;
(function (DBSnapshotAlreadyExistsFault) {
    DBSnapshotAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSnapshotAlreadyExistsFault = exports.DBSnapshotAlreadyExistsFault || (exports.DBSnapshotAlreadyExistsFault = {}));
var InvalidDBSnapshotStateFault;
(function (InvalidDBSnapshotStateFault) {
    InvalidDBSnapshotStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBSnapshotStateFault = exports.InvalidDBSnapshotStateFault || (exports.InvalidDBSnapshotStateFault = {}));
var CopyOptionGroupMessage;
(function (CopyOptionGroupMessage) {
    CopyOptionGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyOptionGroupMessage = exports.CopyOptionGroupMessage || (exports.CopyOptionGroupMessage = {}));
var DBSecurityGroupMembership;
(function (DBSecurityGroupMembership) {
    DBSecurityGroupMembership.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSecurityGroupMembership = exports.DBSecurityGroupMembership || (exports.DBSecurityGroupMembership = {}));
var OptionSetting;
(function (OptionSetting) {
    OptionSetting.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OptionSetting = exports.OptionSetting || (exports.OptionSetting = {}));
var VpcSecurityGroupMembership;
(function (VpcSecurityGroupMembership) {
    VpcSecurityGroupMembership.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VpcSecurityGroupMembership = exports.VpcSecurityGroupMembership || (exports.VpcSecurityGroupMembership = {}));
var Option;
(function (Option) {
    Option.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Option = exports.Option || (exports.Option = {}));
var OptionGroup;
(function (OptionGroup) {
    OptionGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OptionGroup = exports.OptionGroup || (exports.OptionGroup = {}));
var CopyOptionGroupResult;
(function (CopyOptionGroupResult) {
    CopyOptionGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CopyOptionGroupResult = exports.CopyOptionGroupResult || (exports.CopyOptionGroupResult = {}));
var OptionGroupAlreadyExistsFault;
(function (OptionGroupAlreadyExistsFault) {
    OptionGroupAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OptionGroupAlreadyExistsFault = exports.OptionGroupAlreadyExistsFault || (exports.OptionGroupAlreadyExistsFault = {}));
var OptionGroupNotFoundFault;
(function (OptionGroupNotFoundFault) {
    OptionGroupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OptionGroupNotFoundFault = exports.OptionGroupNotFoundFault || (exports.OptionGroupNotFoundFault = {}));
var OptionGroupQuotaExceededFault;
(function (OptionGroupQuotaExceededFault) {
    OptionGroupQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OptionGroupQuotaExceededFault = exports.OptionGroupQuotaExceededFault || (exports.OptionGroupQuotaExceededFault = {}));
var CreateCustomAvailabilityZoneMessage;
(function (CreateCustomAvailabilityZoneMessage) {
    CreateCustomAvailabilityZoneMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCustomAvailabilityZoneMessage = exports.CreateCustomAvailabilityZoneMessage || (exports.CreateCustomAvailabilityZoneMessage = {}));
var VpnDetails;
(function (VpnDetails) {
    VpnDetails.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.VpnPSK && { VpnPSK: smithy_client_1.SENSITIVE_STRING }),
    });
})(VpnDetails = exports.VpnDetails || (exports.VpnDetails = {}));
var CustomAvailabilityZone;
(function (CustomAvailabilityZone) {
    CustomAvailabilityZone.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.VpnDetails && { VpnDetails: VpnDetails.filterSensitiveLog(obj.VpnDetails) }),
    });
})(CustomAvailabilityZone = exports.CustomAvailabilityZone || (exports.CustomAvailabilityZone = {}));
var CreateCustomAvailabilityZoneResult;
(function (CreateCustomAvailabilityZoneResult) {
    CreateCustomAvailabilityZoneResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.CustomAvailabilityZone && {
            CustomAvailabilityZone: CustomAvailabilityZone.filterSensitiveLog(obj.CustomAvailabilityZone),
        }),
    });
})(CreateCustomAvailabilityZoneResult = exports.CreateCustomAvailabilityZoneResult || (exports.CreateCustomAvailabilityZoneResult = {}));
var CustomAvailabilityZoneAlreadyExistsFault;
(function (CustomAvailabilityZoneAlreadyExistsFault) {
    CustomAvailabilityZoneAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomAvailabilityZoneAlreadyExistsFault = exports.CustomAvailabilityZoneAlreadyExistsFault || (exports.CustomAvailabilityZoneAlreadyExistsFault = {}));
var CustomAvailabilityZoneQuotaExceededFault;
(function (CustomAvailabilityZoneQuotaExceededFault) {
    CustomAvailabilityZoneQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomAvailabilityZoneQuotaExceededFault = exports.CustomAvailabilityZoneQuotaExceededFault || (exports.CustomAvailabilityZoneQuotaExceededFault = {}));
var CreateCustomDBEngineVersionMessage;
(function (CreateCustomDBEngineVersionMessage) {
    CreateCustomDBEngineVersionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCustomDBEngineVersionMessage = exports.CreateCustomDBEngineVersionMessage || (exports.CreateCustomDBEngineVersionMessage = {}));
var CustomDBEngineVersionAlreadyExistsFault;
(function (CustomDBEngineVersionAlreadyExistsFault) {
    CustomDBEngineVersionAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomDBEngineVersionAlreadyExistsFault = exports.CustomDBEngineVersionAlreadyExistsFault || (exports.CustomDBEngineVersionAlreadyExistsFault = {}));
var CustomDBEngineVersionQuotaExceededFault;
(function (CustomDBEngineVersionQuotaExceededFault) {
    CustomDBEngineVersionQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomDBEngineVersionQuotaExceededFault = exports.CustomDBEngineVersionQuotaExceededFault || (exports.CustomDBEngineVersionQuotaExceededFault = {}));
var CharacterSet;
(function (CharacterSet) {
    CharacterSet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CharacterSet = exports.CharacterSet || (exports.CharacterSet = {}));
var Timezone;
(function (Timezone) {
    Timezone.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Timezone = exports.Timezone || (exports.Timezone = {}));
var UpgradeTarget;
(function (UpgradeTarget) {
    UpgradeTarget.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpgradeTarget = exports.UpgradeTarget || (exports.UpgradeTarget = {}));
var DBEngineVersion;
(function (DBEngineVersion) {
    DBEngineVersion.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBEngineVersion = exports.DBEngineVersion || (exports.DBEngineVersion = {}));
var ScalingConfiguration;
(function (ScalingConfiguration) {
    ScalingConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ScalingConfiguration = exports.ScalingConfiguration || (exports.ScalingConfiguration = {}));
var CreateDBClusterMessage;
(function (CreateDBClusterMessage) {
    CreateDBClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterMessage = exports.CreateDBClusterMessage || (exports.CreateDBClusterMessage = {}));
var DBClusterRole;
(function (DBClusterRole) {
    DBClusterRole.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterRole = exports.DBClusterRole || (exports.DBClusterRole = {}));
var DBClusterMember;
(function (DBClusterMember) {
    DBClusterMember.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterMember = exports.DBClusterMember || (exports.DBClusterMember = {}));
var DBClusterOptionGroupStatus;
(function (DBClusterOptionGroupStatus) {
    DBClusterOptionGroupStatus.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterOptionGroupStatus = exports.DBClusterOptionGroupStatus || (exports.DBClusterOptionGroupStatus = {}));
var DomainMembership;
(function (DomainMembership) {
    DomainMembership.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DomainMembership = exports.DomainMembership || (exports.DomainMembership = {}));
var WriteForwardingStatus;
(function (WriteForwardingStatus) {
    WriteForwardingStatus["DISABLED"] = "disabled";
    WriteForwardingStatus["DISABLING"] = "disabling";
    WriteForwardingStatus["ENABLED"] = "enabled";
    WriteForwardingStatus["ENABLING"] = "enabling";
    WriteForwardingStatus["UNKNOWN"] = "unknown";
})(WriteForwardingStatus = exports.WriteForwardingStatus || (exports.WriteForwardingStatus = {}));
var PendingCloudwatchLogsExports;
(function (PendingCloudwatchLogsExports) {
    PendingCloudwatchLogsExports.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PendingCloudwatchLogsExports = exports.PendingCloudwatchLogsExports || (exports.PendingCloudwatchLogsExports = {}));
var ClusterPendingModifiedValues;
(function (ClusterPendingModifiedValues) {
    ClusterPendingModifiedValues.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ClusterPendingModifiedValues = exports.ClusterPendingModifiedValues || (exports.ClusterPendingModifiedValues = {}));
var ScalingConfigurationInfo;
(function (ScalingConfigurationInfo) {
    ScalingConfigurationInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ScalingConfigurationInfo = exports.ScalingConfigurationInfo || (exports.ScalingConfigurationInfo = {}));
var DBCluster;
(function (DBCluster) {
    DBCluster.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBCluster = exports.DBCluster || (exports.DBCluster = {}));
var CreateDBClusterResult;
(function (CreateDBClusterResult) {
    CreateDBClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterResult = exports.CreateDBClusterResult || (exports.CreateDBClusterResult = {}));
var DBClusterAlreadyExistsFault;
(function (DBClusterAlreadyExistsFault) {
    DBClusterAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterAlreadyExistsFault = exports.DBClusterAlreadyExistsFault || (exports.DBClusterAlreadyExistsFault = {}));
var DBClusterParameterGroupNotFoundFault;
(function (DBClusterParameterGroupNotFoundFault) {
    DBClusterParameterGroupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterParameterGroupNotFoundFault = exports.DBClusterParameterGroupNotFoundFault || (exports.DBClusterParameterGroupNotFoundFault = {}));
var DBClusterQuotaExceededFault;
(function (DBClusterQuotaExceededFault) {
    DBClusterQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterQuotaExceededFault = exports.DBClusterQuotaExceededFault || (exports.DBClusterQuotaExceededFault = {}));
var DBSubnetGroupDoesNotCoverEnoughAZs;
(function (DBSubnetGroupDoesNotCoverEnoughAZs) {
    DBSubnetGroupDoesNotCoverEnoughAZs.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetGroupDoesNotCoverEnoughAZs = exports.DBSubnetGroupDoesNotCoverEnoughAZs || (exports.DBSubnetGroupDoesNotCoverEnoughAZs = {}));
var DBSubnetGroupNotFoundFault;
(function (DBSubnetGroupNotFoundFault) {
    DBSubnetGroupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetGroupNotFoundFault = exports.DBSubnetGroupNotFoundFault || (exports.DBSubnetGroupNotFoundFault = {}));
var DomainNotFoundFault;
(function (DomainNotFoundFault) {
    DomainNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DomainNotFoundFault = exports.DomainNotFoundFault || (exports.DomainNotFoundFault = {}));
var GlobalClusterNotFoundFault;
(function (GlobalClusterNotFoundFault) {
    GlobalClusterNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GlobalClusterNotFoundFault = exports.GlobalClusterNotFoundFault || (exports.GlobalClusterNotFoundFault = {}));
var InsufficientStorageClusterCapacityFault;
(function (InsufficientStorageClusterCapacityFault) {
    InsufficientStorageClusterCapacityFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InsufficientStorageClusterCapacityFault = exports.InsufficientStorageClusterCapacityFault || (exports.InsufficientStorageClusterCapacityFault = {}));
var InvalidDBSubnetGroupStateFault;
(function (InvalidDBSubnetGroupStateFault) {
    InvalidDBSubnetGroupStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBSubnetGroupStateFault = exports.InvalidDBSubnetGroupStateFault || (exports.InvalidDBSubnetGroupStateFault = {}));
var InvalidGlobalClusterStateFault;
(function (InvalidGlobalClusterStateFault) {
    InvalidGlobalClusterStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidGlobalClusterStateFault = exports.InvalidGlobalClusterStateFault || (exports.InvalidGlobalClusterStateFault = {}));
var InvalidSubnet;
(function (InvalidSubnet) {
    InvalidSubnet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidSubnet = exports.InvalidSubnet || (exports.InvalidSubnet = {}));
var InvalidVPCNetworkStateFault;
(function (InvalidVPCNetworkStateFault) {
    InvalidVPCNetworkStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidVPCNetworkStateFault = exports.InvalidVPCNetworkStateFault || (exports.InvalidVPCNetworkStateFault = {}));
var StorageQuotaExceededFault;
(function (StorageQuotaExceededFault) {
    StorageQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StorageQuotaExceededFault = exports.StorageQuotaExceededFault || (exports.StorageQuotaExceededFault = {}));
var CreateDBClusterEndpointMessage;
(function (CreateDBClusterEndpointMessage) {
    CreateDBClusterEndpointMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterEndpointMessage = exports.CreateDBClusterEndpointMessage || (exports.CreateDBClusterEndpointMessage = {}));
var DBClusterEndpoint;
(function (DBClusterEndpoint) {
    DBClusterEndpoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterEndpoint = exports.DBClusterEndpoint || (exports.DBClusterEndpoint = {}));
var DBClusterEndpointAlreadyExistsFault;
(function (DBClusterEndpointAlreadyExistsFault) {
    DBClusterEndpointAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterEndpointAlreadyExistsFault = exports.DBClusterEndpointAlreadyExistsFault || (exports.DBClusterEndpointAlreadyExistsFault = {}));
var DBClusterEndpointQuotaExceededFault;
(function (DBClusterEndpointQuotaExceededFault) {
    DBClusterEndpointQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterEndpointQuotaExceededFault = exports.DBClusterEndpointQuotaExceededFault || (exports.DBClusterEndpointQuotaExceededFault = {}));
var CreateDBClusterParameterGroupMessage;
(function (CreateDBClusterParameterGroupMessage) {
    CreateDBClusterParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterParameterGroupMessage = exports.CreateDBClusterParameterGroupMessage || (exports.CreateDBClusterParameterGroupMessage = {}));
var CreateDBClusterParameterGroupResult;
(function (CreateDBClusterParameterGroupResult) {
    CreateDBClusterParameterGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterParameterGroupResult = exports.CreateDBClusterParameterGroupResult || (exports.CreateDBClusterParameterGroupResult = {}));
var CreateDBClusterSnapshotMessage;
(function (CreateDBClusterSnapshotMessage) {
    CreateDBClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterSnapshotMessage = exports.CreateDBClusterSnapshotMessage || (exports.CreateDBClusterSnapshotMessage = {}));
var CreateDBClusterSnapshotResult;
(function (CreateDBClusterSnapshotResult) {
    CreateDBClusterSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBClusterSnapshotResult = exports.CreateDBClusterSnapshotResult || (exports.CreateDBClusterSnapshotResult = {}));
var AuthorizationNotFoundFault;
(function (AuthorizationNotFoundFault) {
    AuthorizationNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AuthorizationNotFoundFault = exports.AuthorizationNotFoundFault || (exports.AuthorizationNotFoundFault = {}));
var BackupPolicyNotFoundFault;
(function (BackupPolicyNotFoundFault) {
    BackupPolicyNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BackupPolicyNotFoundFault = exports.BackupPolicyNotFoundFault || (exports.BackupPolicyNotFoundFault = {}));
var CreateDBInstanceMessage;
(function (CreateDBInstanceMessage) {
    CreateDBInstanceMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBInstanceMessage = exports.CreateDBInstanceMessage || (exports.CreateDBInstanceMessage = {}));
var DBInstanceRole;
(function (DBInstanceRole) {
    DBInstanceRole.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceRole = exports.DBInstanceRole || (exports.DBInstanceRole = {}));
var AutomationMode;
(function (AutomationMode) {
    AutomationMode["ALL_PAUSED"] = "all-paused";
    AutomationMode["FULL"] = "full";
})(AutomationMode = exports.AutomationMode || (exports.AutomationMode = {}));
var DBInstanceAutomatedBackupsReplication;
(function (DBInstanceAutomatedBackupsReplication) {
    DBInstanceAutomatedBackupsReplication.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceAutomatedBackupsReplication = exports.DBInstanceAutomatedBackupsReplication || (exports.DBInstanceAutomatedBackupsReplication = {}));
var DBParameterGroupStatus;
(function (DBParameterGroupStatus) {
    DBParameterGroupStatus.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroupStatus = exports.DBParameterGroupStatus || (exports.DBParameterGroupStatus = {}));
var AvailabilityZone;
(function (AvailabilityZone) {
    AvailabilityZone.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AvailabilityZone = exports.AvailabilityZone || (exports.AvailabilityZone = {}));
var Outpost;
(function (Outpost) {
    Outpost.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Outpost = exports.Outpost || (exports.Outpost = {}));
var Subnet;
(function (Subnet) {
    Subnet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Subnet = exports.Subnet || (exports.Subnet = {}));
var DBSubnetGroup;
(function (DBSubnetGroup) {
    DBSubnetGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetGroup = exports.DBSubnetGroup || (exports.DBSubnetGroup = {}));
var Endpoint;
(function (Endpoint) {
    Endpoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Endpoint = exports.Endpoint || (exports.Endpoint = {}));
var OptionGroupMembership;
(function (OptionGroupMembership) {
    OptionGroupMembership.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OptionGroupMembership = exports.OptionGroupMembership || (exports.OptionGroupMembership = {}));
var PendingModifiedValues;
(function (PendingModifiedValues) {
    PendingModifiedValues.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PendingModifiedValues = exports.PendingModifiedValues || (exports.PendingModifiedValues = {}));
var ReplicaMode;
(function (ReplicaMode) {
    ReplicaMode["MOUNTED"] = "mounted";
    ReplicaMode["OPEN_READ_ONLY"] = "open-read-only";
})(ReplicaMode = exports.ReplicaMode || (exports.ReplicaMode = {}));
var DBInstanceStatusInfo;
(function (DBInstanceStatusInfo) {
    DBInstanceStatusInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceStatusInfo = exports.DBInstanceStatusInfo || (exports.DBInstanceStatusInfo = {}));
var DBInstance;
(function (DBInstance) {
    DBInstance.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstance = exports.DBInstance || (exports.DBInstance = {}));
var CreateDBInstanceResult;
(function (CreateDBInstanceResult) {
    CreateDBInstanceResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBInstanceResult = exports.CreateDBInstanceResult || (exports.CreateDBInstanceResult = {}));
var DBInstanceAlreadyExistsFault;
(function (DBInstanceAlreadyExistsFault) {
    DBInstanceAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceAlreadyExistsFault = exports.DBInstanceAlreadyExistsFault || (exports.DBInstanceAlreadyExistsFault = {}));
var InstanceQuotaExceededFault;
(function (InstanceQuotaExceededFault) {
    InstanceQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InstanceQuotaExceededFault = exports.InstanceQuotaExceededFault || (exports.InstanceQuotaExceededFault = {}));
var InsufficientDBInstanceCapacityFault;
(function (InsufficientDBInstanceCapacityFault) {
    InsufficientDBInstanceCapacityFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InsufficientDBInstanceCapacityFault = exports.InsufficientDBInstanceCapacityFault || (exports.InsufficientDBInstanceCapacityFault = {}));
var ProvisionedIopsNotAvailableInAZFault;
(function (ProvisionedIopsNotAvailableInAZFault) {
    ProvisionedIopsNotAvailableInAZFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ProvisionedIopsNotAvailableInAZFault = exports.ProvisionedIopsNotAvailableInAZFault || (exports.ProvisionedIopsNotAvailableInAZFault = {}));
var StorageTypeNotSupportedFault;
(function (StorageTypeNotSupportedFault) {
    StorageTypeNotSupportedFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StorageTypeNotSupportedFault = exports.StorageTypeNotSupportedFault || (exports.StorageTypeNotSupportedFault = {}));
var CreateDBInstanceReadReplicaMessage;
(function (CreateDBInstanceReadReplicaMessage) {
    CreateDBInstanceReadReplicaMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBInstanceReadReplicaMessage = exports.CreateDBInstanceReadReplicaMessage || (exports.CreateDBInstanceReadReplicaMessage = {}));
var CreateDBInstanceReadReplicaResult;
(function (CreateDBInstanceReadReplicaResult) {
    CreateDBInstanceReadReplicaResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBInstanceReadReplicaResult = exports.CreateDBInstanceReadReplicaResult || (exports.CreateDBInstanceReadReplicaResult = {}));
var DBSubnetGroupNotAllowedFault;
(function (DBSubnetGroupNotAllowedFault) {
    DBSubnetGroupNotAllowedFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetGroupNotAllowedFault = exports.DBSubnetGroupNotAllowedFault || (exports.DBSubnetGroupNotAllowedFault = {}));
var InvalidDBSubnetGroupFault;
(function (InvalidDBSubnetGroupFault) {
    InvalidDBSubnetGroupFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBSubnetGroupFault = exports.InvalidDBSubnetGroupFault || (exports.InvalidDBSubnetGroupFault = {}));
var CreateDBParameterGroupMessage;
(function (CreateDBParameterGroupMessage) {
    CreateDBParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBParameterGroupMessage = exports.CreateDBParameterGroupMessage || (exports.CreateDBParameterGroupMessage = {}));
var CreateDBParameterGroupResult;
(function (CreateDBParameterGroupResult) {
    CreateDBParameterGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBParameterGroupResult = exports.CreateDBParameterGroupResult || (exports.CreateDBParameterGroupResult = {}));
var AuthScheme;
(function (AuthScheme) {
    AuthScheme["SECRETS"] = "SECRETS";
})(AuthScheme = exports.AuthScheme || (exports.AuthScheme = {}));
var IAMAuthMode;
(function (IAMAuthMode) {
    IAMAuthMode["DISABLED"] = "DISABLED";
    IAMAuthMode["REQUIRED"] = "REQUIRED";
})(IAMAuthMode = exports.IAMAuthMode || (exports.IAMAuthMode = {}));
var UserAuthConfig;
(function (UserAuthConfig) {
    UserAuthConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UserAuthConfig = exports.UserAuthConfig || (exports.UserAuthConfig = {}));
var EngineFamily;
(function (EngineFamily) {
    EngineFamily["MYSQL"] = "MYSQL";
    EngineFamily["POSTGRESQL"] = "POSTGRESQL";
})(EngineFamily = exports.EngineFamily || (exports.EngineFamily = {}));
var CreateDBProxyRequest;
(function (CreateDBProxyRequest) {
    CreateDBProxyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBProxyRequest = exports.CreateDBProxyRequest || (exports.CreateDBProxyRequest = {}));
var UserAuthConfigInfo;
(function (UserAuthConfigInfo) {
    UserAuthConfigInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UserAuthConfigInfo = exports.UserAuthConfigInfo || (exports.UserAuthConfigInfo = {}));
var DBProxyStatus;
(function (DBProxyStatus) {
    DBProxyStatus["AVAILABLE"] = "available";
    DBProxyStatus["CREATING"] = "creating";
    DBProxyStatus["DELETING"] = "deleting";
    DBProxyStatus["INCOMPATIBLE_NETWORK"] = "incompatible-network";
    DBProxyStatus["INSUFFICIENT_RESOURCE_LIMITS"] = "insufficient-resource-limits";
    DBProxyStatus["MODIFYING"] = "modifying";
    DBProxyStatus["REACTIVATING"] = "reactivating";
    DBProxyStatus["SUSPENDED"] = "suspended";
    DBProxyStatus["SUSPENDING"] = "suspending";
})(DBProxyStatus = exports.DBProxyStatus || (exports.DBProxyStatus = {}));
var DBProxy;
(function (DBProxy) {
    DBProxy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxy = exports.DBProxy || (exports.DBProxy = {}));
var CreateDBProxyResponse;
(function (CreateDBProxyResponse) {
    CreateDBProxyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBProxyResponse = exports.CreateDBProxyResponse || (exports.CreateDBProxyResponse = {}));
var DBProxyAlreadyExistsFault;
(function (DBProxyAlreadyExistsFault) {
    DBProxyAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyAlreadyExistsFault = exports.DBProxyAlreadyExistsFault || (exports.DBProxyAlreadyExistsFault = {}));
var DBProxyQuotaExceededFault;
(function (DBProxyQuotaExceededFault) {
    DBProxyQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyQuotaExceededFault = exports.DBProxyQuotaExceededFault || (exports.DBProxyQuotaExceededFault = {}));
var DBProxyEndpointTargetRole;
(function (DBProxyEndpointTargetRole) {
    DBProxyEndpointTargetRole["READ_ONLY"] = "READ_ONLY";
    DBProxyEndpointTargetRole["READ_WRITE"] = "READ_WRITE";
})(DBProxyEndpointTargetRole = exports.DBProxyEndpointTargetRole || (exports.DBProxyEndpointTargetRole = {}));
var CreateDBProxyEndpointRequest;
(function (CreateDBProxyEndpointRequest) {
    CreateDBProxyEndpointRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBProxyEndpointRequest = exports.CreateDBProxyEndpointRequest || (exports.CreateDBProxyEndpointRequest = {}));
var DBProxyEndpointStatus;
(function (DBProxyEndpointStatus) {
    DBProxyEndpointStatus["AVAILABLE"] = "available";
    DBProxyEndpointStatus["CREATING"] = "creating";
    DBProxyEndpointStatus["DELETING"] = "deleting";
    DBProxyEndpointStatus["INCOMPATIBLE_NETWORK"] = "incompatible-network";
    DBProxyEndpointStatus["INSUFFICIENT_RESOURCE_LIMITS"] = "insufficient-resource-limits";
    DBProxyEndpointStatus["MODIFYING"] = "modifying";
})(DBProxyEndpointStatus = exports.DBProxyEndpointStatus || (exports.DBProxyEndpointStatus = {}));
var DBProxyEndpoint;
(function (DBProxyEndpoint) {
    DBProxyEndpoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyEndpoint = exports.DBProxyEndpoint || (exports.DBProxyEndpoint = {}));
var CreateDBProxyEndpointResponse;
(function (CreateDBProxyEndpointResponse) {
    CreateDBProxyEndpointResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBProxyEndpointResponse = exports.CreateDBProxyEndpointResponse || (exports.CreateDBProxyEndpointResponse = {}));
var DBProxyEndpointAlreadyExistsFault;
(function (DBProxyEndpointAlreadyExistsFault) {
    DBProxyEndpointAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyEndpointAlreadyExistsFault = exports.DBProxyEndpointAlreadyExistsFault || (exports.DBProxyEndpointAlreadyExistsFault = {}));
var DBProxyEndpointQuotaExceededFault;
(function (DBProxyEndpointQuotaExceededFault) {
    DBProxyEndpointQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyEndpointQuotaExceededFault = exports.DBProxyEndpointQuotaExceededFault || (exports.DBProxyEndpointQuotaExceededFault = {}));
var InvalidDBProxyStateFault;
(function (InvalidDBProxyStateFault) {
    InvalidDBProxyStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBProxyStateFault = exports.InvalidDBProxyStateFault || (exports.InvalidDBProxyStateFault = {}));
var CreateDBSecurityGroupMessage;
(function (CreateDBSecurityGroupMessage) {
    CreateDBSecurityGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBSecurityGroupMessage = exports.CreateDBSecurityGroupMessage || (exports.CreateDBSecurityGroupMessage = {}));
var CreateDBSecurityGroupResult;
(function (CreateDBSecurityGroupResult) {
    CreateDBSecurityGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBSecurityGroupResult = exports.CreateDBSecurityGroupResult || (exports.CreateDBSecurityGroupResult = {}));
var DBSecurityGroupAlreadyExistsFault;
(function (DBSecurityGroupAlreadyExistsFault) {
    DBSecurityGroupAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSecurityGroupAlreadyExistsFault = exports.DBSecurityGroupAlreadyExistsFault || (exports.DBSecurityGroupAlreadyExistsFault = {}));
var DBSecurityGroupNotSupportedFault;
(function (DBSecurityGroupNotSupportedFault) {
    DBSecurityGroupNotSupportedFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSecurityGroupNotSupportedFault = exports.DBSecurityGroupNotSupportedFault || (exports.DBSecurityGroupNotSupportedFault = {}));
var DBSecurityGroupQuotaExceededFault;
(function (DBSecurityGroupQuotaExceededFault) {
    DBSecurityGroupQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSecurityGroupQuotaExceededFault = exports.DBSecurityGroupQuotaExceededFault || (exports.DBSecurityGroupQuotaExceededFault = {}));
var CreateDBSnapshotMessage;
(function (CreateDBSnapshotMessage) {
    CreateDBSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBSnapshotMessage = exports.CreateDBSnapshotMessage || (exports.CreateDBSnapshotMessage = {}));
var CreateDBSnapshotResult;
(function (CreateDBSnapshotResult) {
    CreateDBSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBSnapshotResult = exports.CreateDBSnapshotResult || (exports.CreateDBSnapshotResult = {}));
var CreateDBSubnetGroupMessage;
(function (CreateDBSubnetGroupMessage) {
    CreateDBSubnetGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBSubnetGroupMessage = exports.CreateDBSubnetGroupMessage || (exports.CreateDBSubnetGroupMessage = {}));
var CreateDBSubnetGroupResult;
(function (CreateDBSubnetGroupResult) {
    CreateDBSubnetGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateDBSubnetGroupResult = exports.CreateDBSubnetGroupResult || (exports.CreateDBSubnetGroupResult = {}));
var DBSubnetGroupAlreadyExistsFault;
(function (DBSubnetGroupAlreadyExistsFault) {
    DBSubnetGroupAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetGroupAlreadyExistsFault = exports.DBSubnetGroupAlreadyExistsFault || (exports.DBSubnetGroupAlreadyExistsFault = {}));
var DBSubnetGroupQuotaExceededFault;
(function (DBSubnetGroupQuotaExceededFault) {
    DBSubnetGroupQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetGroupQuotaExceededFault = exports.DBSubnetGroupQuotaExceededFault || (exports.DBSubnetGroupQuotaExceededFault = {}));
var DBSubnetQuotaExceededFault;
(function (DBSubnetQuotaExceededFault) {
    DBSubnetQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBSubnetQuotaExceededFault = exports.DBSubnetQuotaExceededFault || (exports.DBSubnetQuotaExceededFault = {}));
var CreateEventSubscriptionMessage;
(function (CreateEventSubscriptionMessage) {
    CreateEventSubscriptionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateEventSubscriptionMessage = exports.CreateEventSubscriptionMessage || (exports.CreateEventSubscriptionMessage = {}));
var CreateEventSubscriptionResult;
(function (CreateEventSubscriptionResult) {
    CreateEventSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateEventSubscriptionResult = exports.CreateEventSubscriptionResult || (exports.CreateEventSubscriptionResult = {}));
var EventSubscriptionQuotaExceededFault;
(function (EventSubscriptionQuotaExceededFault) {
    EventSubscriptionQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EventSubscriptionQuotaExceededFault = exports.EventSubscriptionQuotaExceededFault || (exports.EventSubscriptionQuotaExceededFault = {}));
var SNSInvalidTopicFault;
(function (SNSInvalidTopicFault) {
    SNSInvalidTopicFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SNSInvalidTopicFault = exports.SNSInvalidTopicFault || (exports.SNSInvalidTopicFault = {}));
var SNSNoAuthorizationFault;
(function (SNSNoAuthorizationFault) {
    SNSNoAuthorizationFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SNSNoAuthorizationFault = exports.SNSNoAuthorizationFault || (exports.SNSNoAuthorizationFault = {}));
var SNSTopicArnNotFoundFault;
(function (SNSTopicArnNotFoundFault) {
    SNSTopicArnNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SNSTopicArnNotFoundFault = exports.SNSTopicArnNotFoundFault || (exports.SNSTopicArnNotFoundFault = {}));
var SubscriptionAlreadyExistFault;
(function (SubscriptionAlreadyExistFault) {
    SubscriptionAlreadyExistFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SubscriptionAlreadyExistFault = exports.SubscriptionAlreadyExistFault || (exports.SubscriptionAlreadyExistFault = {}));
var SubscriptionCategoryNotFoundFault;
(function (SubscriptionCategoryNotFoundFault) {
    SubscriptionCategoryNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SubscriptionCategoryNotFoundFault = exports.SubscriptionCategoryNotFoundFault || (exports.SubscriptionCategoryNotFoundFault = {}));
var CreateGlobalClusterMessage;
(function (CreateGlobalClusterMessage) {
    CreateGlobalClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateGlobalClusterMessage = exports.CreateGlobalClusterMessage || (exports.CreateGlobalClusterMessage = {}));
var FailoverStatus;
(function (FailoverStatus) {
    FailoverStatus["CANCELLING"] = "cancelling";
    FailoverStatus["FAILING_OVER"] = "failing-over";
    FailoverStatus["PENDING"] = "pending";
})(FailoverStatus = exports.FailoverStatus || (exports.FailoverStatus = {}));
var FailoverState;
(function (FailoverState) {
    FailoverState.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FailoverState = exports.FailoverState || (exports.FailoverState = {}));
var GlobalClusterMember;
(function (GlobalClusterMember) {
    GlobalClusterMember.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GlobalClusterMember = exports.GlobalClusterMember || (exports.GlobalClusterMember = {}));
var GlobalCluster;
(function (GlobalCluster) {
    GlobalCluster.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GlobalCluster = exports.GlobalCluster || (exports.GlobalCluster = {}));
var CreateGlobalClusterResult;
(function (CreateGlobalClusterResult) {
    CreateGlobalClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateGlobalClusterResult = exports.CreateGlobalClusterResult || (exports.CreateGlobalClusterResult = {}));
var GlobalClusterAlreadyExistsFault;
(function (GlobalClusterAlreadyExistsFault) {
    GlobalClusterAlreadyExistsFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GlobalClusterAlreadyExistsFault = exports.GlobalClusterAlreadyExistsFault || (exports.GlobalClusterAlreadyExistsFault = {}));
var GlobalClusterQuotaExceededFault;
(function (GlobalClusterQuotaExceededFault) {
    GlobalClusterQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GlobalClusterQuotaExceededFault = exports.GlobalClusterQuotaExceededFault || (exports.GlobalClusterQuotaExceededFault = {}));
var CreateOptionGroupMessage;
(function (CreateOptionGroupMessage) {
    CreateOptionGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateOptionGroupMessage = exports.CreateOptionGroupMessage || (exports.CreateOptionGroupMessage = {}));
var CreateOptionGroupResult;
(function (CreateOptionGroupResult) {
    CreateOptionGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateOptionGroupResult = exports.CreateOptionGroupResult || (exports.CreateOptionGroupResult = {}));
var DeleteCustomAvailabilityZoneMessage;
(function (DeleteCustomAvailabilityZoneMessage) {
    DeleteCustomAvailabilityZoneMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCustomAvailabilityZoneMessage = exports.DeleteCustomAvailabilityZoneMessage || (exports.DeleteCustomAvailabilityZoneMessage = {}));
var DeleteCustomAvailabilityZoneResult;
(function (DeleteCustomAvailabilityZoneResult) {
    DeleteCustomAvailabilityZoneResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.CustomAvailabilityZone && {
            CustomAvailabilityZone: CustomAvailabilityZone.filterSensitiveLog(obj.CustomAvailabilityZone),
        }),
    });
})(DeleteCustomAvailabilityZoneResult = exports.DeleteCustomAvailabilityZoneResult || (exports.DeleteCustomAvailabilityZoneResult = {}));
var CustomDBEngineVersionNotFoundFault;
(function (CustomDBEngineVersionNotFoundFault) {
    CustomDBEngineVersionNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomDBEngineVersionNotFoundFault = exports.CustomDBEngineVersionNotFoundFault || (exports.CustomDBEngineVersionNotFoundFault = {}));
var DeleteCustomDBEngineVersionMessage;
(function (DeleteCustomDBEngineVersionMessage) {
    DeleteCustomDBEngineVersionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCustomDBEngineVersionMessage = exports.DeleteCustomDBEngineVersionMessage || (exports.DeleteCustomDBEngineVersionMessage = {}));
var InvalidCustomDBEngineVersionStateFault;
(function (InvalidCustomDBEngineVersionStateFault) {
    InvalidCustomDBEngineVersionStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidCustomDBEngineVersionStateFault = exports.InvalidCustomDBEngineVersionStateFault || (exports.InvalidCustomDBEngineVersionStateFault = {}));
var DeleteDBClusterMessage;
(function (DeleteDBClusterMessage) {
    DeleteDBClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBClusterMessage = exports.DeleteDBClusterMessage || (exports.DeleteDBClusterMessage = {}));
var DeleteDBClusterResult;
(function (DeleteDBClusterResult) {
    DeleteDBClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBClusterResult = exports.DeleteDBClusterResult || (exports.DeleteDBClusterResult = {}));
var DBClusterEndpointNotFoundFault;
(function (DBClusterEndpointNotFoundFault) {
    DBClusterEndpointNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterEndpointNotFoundFault = exports.DBClusterEndpointNotFoundFault || (exports.DBClusterEndpointNotFoundFault = {}));
var DeleteDBClusterEndpointMessage;
(function (DeleteDBClusterEndpointMessage) {
    DeleteDBClusterEndpointMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBClusterEndpointMessage = exports.DeleteDBClusterEndpointMessage || (exports.DeleteDBClusterEndpointMessage = {}));
var InvalidDBClusterEndpointStateFault;
(function (InvalidDBClusterEndpointStateFault) {
    InvalidDBClusterEndpointStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBClusterEndpointStateFault = exports.InvalidDBClusterEndpointStateFault || (exports.InvalidDBClusterEndpointStateFault = {}));
var DeleteDBClusterParameterGroupMessage;
(function (DeleteDBClusterParameterGroupMessage) {
    DeleteDBClusterParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBClusterParameterGroupMessage = exports.DeleteDBClusterParameterGroupMessage || (exports.DeleteDBClusterParameterGroupMessage = {}));
var InvalidDBParameterGroupStateFault;
(function (InvalidDBParameterGroupStateFault) {
    InvalidDBParameterGroupStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBParameterGroupStateFault = exports.InvalidDBParameterGroupStateFault || (exports.InvalidDBParameterGroupStateFault = {}));
var DeleteDBClusterSnapshotMessage;
(function (DeleteDBClusterSnapshotMessage) {
    DeleteDBClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBClusterSnapshotMessage = exports.DeleteDBClusterSnapshotMessage || (exports.DeleteDBClusterSnapshotMessage = {}));
var DeleteDBClusterSnapshotResult;
(function (DeleteDBClusterSnapshotResult) {
    DeleteDBClusterSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBClusterSnapshotResult = exports.DeleteDBClusterSnapshotResult || (exports.DeleteDBClusterSnapshotResult = {}));
var DBInstanceAutomatedBackupQuotaExceededFault;
(function (DBInstanceAutomatedBackupQuotaExceededFault) {
    DBInstanceAutomatedBackupQuotaExceededFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceAutomatedBackupQuotaExceededFault = exports.DBInstanceAutomatedBackupQuotaExceededFault || (exports.DBInstanceAutomatedBackupQuotaExceededFault = {}));
var DeleteDBInstanceMessage;
(function (DeleteDBInstanceMessage) {
    DeleteDBInstanceMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBInstanceMessage = exports.DeleteDBInstanceMessage || (exports.DeleteDBInstanceMessage = {}));
var DeleteDBInstanceResult;
(function (DeleteDBInstanceResult) {
    DeleteDBInstanceResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBInstanceResult = exports.DeleteDBInstanceResult || (exports.DeleteDBInstanceResult = {}));
var DBInstanceAutomatedBackupNotFoundFault;
(function (DBInstanceAutomatedBackupNotFoundFault) {
    DBInstanceAutomatedBackupNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceAutomatedBackupNotFoundFault = exports.DBInstanceAutomatedBackupNotFoundFault || (exports.DBInstanceAutomatedBackupNotFoundFault = {}));
var DeleteDBInstanceAutomatedBackupMessage;
(function (DeleteDBInstanceAutomatedBackupMessage) {
    DeleteDBInstanceAutomatedBackupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBInstanceAutomatedBackupMessage = exports.DeleteDBInstanceAutomatedBackupMessage || (exports.DeleteDBInstanceAutomatedBackupMessage = {}));
var RestoreWindow;
(function (RestoreWindow) {
    RestoreWindow.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreWindow = exports.RestoreWindow || (exports.RestoreWindow = {}));
var DBInstanceAutomatedBackup;
(function (DBInstanceAutomatedBackup) {
    DBInstanceAutomatedBackup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceAutomatedBackup = exports.DBInstanceAutomatedBackup || (exports.DBInstanceAutomatedBackup = {}));
var DeleteDBInstanceAutomatedBackupResult;
(function (DeleteDBInstanceAutomatedBackupResult) {
    DeleteDBInstanceAutomatedBackupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBInstanceAutomatedBackupResult = exports.DeleteDBInstanceAutomatedBackupResult || (exports.DeleteDBInstanceAutomatedBackupResult = {}));
var InvalidDBInstanceAutomatedBackupStateFault;
(function (InvalidDBInstanceAutomatedBackupStateFault) {
    InvalidDBInstanceAutomatedBackupStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBInstanceAutomatedBackupStateFault = exports.InvalidDBInstanceAutomatedBackupStateFault || (exports.InvalidDBInstanceAutomatedBackupStateFault = {}));
var DeleteDBParameterGroupMessage;
(function (DeleteDBParameterGroupMessage) {
    DeleteDBParameterGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBParameterGroupMessage = exports.DeleteDBParameterGroupMessage || (exports.DeleteDBParameterGroupMessage = {}));
var DeleteDBProxyRequest;
(function (DeleteDBProxyRequest) {
    DeleteDBProxyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBProxyRequest = exports.DeleteDBProxyRequest || (exports.DeleteDBProxyRequest = {}));
var DeleteDBProxyResponse;
(function (DeleteDBProxyResponse) {
    DeleteDBProxyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBProxyResponse = exports.DeleteDBProxyResponse || (exports.DeleteDBProxyResponse = {}));
var DBProxyEndpointNotFoundFault;
(function (DBProxyEndpointNotFoundFault) {
    DBProxyEndpointNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyEndpointNotFoundFault = exports.DBProxyEndpointNotFoundFault || (exports.DBProxyEndpointNotFoundFault = {}));
var DeleteDBProxyEndpointRequest;
(function (DeleteDBProxyEndpointRequest) {
    DeleteDBProxyEndpointRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBProxyEndpointRequest = exports.DeleteDBProxyEndpointRequest || (exports.DeleteDBProxyEndpointRequest = {}));
var DeleteDBProxyEndpointResponse;
(function (DeleteDBProxyEndpointResponse) {
    DeleteDBProxyEndpointResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBProxyEndpointResponse = exports.DeleteDBProxyEndpointResponse || (exports.DeleteDBProxyEndpointResponse = {}));
var InvalidDBProxyEndpointStateFault;
(function (InvalidDBProxyEndpointStateFault) {
    InvalidDBProxyEndpointStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBProxyEndpointStateFault = exports.InvalidDBProxyEndpointStateFault || (exports.InvalidDBProxyEndpointStateFault = {}));
var DeleteDBSecurityGroupMessage;
(function (DeleteDBSecurityGroupMessage) {
    DeleteDBSecurityGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBSecurityGroupMessage = exports.DeleteDBSecurityGroupMessage || (exports.DeleteDBSecurityGroupMessage = {}));
var DeleteDBSnapshotMessage;
(function (DeleteDBSnapshotMessage) {
    DeleteDBSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBSnapshotMessage = exports.DeleteDBSnapshotMessage || (exports.DeleteDBSnapshotMessage = {}));
var DeleteDBSnapshotResult;
(function (DeleteDBSnapshotResult) {
    DeleteDBSnapshotResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBSnapshotResult = exports.DeleteDBSnapshotResult || (exports.DeleteDBSnapshotResult = {}));
var DeleteDBSubnetGroupMessage;
(function (DeleteDBSubnetGroupMessage) {
    DeleteDBSubnetGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDBSubnetGroupMessage = exports.DeleteDBSubnetGroupMessage || (exports.DeleteDBSubnetGroupMessage = {}));
var InvalidDBSubnetStateFault;
(function (InvalidDBSubnetStateFault) {
    InvalidDBSubnetStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidDBSubnetStateFault = exports.InvalidDBSubnetStateFault || (exports.InvalidDBSubnetStateFault = {}));
var DeleteEventSubscriptionMessage;
(function (DeleteEventSubscriptionMessage) {
    DeleteEventSubscriptionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteEventSubscriptionMessage = exports.DeleteEventSubscriptionMessage || (exports.DeleteEventSubscriptionMessage = {}));
var DeleteEventSubscriptionResult;
(function (DeleteEventSubscriptionResult) {
    DeleteEventSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteEventSubscriptionResult = exports.DeleteEventSubscriptionResult || (exports.DeleteEventSubscriptionResult = {}));
var InvalidEventSubscriptionStateFault;
(function (InvalidEventSubscriptionStateFault) {
    InvalidEventSubscriptionStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidEventSubscriptionStateFault = exports.InvalidEventSubscriptionStateFault || (exports.InvalidEventSubscriptionStateFault = {}));
var DeleteGlobalClusterMessage;
(function (DeleteGlobalClusterMessage) {
    DeleteGlobalClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteGlobalClusterMessage = exports.DeleteGlobalClusterMessage || (exports.DeleteGlobalClusterMessage = {}));
var DeleteGlobalClusterResult;
(function (DeleteGlobalClusterResult) {
    DeleteGlobalClusterResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteGlobalClusterResult = exports.DeleteGlobalClusterResult || (exports.DeleteGlobalClusterResult = {}));
var DeleteInstallationMediaMessage;
(function (DeleteInstallationMediaMessage) {
    DeleteInstallationMediaMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteInstallationMediaMessage = exports.DeleteInstallationMediaMessage || (exports.DeleteInstallationMediaMessage = {}));
var InstallationMediaFailureCause;
(function (InstallationMediaFailureCause) {
    InstallationMediaFailureCause.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InstallationMediaFailureCause = exports.InstallationMediaFailureCause || (exports.InstallationMediaFailureCause = {}));
var InstallationMedia;
(function (InstallationMedia) {
    InstallationMedia.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InstallationMedia = exports.InstallationMedia || (exports.InstallationMedia = {}));
var InstallationMediaNotFoundFault;
(function (InstallationMediaNotFoundFault) {
    InstallationMediaNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InstallationMediaNotFoundFault = exports.InstallationMediaNotFoundFault || (exports.InstallationMediaNotFoundFault = {}));
var DeleteOptionGroupMessage;
(function (DeleteOptionGroupMessage) {
    DeleteOptionGroupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteOptionGroupMessage = exports.DeleteOptionGroupMessage || (exports.DeleteOptionGroupMessage = {}));
var InvalidOptionGroupStateFault;
(function (InvalidOptionGroupStateFault) {
    InvalidOptionGroupStateFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidOptionGroupStateFault = exports.InvalidOptionGroupStateFault || (exports.InvalidOptionGroupStateFault = {}));
var DBProxyTargetNotFoundFault;
(function (DBProxyTargetNotFoundFault) {
    DBProxyTargetNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyTargetNotFoundFault = exports.DBProxyTargetNotFoundFault || (exports.DBProxyTargetNotFoundFault = {}));
var DeregisterDBProxyTargetsRequest;
(function (DeregisterDBProxyTargetsRequest) {
    DeregisterDBProxyTargetsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeregisterDBProxyTargetsRequest = exports.DeregisterDBProxyTargetsRequest || (exports.DeregisterDBProxyTargetsRequest = {}));
var DeregisterDBProxyTargetsResponse;
(function (DeregisterDBProxyTargetsResponse) {
    DeregisterDBProxyTargetsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeregisterDBProxyTargetsResponse = exports.DeregisterDBProxyTargetsResponse || (exports.DeregisterDBProxyTargetsResponse = {}));
var DescribeAccountAttributesMessage;
(function (DescribeAccountAttributesMessage) {
    DescribeAccountAttributesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAccountAttributesMessage = exports.DescribeAccountAttributesMessage || (exports.DescribeAccountAttributesMessage = {}));
var Certificate;
(function (Certificate) {
    Certificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Certificate = exports.Certificate || (exports.Certificate = {}));
var CertificateMessage;
(function (CertificateMessage) {
    CertificateMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CertificateMessage = exports.CertificateMessage || (exports.CertificateMessage = {}));
var CertificateNotFoundFault;
(function (CertificateNotFoundFault) {
    CertificateNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CertificateNotFoundFault = exports.CertificateNotFoundFault || (exports.CertificateNotFoundFault = {}));
var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Filter = exports.Filter || (exports.Filter = {}));
var DescribeCertificatesMessage;
(function (DescribeCertificatesMessage) {
    DescribeCertificatesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCertificatesMessage = exports.DescribeCertificatesMessage || (exports.DescribeCertificatesMessage = {}));
var CustomAvailabilityZoneMessage;
(function (CustomAvailabilityZoneMessage) {
    CustomAvailabilityZoneMessage.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.CustomAvailabilityZones && {
            CustomAvailabilityZones: obj.CustomAvailabilityZones.map((item) => CustomAvailabilityZone.filterSensitiveLog(item)),
        }),
    });
})(CustomAvailabilityZoneMessage = exports.CustomAvailabilityZoneMessage || (exports.CustomAvailabilityZoneMessage = {}));
var DescribeCustomAvailabilityZonesMessage;
(function (DescribeCustomAvailabilityZonesMessage) {
    DescribeCustomAvailabilityZonesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCustomAvailabilityZonesMessage = exports.DescribeCustomAvailabilityZonesMessage || (exports.DescribeCustomAvailabilityZonesMessage = {}));
var DBClusterBacktrackMessage;
(function (DBClusterBacktrackMessage) {
    DBClusterBacktrackMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterBacktrackMessage = exports.DBClusterBacktrackMessage || (exports.DBClusterBacktrackMessage = {}));
var DBClusterBacktrackNotFoundFault;
(function (DBClusterBacktrackNotFoundFault) {
    DBClusterBacktrackNotFoundFault.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterBacktrackNotFoundFault = exports.DBClusterBacktrackNotFoundFault || (exports.DBClusterBacktrackNotFoundFault = {}));
var DescribeDBClusterBacktracksMessage;
(function (DescribeDBClusterBacktracksMessage) {
    DescribeDBClusterBacktracksMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterBacktracksMessage = exports.DescribeDBClusterBacktracksMessage || (exports.DescribeDBClusterBacktracksMessage = {}));
var DBClusterEndpointMessage;
(function (DBClusterEndpointMessage) {
    DBClusterEndpointMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterEndpointMessage = exports.DBClusterEndpointMessage || (exports.DBClusterEndpointMessage = {}));
var DescribeDBClusterEndpointsMessage;
(function (DescribeDBClusterEndpointsMessage) {
    DescribeDBClusterEndpointsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterEndpointsMessage = exports.DescribeDBClusterEndpointsMessage || (exports.DescribeDBClusterEndpointsMessage = {}));
var DBClusterParameterGroupsMessage;
(function (DBClusterParameterGroupsMessage) {
    DBClusterParameterGroupsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterParameterGroupsMessage = exports.DBClusterParameterGroupsMessage || (exports.DBClusterParameterGroupsMessage = {}));
var DescribeDBClusterParameterGroupsMessage;
(function (DescribeDBClusterParameterGroupsMessage) {
    DescribeDBClusterParameterGroupsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterParameterGroupsMessage = exports.DescribeDBClusterParameterGroupsMessage || (exports.DescribeDBClusterParameterGroupsMessage = {}));
var Parameter;
(function (Parameter) {
    Parameter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Parameter = exports.Parameter || (exports.Parameter = {}));
var DBClusterParameterGroupDetails;
(function (DBClusterParameterGroupDetails) {
    DBClusterParameterGroupDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterParameterGroupDetails = exports.DBClusterParameterGroupDetails || (exports.DBClusterParameterGroupDetails = {}));
var DescribeDBClusterParametersMessage;
(function (DescribeDBClusterParametersMessage) {
    DescribeDBClusterParametersMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterParametersMessage = exports.DescribeDBClusterParametersMessage || (exports.DescribeDBClusterParametersMessage = {}));
var DBClusterMessage;
(function (DBClusterMessage) {
    DBClusterMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterMessage = exports.DBClusterMessage || (exports.DBClusterMessage = {}));
var DescribeDBClustersMessage;
(function (DescribeDBClustersMessage) {
    DescribeDBClustersMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClustersMessage = exports.DescribeDBClustersMessage || (exports.DescribeDBClustersMessage = {}));
var DescribeDBClusterSnapshotAttributesMessage;
(function (DescribeDBClusterSnapshotAttributesMessage) {
    DescribeDBClusterSnapshotAttributesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterSnapshotAttributesMessage = exports.DescribeDBClusterSnapshotAttributesMessage || (exports.DescribeDBClusterSnapshotAttributesMessage = {}));
var DBClusterSnapshotAttribute;
(function (DBClusterSnapshotAttribute) {
    DBClusterSnapshotAttribute.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterSnapshotAttribute = exports.DBClusterSnapshotAttribute || (exports.DBClusterSnapshotAttribute = {}));
var DBClusterSnapshotAttributesResult;
(function (DBClusterSnapshotAttributesResult) {
    DBClusterSnapshotAttributesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterSnapshotAttributesResult = exports.DBClusterSnapshotAttributesResult || (exports.DBClusterSnapshotAttributesResult = {}));
var DescribeDBClusterSnapshotAttributesResult;
(function (DescribeDBClusterSnapshotAttributesResult) {
    DescribeDBClusterSnapshotAttributesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterSnapshotAttributesResult = exports.DescribeDBClusterSnapshotAttributesResult || (exports.DescribeDBClusterSnapshotAttributesResult = {}));
var DBClusterSnapshotMessage;
(function (DBClusterSnapshotMessage) {
    DBClusterSnapshotMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBClusterSnapshotMessage = exports.DBClusterSnapshotMessage || (exports.DBClusterSnapshotMessage = {}));
var DescribeDBClusterSnapshotsMessage;
(function (DescribeDBClusterSnapshotsMessage) {
    DescribeDBClusterSnapshotsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBClusterSnapshotsMessage = exports.DescribeDBClusterSnapshotsMessage || (exports.DescribeDBClusterSnapshotsMessage = {}));
var DBEngineVersionMessage;
(function (DBEngineVersionMessage) {
    DBEngineVersionMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBEngineVersionMessage = exports.DBEngineVersionMessage || (exports.DBEngineVersionMessage = {}));
var DescribeDBEngineVersionsMessage;
(function (DescribeDBEngineVersionsMessage) {
    DescribeDBEngineVersionsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBEngineVersionsMessage = exports.DescribeDBEngineVersionsMessage || (exports.DescribeDBEngineVersionsMessage = {}));
var DBInstanceAutomatedBackupMessage;
(function (DBInstanceAutomatedBackupMessage) {
    DBInstanceAutomatedBackupMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceAutomatedBackupMessage = exports.DBInstanceAutomatedBackupMessage || (exports.DBInstanceAutomatedBackupMessage = {}));
var DescribeDBInstanceAutomatedBackupsMessage;
(function (DescribeDBInstanceAutomatedBackupsMessage) {
    DescribeDBInstanceAutomatedBackupsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBInstanceAutomatedBackupsMessage = exports.DescribeDBInstanceAutomatedBackupsMessage || (exports.DescribeDBInstanceAutomatedBackupsMessage = {}));
var DBInstanceMessage;
(function (DBInstanceMessage) {
    DBInstanceMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBInstanceMessage = exports.DBInstanceMessage || (exports.DBInstanceMessage = {}));
var DescribeDBInstancesMessage;
(function (DescribeDBInstancesMessage) {
    DescribeDBInstancesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBInstancesMessage = exports.DescribeDBInstancesMessage || (exports.DescribeDBInstancesMessage = {}));
var DescribeDBLogFilesMessage;
(function (DescribeDBLogFilesMessage) {
    DescribeDBLogFilesMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBLogFilesMessage = exports.DescribeDBLogFilesMessage || (exports.DescribeDBLogFilesMessage = {}));
var DescribeDBLogFilesDetails;
(function (DescribeDBLogFilesDetails) {
    DescribeDBLogFilesDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBLogFilesDetails = exports.DescribeDBLogFilesDetails || (exports.DescribeDBLogFilesDetails = {}));
var DescribeDBLogFilesResponse;
(function (DescribeDBLogFilesResponse) {
    DescribeDBLogFilesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBLogFilesResponse = exports.DescribeDBLogFilesResponse || (exports.DescribeDBLogFilesResponse = {}));
var DBParameterGroupsMessage;
(function (DBParameterGroupsMessage) {
    DBParameterGroupsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroupsMessage = exports.DBParameterGroupsMessage || (exports.DBParameterGroupsMessage = {}));
var DescribeDBParameterGroupsMessage;
(function (DescribeDBParameterGroupsMessage) {
    DescribeDBParameterGroupsMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBParameterGroupsMessage = exports.DescribeDBParameterGroupsMessage || (exports.DescribeDBParameterGroupsMessage = {}));
var DBParameterGroupDetails;
(function (DBParameterGroupDetails) {
    DBParameterGroupDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBParameterGroupDetails = exports.DBParameterGroupDetails || (exports.DBParameterGroupDetails = {}));
var DescribeDBParametersMessage;
(function (DescribeDBParametersMessage) {
    DescribeDBParametersMessage.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBParametersMessage = exports.DescribeDBParametersMessage || (exports.DescribeDBParametersMessage = {}));
var DescribeDBProxiesRequest;
(function (DescribeDBProxiesRequest) {
    DescribeDBProxiesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxiesRequest = exports.DescribeDBProxiesRequest || (exports.DescribeDBProxiesRequest = {}));
var DescribeDBProxiesResponse;
(function (DescribeDBProxiesResponse) {
    DescribeDBProxiesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxiesResponse = exports.DescribeDBProxiesResponse || (exports.DescribeDBProxiesResponse = {}));
var DescribeDBProxyEndpointsRequest;
(function (DescribeDBProxyEndpointsRequest) {
    DescribeDBProxyEndpointsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxyEndpointsRequest = exports.DescribeDBProxyEndpointsRequest || (exports.DescribeDBProxyEndpointsRequest = {}));
var DescribeDBProxyEndpointsResponse;
(function (DescribeDBProxyEndpointsResponse) {
    DescribeDBProxyEndpointsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxyEndpointsResponse = exports.DescribeDBProxyEndpointsResponse || (exports.DescribeDBProxyEndpointsResponse = {}));
var DescribeDBProxyTargetGroupsRequest;
(function (DescribeDBProxyTargetGroupsRequest) {
    DescribeDBProxyTargetGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxyTargetGroupsRequest = exports.DescribeDBProxyTargetGroupsRequest || (exports.DescribeDBProxyTargetGroupsRequest = {}));
var ConnectionPoolConfigurationInfo;
(function (ConnectionPoolConfigurationInfo) {
    ConnectionPoolConfigurationInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConnectionPoolConfigurationInfo = exports.ConnectionPoolConfigurationInfo || (exports.ConnectionPoolConfigurationInfo = {}));
var DBProxyTargetGroup;
(function (DBProxyTargetGroup) {
    DBProxyTargetGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DBProxyTargetGroup = exports.DBProxyTargetGroup || (exports.DBProxyTargetGroup = {}));
var DescribeDBProxyTargetGroupsResponse;
(function (DescribeDBProxyTargetGroupsResponse) {
    DescribeDBProxyTargetGroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxyTargetGroupsResponse = exports.DescribeDBProxyTargetGroupsResponse || (exports.DescribeDBProxyTargetGroupsResponse = {}));
var DescribeDBProxyTargetsRequest;
(function (DescribeDBProxyTargetsRequest) {
    DescribeDBProxyTargetsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDBProxyTargetsRequest = exports.DescribeDBProxyTargetsRequest || (exports.DescribeDBProxyTargetsRequest = {}));
